{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2092ef92-81df-4128-a1fa-f728c558bac0",
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import tqdm as tqdm_module\n",
    "import requests\n",
    "import re\n",
    "import json\n",
    "import pbmd_tools as pbmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1e66c109-89bd-4f6f-8d99-89ed18a4be3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "db = 'pubmed'\n",
    "domain = 'https://www.ncbi.nlm.nih.gov/entrez/eutils'\n",
    "nresults = 100\n",
    "query = '\"github.com\"[Title/Abstract] NOT \"github.com\"[Title]'\n",
    "retmode = 'json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e774752a-5e60-4e94-9242-b8a1015a9767",
   "metadata": {},
   "outputs": [],
   "source": [
    "queryLinkSearch = f'{domain}/esearch.fcgi?db={db}&retmax={nresults}&retmode={retmode}&term={query}'\n",
    "response = requests.get(queryLinkSearch)\n",
    "pubmedJson = response.json()\n",
    "results = []\n",
    "for paperId in pubmedJson[\"esearchresult\"][\"idlist\"]:\n",
    "    queryLinkSummary = f'{domain}/esummary.fcgi?db={db}&id={paperId}&retmode={retmode}'\n",
    "    results.append({'paperId': paperId, 'metadata': requests.get(queryLinkSummary).json()})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5b3727b0-68a6-4591-bfec-26764dd2fcfd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36760999</td>\n",
       "      <td>2023-01-24</td>\n",
       "      <td>10.3389/fgene.2023.1082032</td>\n",
       "      <td>Frontiers in genetics</td>\n",
       "      <td>Identification of functional gene modules by i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36759692</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1038/s41598-023-29320-6</td>\n",
       "      <td>Scientific reports</td>\n",
       "      <td>Deep consistency-preserving hash auto-encoders...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36759336</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1093/bib/bbad044</td>\n",
       "      <td>Briefings in bioinformatics</td>\n",
       "      <td>HiConfidence: a novel approach uncovering the ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36756726</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1111/1755-0998.13765</td>\n",
       "      <td>Molecular ecology resources</td>\n",
       "      <td>Long amplicon HiFi sequencing for mitochondria...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36756173</td>\n",
       "      <td>2022-10-23</td>\n",
       "      <td>10.1016/j.csbj.2022.10.016</td>\n",
       "      <td>Computational and structural biotechnology jou...</td>\n",
       "      <td>Multi-task deep autoencoder to predict Alzheim...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36711471</td>\n",
       "      <td>2023-01-20</td>\n",
       "      <td>10.1101/2023.01.17.524477</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Combining protein sequences and structures wit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36710930</td>\n",
       "      <td>2023-01-13</td>\n",
       "      <td>10.3389/fnmol.2022.1037565</td>\n",
       "      <td>Frontiers in molecular neuroscience</td>\n",
       "      <td>Inferring cell developmental stage-specific ln...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36710872</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>10.7717/peerj.14706</td>\n",
       "      <td>PeerJ</td>\n",
       "      <td>A clustering method for small scRNA-seq data b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36709790</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>10.1016/j.ymeth.2023.01.006</td>\n",
       "      <td>Methods (San Diego, Calif.)</td>\n",
       "      <td>Predicting latent lncRNA and cancer metastatic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36709557</td>\n",
       "      <td>2023-01-15</td>\n",
       "      <td>10.1016/j.cmpb.2023.107355</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>High-throughput 3DRA segmentation of brain vas...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                          DOI  \\\n",
       "0   36760999  2023-01-24   10.3389/fgene.2023.1082032   \n",
       "1   36759692  2023-02-09   10.1038/s41598-023-29320-6   \n",
       "2   36759336  2023-02-09          10.1093/bib/bbad044   \n",
       "3   36756726  2023-02-09      10.1111/1755-0998.13765   \n",
       "4   36756173  2022-10-23   10.1016/j.csbj.2022.10.016   \n",
       "..       ...         ...                          ...   \n",
       "95  36711471  2023-01-20    10.1101/2023.01.17.524477   \n",
       "96  36710930  2023-01-13   10.3389/fnmol.2022.1037565   \n",
       "97  36710872  2023-01-23          10.7717/peerj.14706   \n",
       "98  36709790  2023-01-26  10.1016/j.ymeth.2023.01.006   \n",
       "99  36709557  2023-01-15   10.1016/j.cmpb.2023.107355   \n",
       "\n",
       "                                              Journal  \\\n",
       "0                               Frontiers in genetics   \n",
       "1                                  Scientific reports   \n",
       "2                         Briefings in bioinformatics   \n",
       "3                         Molecular ecology resources   \n",
       "4   Computational and structural biotechnology jou...   \n",
       "..                                                ...   \n",
       "95          bioRxiv : the preprint server for biology   \n",
       "96                Frontiers in molecular neuroscience   \n",
       "97                                              PeerJ   \n",
       "98                        Methods (San Diego, Calif.)   \n",
       "99       Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                Title  \n",
       "0   Identification of functional gene modules by i...  \n",
       "1   Deep consistency-preserving hash auto-encoders...  \n",
       "2   HiConfidence: a novel approach uncovering the ...  \n",
       "3   Long amplicon HiFi sequencing for mitochondria...  \n",
       "4   Multi-task deep autoencoder to predict Alzheim...  \n",
       "..                                                ...  \n",
       "95  Combining protein sequences and structures wit...  \n",
       "96  Inferring cell developmental stage-specific ln...  \n",
       "97  A clustering method for small scRNA-seq data b...  \n",
       "98  Predicting latent lncRNA and cancer metastatic...  \n",
       "99  High-throughput 3DRA segmentation of brain vas...  \n",
       "\n",
       "[100 rows x 5 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = pd.DataFrame.from_records(results)\n",
    "df1[\"result\"] = df1[\"metadata\"].apply(lambda x: x[\"result\"])\n",
    "df1[\"PMID\"] = df1[\"paperId\"]\n",
    "df1[\"PubDate\"] = [d[k][\"sortpubdate\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\"]\n",
    "df1[\"PubDate\"] = df1[\"PubDate\"].apply(lambda x: datetime.strptime(x, \"%Y/%m/%d %H:%M\").strftime(\"%Y-%m-%d\") if x else None)\n",
    "df1[\"DOI\"] = [d[k][\"articleids\"][i][\"value\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\" for i in range(len(d[k][\"articleids\"])) if d[k][\"articleids\"][i][\"idtype\"] == \"doi\"] \n",
    "df1[\"Journal\"] = [d[k][\"fulljournalname\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\"]\n",
    "df1[\"Title\"] = [d[k][\"title\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\"]\n",
    "df1 = df1.drop([\"paperId\", \"result\", \"metadata\"], axis=1)\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "01753df2-f9cb-4557-b676-9bdb17d56207",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_abstract(PMID):\n",
    "    queryLinkSearch = f'{domain}/efetch.fcgi?db={db}&id={PMID}&retmax={nresults}&retmode={retmode}&rettype=abstract'\n",
    "    response = requests.get(queryLinkSearch)\n",
    "    return response.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb9e5ea9-f932-41aa-8a6e-db1628a84b8c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36760999</td>\n",
       "      <td>2023-01-24</td>\n",
       "      <td>10.3389/fgene.2023.1082032</td>\n",
       "      <td>Frontiers in genetics</td>\n",
       "      <td>Identification of functional gene modules by i...</td>\n",
       "      <td>Front Genet. 2023 Jan 24;14:1082032. doi: 10.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36759692</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1038/s41598-023-29320-6</td>\n",
       "      <td>Scientific reports</td>\n",
       "      <td>Deep consistency-preserving hash auto-encoders...</td>\n",
       "      <td>Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36759336</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1093/bib/bbad044</td>\n",
       "      <td>Briefings in bioinformatics</td>\n",
       "      <td>HiConfidence: a novel approach uncovering the ...</td>\n",
       "      <td>Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36756726</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1111/1755-0998.13765</td>\n",
       "      <td>Molecular ecology resources</td>\n",
       "      <td>Long amplicon HiFi sequencing for mitochondria...</td>\n",
       "      <td>Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36756173</td>\n",
       "      <td>2022-10-23</td>\n",
       "      <td>10.1016/j.csbj.2022.10.016</td>\n",
       "      <td>Computational and structural biotechnology jou...</td>\n",
       "      <td>Multi-task deep autoencoder to predict Alzheim...</td>\n",
       "      <td>Comput Struct Biotechnol J. 2022 Oct 23;20:576...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36711471</td>\n",
       "      <td>2023-01-20</td>\n",
       "      <td>10.1101/2023.01.17.524477</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Combining protein sequences and structures wit...</td>\n",
       "      <td>bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36710930</td>\n",
       "      <td>2023-01-13</td>\n",
       "      <td>10.3389/fnmol.2022.1037565</td>\n",
       "      <td>Frontiers in molecular neuroscience</td>\n",
       "      <td>Inferring cell developmental stage-specific ln...</td>\n",
       "      <td>Front Mol Neurosci. 2023 Jan 13;15:1037565. do...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36710872</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>10.7717/peerj.14706</td>\n",
       "      <td>PeerJ</td>\n",
       "      <td>A clustering method for small scRNA-seq data b...</td>\n",
       "      <td>PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/pee...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36709790</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>10.1016/j.ymeth.2023.01.006</td>\n",
       "      <td>Methods (San Diego, Calif.)</td>\n",
       "      <td>Predicting latent lncRNA and cancer metastatic...</td>\n",
       "      <td>Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.y...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36709557</td>\n",
       "      <td>2023-01-15</td>\n",
       "      <td>10.1016/j.cmpb.2023.107355</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>High-throughput 3DRA segmentation of brain vas...</td>\n",
       "      <td>Comput Methods Programs Biomed. 2023 Jan 15;23...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                          DOI  \\\n",
       "0   36760999  2023-01-24   10.3389/fgene.2023.1082032   \n",
       "1   36759692  2023-02-09   10.1038/s41598-023-29320-6   \n",
       "2   36759336  2023-02-09          10.1093/bib/bbad044   \n",
       "3   36756726  2023-02-09      10.1111/1755-0998.13765   \n",
       "4   36756173  2022-10-23   10.1016/j.csbj.2022.10.016   \n",
       "..       ...         ...                          ...   \n",
       "95  36711471  2023-01-20    10.1101/2023.01.17.524477   \n",
       "96  36710930  2023-01-13   10.3389/fnmol.2022.1037565   \n",
       "97  36710872  2023-01-23          10.7717/peerj.14706   \n",
       "98  36709790  2023-01-26  10.1016/j.ymeth.2023.01.006   \n",
       "99  36709557  2023-01-15   10.1016/j.cmpb.2023.107355   \n",
       "\n",
       "                                              Journal  \\\n",
       "0                               Frontiers in genetics   \n",
       "1                                  Scientific reports   \n",
       "2                         Briefings in bioinformatics   \n",
       "3                         Molecular ecology resources   \n",
       "4   Computational and structural biotechnology jou...   \n",
       "..                                                ...   \n",
       "95          bioRxiv : the preprint server for biology   \n",
       "96                Frontiers in molecular neuroscience   \n",
       "97                                              PeerJ   \n",
       "98                        Methods (San Diego, Calif.)   \n",
       "99       Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                Title  \\\n",
       "0   Identification of functional gene modules by i...   \n",
       "1   Deep consistency-preserving hash auto-encoders...   \n",
       "2   HiConfidence: a novel approach uncovering the ...   \n",
       "3   Long amplicon HiFi sequencing for mitochondria...   \n",
       "4   Multi-task deep autoencoder to predict Alzheim...   \n",
       "..                                                ...   \n",
       "95  Combining protein sequences and structures wit...   \n",
       "96  Inferring cell developmental stage-specific ln...   \n",
       "97  A clustering method for small scRNA-seq data b...   \n",
       "98  Predicting latent lncRNA and cancer metastatic...   \n",
       "99  High-throughput 3DRA segmentation of brain vas...   \n",
       "\n",
       "                                             Abstract  \n",
       "0   Front Genet. 2023 Jan 24;14:1082032. doi: 10.3...  \n",
       "1   Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s...  \n",
       "2   Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1...  \n",
       "3   Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755...  \n",
       "4   Comput Struct Biotechnol J. 2022 Oct 23;20:576...  \n",
       "..                                                ...  \n",
       "95  bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 1...  \n",
       "96  Front Mol Neurosci. 2023 Jan 13;15:1037565. do...  \n",
       "97  PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/pee...  \n",
       "98  Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.y...  \n",
       "99  Comput Methods Programs Biomed. 2023 Jan 15;23...  \n",
       "\n",
       "[100 rows x 6 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rate_limit = 3\n",
    "request_count = 0\n",
    "\n",
    "for id_iter in df1[\"PMID\"]:  \n",
    "    \n",
    "    if request_count >= rate_limit:\n",
    "        \n",
    "        time.sleep(1)\n",
    "        request_count = 0\n",
    "        \n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"Abstract\"] = get_abstract(id_iter)\n",
    "    \n",
    "    request_count += 1\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cc71745-20f7-4756-8bc7-99e3923b5f5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_link(df, PMID):\n",
    "    \n",
    "    regex = [\"https:\\/[^ ]*[./]github.com[^\\n .,)]*\"]\n",
    "    links_with_point = ''\n",
    "    \n",
    "    for rgx in regex:\n",
    "        if (links_with_point == '') and re.search(rgx, str(df.loc[df[\"PMID\"] == PMID, \"Abstract\"].values[0]), re.IGNORECASE):\n",
    "                links_with_point = re.findall(rgx, str(df.loc[df[\"PMID\"] == PMID, \"Abstract\"].values[0]), re.IGNORECASE)\n",
    "    links = []\n",
    "    for link in links_with_point :\n",
    "        links.append(link) \n",
    "        \n",
    "    return links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3bfcf0e-2e76-4e9f-b8b5-f305e6c58d0d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36760999</td>\n",
       "      <td>2023-01-24</td>\n",
       "      <td>10.3389/fgene.2023.1082032</td>\n",
       "      <td>Frontiers in genetics</td>\n",
       "      <td>Identification of functional gene modules by i...</td>\n",
       "      <td>Front Genet. 2023 Jan 24;14:1082032. doi: 10.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36759692</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1038/s41598-023-29320-6</td>\n",
       "      <td>Scientific reports</td>\n",
       "      <td>Deep consistency-preserving hash auto-encoders...</td>\n",
       "      <td>Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36759336</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1093/bib/bbad044</td>\n",
       "      <td>Briefings in bioinformatics</td>\n",
       "      <td>HiConfidence: a novel approach uncovering the ...</td>\n",
       "      <td>Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36756726</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1111/1755-0998.13765</td>\n",
       "      <td>Molecular ecology resources</td>\n",
       "      <td>Long amplicon HiFi sequencing for mitochondria...</td>\n",
       "      <td>Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36756173</td>\n",
       "      <td>2022-10-23</td>\n",
       "      <td>10.1016/j.csbj.2022.10.016</td>\n",
       "      <td>Computational and structural biotechnology jou...</td>\n",
       "      <td>Multi-task deep autoencoder to predict Alzheim...</td>\n",
       "      <td>Comput Struct Biotechnol J. 2022 Oct 23;20:576...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36711471</td>\n",
       "      <td>2023-01-20</td>\n",
       "      <td>10.1101/2023.01.17.524477</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Combining protein sequences and structures wit...</td>\n",
       "      <td>bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36710930</td>\n",
       "      <td>2023-01-13</td>\n",
       "      <td>10.3389/fnmol.2022.1037565</td>\n",
       "      <td>Frontiers in molecular neuroscience</td>\n",
       "      <td>Inferring cell developmental stage-specific ln...</td>\n",
       "      <td>Front Mol Neurosci. 2023 Jan 13;15:1037565. do...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36710872</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>10.7717/peerj.14706</td>\n",
       "      <td>PeerJ</td>\n",
       "      <td>A clustering method for small scRNA-seq data b...</td>\n",
       "      <td>PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/pee...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36709790</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>10.1016/j.ymeth.2023.01.006</td>\n",
       "      <td>Methods (San Diego, Calif.)</td>\n",
       "      <td>Predicting latent lncRNA and cancer metastatic...</td>\n",
       "      <td>Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.y...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36709557</td>\n",
       "      <td>2023-01-15</td>\n",
       "      <td>10.1016/j.cmpb.2023.107355</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>High-throughput 3DRA segmentation of brain vas...</td>\n",
       "      <td>Comput Methods Programs Biomed. 2023 Jan 15;23...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                          DOI  \\\n",
       "0   36760999  2023-01-24   10.3389/fgene.2023.1082032   \n",
       "1   36759692  2023-02-09   10.1038/s41598-023-29320-6   \n",
       "2   36759336  2023-02-09          10.1093/bib/bbad044   \n",
       "3   36756726  2023-02-09      10.1111/1755-0998.13765   \n",
       "4   36756173  2022-10-23   10.1016/j.csbj.2022.10.016   \n",
       "..       ...         ...                          ...   \n",
       "95  36711471  2023-01-20    10.1101/2023.01.17.524477   \n",
       "96  36710930  2023-01-13   10.3389/fnmol.2022.1037565   \n",
       "97  36710872  2023-01-23          10.7717/peerj.14706   \n",
       "98  36709790  2023-01-26  10.1016/j.ymeth.2023.01.006   \n",
       "99  36709557  2023-01-15   10.1016/j.cmpb.2023.107355   \n",
       "\n",
       "                                              Journal  \\\n",
       "0                               Frontiers in genetics   \n",
       "1                                  Scientific reports   \n",
       "2                         Briefings in bioinformatics   \n",
       "3                         Molecular ecology resources   \n",
       "4   Computational and structural biotechnology jou...   \n",
       "..                                                ...   \n",
       "95          bioRxiv : the preprint server for biology   \n",
       "96                Frontiers in molecular neuroscience   \n",
       "97                                              PeerJ   \n",
       "98                        Methods (San Diego, Calif.)   \n",
       "99       Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                Title  \\\n",
       "0   Identification of functional gene modules by i...   \n",
       "1   Deep consistency-preserving hash auto-encoders...   \n",
       "2   HiConfidence: a novel approach uncovering the ...   \n",
       "3   Long amplicon HiFi sequencing for mitochondria...   \n",
       "4   Multi-task deep autoencoder to predict Alzheim...   \n",
       "..                                                ...   \n",
       "95  Combining protein sequences and structures wit...   \n",
       "96  Inferring cell developmental stage-specific ln...   \n",
       "97  A clustering method for small scRNA-seq data b...   \n",
       "98  Predicting latent lncRNA and cancer metastatic...   \n",
       "99  High-throughput 3DRA segmentation of brain vas...   \n",
       "\n",
       "                                             Abstract  \n",
       "0   Front Genet. 2023 Jan 24;14:1082032. doi: 10.3...  \n",
       "1   Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s...  \n",
       "2   Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1...  \n",
       "3   Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755...  \n",
       "4   Comput Struct Biotechnol J. 2022 Oct 23;20:576...  \n",
       "..                                                ...  \n",
       "95  bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 1...  \n",
       "96  Front Mol Neurosci. 2023 Jan 13;15:1037565. do...  \n",
       "97  PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/pee...  \n",
       "98  Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.y...  \n",
       "99  Comput Methods Programs Biomed. 2023 Jan 15;23...  \n",
       "\n",
       "[100 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ad2241b-a3df-41b6-b311-6f21cd1ed1e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_link(link_array):\n",
    "    links = []\n",
    "    for link in link_array:\n",
    "        if link != \"\":\n",
    "            if not link.startswith(\"https://\"):\n",
    "                link = \"https://\" + link\n",
    "            if link[-2] == \")\":\n",
    "                link = link[:-2]\n",
    "            if link[-1] == \")\" or link[-1] == \".\" or link[-1] == \",\" or link[-1] == \",\":\n",
    "                link = link[:-1]\n",
    "        links.append(link)\n",
    "        \n",
    "    return links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "689f13df-c3f9-4c33-985e-18e2ba522cfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_phrase_with_link(df, PMID):\n",
    "    \n",
    "    regex = [\"\\.[^.]*[./]github[^ ]* [^.]*[^ ]*\\.\", \"\\.[^.]*[./]github[^ ]*\", \"[^.]*github[^ ]*\", \"[a-zA-Z0-9 .,/:\\'\\\"!?]{101}github[^ ]*[a-zA-Z0-9 .,/:\\'\\\"!?()]{100}\"]\n",
    "    phrase_with_point = ''\n",
    "    \n",
    "    for rgx in regex:\n",
    "        if (phrase_with_point == \"\") and re.search(rgx, str(df.loc[df[\"PMID\"] == PMID, \"Abstract\"].values[0]), re.IGNORECASE):\n",
    "            phrase_with_point = re.findall(rgx, str(df.loc[df[\"PMID\"] == PMID, \"Abstract\"].values[0]), re.IGNORECASE)\n",
    "    phrases = []\n",
    "    for phrase in phrase_with_point :\n",
    "        phrase = phrase[2:]\n",
    "        phrases.append(phrase)\n",
    "        \n",
    "    return ' '.join(phrases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "baaeface-4d91-46c1-8ee4-f972180a1fd1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "      <th>Abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36760999</td>\n",
       "      <td>2023-01-24</td>\n",
       "      <td>10.3389/fgene.2023.1082032</td>\n",
       "      <td>https://github.com/free1234hm/CLAM</td>\n",
       "      <td>https://github.com/free1234hm/CLAM</td>\n",
       "      <td>Frontiers in genetics</td>\n",
       "      <td>Identification of functional gene modules by integrating multi-omics data and known molecular interactions.</td>\n",
       "      <td>We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.</td>\n",
       "      <td>Front Genet. 2023 Jan 24;14:1082032. doi: 10.3389/fgene.2023.1082032. \\neCollection 2023.\\n\\nIdentification of functional gene modules by integrating multi-omics data and \\nknown molecular interactions.\\n\\nChen X(1)(2), Han M(2), Li Y(3), Li X(2), Zhang J(2), Zhu Y(1)(2).\\n\\nAuthor information:\\n(1)Basic Medical School, Anhui Medical University, Hefei, China.\\n(2)National Center for Protein Sciences (Beijing), Beijing Proteome Research \\nCenter, Beijing Institute of Lifeomics, Beijing, China.\\n(3)Central Research Laboratory, Peking Union Medical College Hospital, Chinese \\nAcademy of Medical Sciences and Peking Union Medical College, Beijing, China.\\n\\nMulti-omics data integration has emerged as a promising approach to identify \\npatient subgroups. However, in terms of grouping genes (or gene products) into \\nco-expression modules, data integration methods suffer from two main drawbacks. \\nFirst, most existing methods only consider genes or samples measured in all \\ndifferent datasets. Second, known molecular interactions (e.g., transcriptional \\nregulatory interactions, protein-protein interactions and biological pathways) \\ncannot be utilized to assist in module detection. Herein, we present a novel \\ndata integration framework, Correlation-based Local Approximation of Membership \\n(CLAM), which provides two methodological innovations to address these \\nlimitations: 1) constructing a trans-omics neighborhood matrix by integrating \\nmulti-omics datasets and known molecular interactions, and 2) using a local \\napproximation procedure to define gene modules from the matrix. Applying \\nCorrelation-based Local Approximation of Membership to human colorectal cancer \\n(CRC) and mouse B-cell differentiation multi-omics data obtained from The Cancer \\nGenome Atlas (TCGA), Clinical Proteomics Tumor Analysis Consortium (CPTAC), Gene \\nExpression Omnibus (GEO) and ProteomeXchange database, we demonstrated its \\nsuperior ability to recover biologically relevant modules and gene ontology (GO) \\nterms. Further investigation of the colorectal cancer modules revealed numerous \\ntranscription factors and KEGG pathways that played crucial roles in colorectal \\ncancer progression. Module-based survival analysis constructed four \\nsurvival-related networks in which pairwise gene correlations were significantly \\ncorrelated with colorectal cancer patient survival. Overall, the series of \\nevaluations demonstrated the great potential of Correlation-based Local \\nApproximation of Membership for identifying modular biomarkers for complex \\ndiseases. We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.\\n\\nDOI: 10.3389/fgene.2023.1082032\\nPMCID: PMC9902936\\nPMID: 36760999\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36759692</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1038/s41598-023-29320-6</td>\n",
       "      <td>https://github.com/Socrates023/DCPHA</td>\n",
       "      <td>https://github.com/Socrates023/DCPHA</td>\n",
       "      <td>Scientific reports</td>\n",
       "      <td>Deep consistency-preserving hash auto-encoders for neuroimage cross-modal retrieval.</td>\n",
       "      <td>We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .</td>\n",
       "      <td>Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s41598-023-29320-6.\\n\\nDeep consistency-preserving hash auto-encoders for neuroimage cross-modal \\nretrieval.\\n\\nWang X(1), Zeng X(2).\\n\\nAuthor information:\\n(1)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China.\\n(2)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China. zengxh@cqupt.edu.cn.\\n\\nCross-modal hashing is an efficient method to embed high-dimensional \\nheterogeneous modal feature descriptors into a consistency-preserving Hamming \\nspace with low-dimensional. Most existing cross-modal hashing methods have been \\nable to bridge the heterogeneous modality gap, but there are still two \\nchallenges resulting in limited retrieval accuracy: (1) ignoring the continuous \\nsimilarity of samples on manifold; (2) lack of discriminability of hash codes \\nwith the same semantics. To cope with these problems, we propose a Deep \\nConsistency-Preserving Hash Auto-encoders model, called DCPHA, based on the \\nmulti-manifold property of the feature distribution. Specifically, DCPHA \\nconsists of a pair of asymmetric auto-encoders and two semantics-preserving \\nattention branches working in the encoding and decoding stages, respectively. \\nWhen the number of input medical image modalities is greater than 2, the encoder \\nis a multiple pseudo-Siamese network designed to extract specific modality \\nfeatures of different medical image modalities. In addition, we define the \\ncontinuous similarity of heterogeneous and homogeneous samples on Riemann \\nmanifold from the perspective of multiple sub-manifolds, respectively, and the \\ntwo constraints, i.e., multi-semantic consistency and multi-manifold \\nsimilarity-preserving, are embedded in the learning of hash codes to obtain \\nhigh-quality hash codes with consistency-preserving. The extensive experiments \\nshow that the proposed DCPHA has the most stable and state-of-the-art \\nperformance. We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .\\n\\nÂ© 2023. The Author(s).\\n\\nDOI: 10.1038/s41598-023-29320-6\\nPMID: 36759692\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36759336</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1093/bib/bbad044</td>\n",
       "      <td>https://github.com/victorykobets/HiConfidence</td>\n",
       "      <td>https://github.com/victorykobets/HiConfidence</td>\n",
       "      <td>Briefings in bioinformatics</td>\n",
       "      <td>HiConfidence: a novel approach uncovering the biological signal in Hi-C data affected by technical biases.</td>\n",
       "      <td>The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023.</td>\n",
       "      <td>Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1093/bib/bbad044. Online ahead of \\nprint.\\n\\nHiConfidence: a novel approach uncovering the biological signal in Hi-C data \\naffected by technical biases.\\n\\nKobets VA(1), Ulianov SV(2)(3), Galitsyna AA(1)(2)(4), Doronin SA(5), Mikhaleva \\nEA(5), Gelfand MS(1)(4), Shevelyov YY(5), Razin SV(2)(3), Khrameeva EE(1).\\n\\nAuthor information:\\n(1)Skolkovo Institute of Science and Technology, Moscow, 121205, Russia.\\n(2)Institute of Gene Biology, Russian Academy of Sciences, Moscow, 119334, \\nRussia.\\n(3)Faculty of Biology, M.V. Lomonosov Moscow State University, Moscow, 119992, \\nRussia.\\n(4)A.A. Kharkevich Institute for Information Transmission Problems, Russian \\nAcademy of Sciences, Moscow, 127051, Russia.\\n(5)Institute of Molecular Genetics of National Research Centre \"Kurchatov \\nInstitute\", Moscow, 123182, Russia.\\n\\nThe chromatin interaction assays, particularly Hi-C, enable detailed studies of \\ngenome architecture in multiple organisms and model systems, resulting in a \\ndeeper understanding of gene expression regulation mechanisms mediated by \\nepigenetics. However, the analysis and interpretation of Hi-C data remain \\nchallenging due to technical biases, limiting direct comparisons of datasets \\nobtained in different experiments and laboratories. As a result, removing biases \\nfrom Hi-C-generated chromatin contact matrices is a critical data analysis step. \\nOur novel approach, HiConfidence, eliminates biases from the Hi-C data by \\nweighing chromatin contacts according to their consistency between replicates so \\nthat low-quality replicates do not substantially influence the result. The \\nalgorithm is effective for the analysis of global changes in chromatin \\nstructures such as compartments and topologically associating domains. We apply \\nthe HiConfidence approach to several Hi-C datasets with significant technical \\nbiases, that could not be analyzed effectively using existing methods, and \\nobtain meaningful biological conclusions. In particular, HiConfidence aids in \\nthe study of how changes in histone acetylation pattern affect chromatin \\norganization in Drosophila melanogaster S2 cells. The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023. Published by Oxford University Press.\\n\\nDOI: 10.1093/bib/bbad044\\nPMID: 36759336\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36756726</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1111/1755-0998.13765</td>\n",
       "      <td>https://github.com/Caizf-script/HQGR</td>\n",
       "      <td>https://github.com/Caizf-script/HQGR</td>\n",
       "      <td>Molecular ecology resources</td>\n",
       "      <td>Long amplicon HiFi sequencing for mitochondrial DNA genomes.</td>\n",
       "      <td>The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.</td>\n",
       "      <td>Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755-0998.13765. Online ahead of \\nprint.\\n\\nLong amplicon HiFi sequencing for mitochondrial DNA genomes.\\n\\nCai ZF(#)(1)(2), Hu JY(#)(3), Yin TT(#)(2), Wang D(3), Shen QK(2), Ma C(2)(4), \\nOu DQ(5), Xu MM(2)(4), Shi X(2)(4), Li QL(1)(2), Wu RN(2), Ajuma L(2)(4), Adeola \\nAC(2), Zhang YP(1)(2)(4), Peng MS(2)(4).\\n\\nAuthor information:\\n(1)State Key Laboratory for Conservation and Utilization of Bio-resources in \\nYunnan, Yunnan University, Kunming, China.\\n(2)State Key Laboratory of Genetic Resources and Evolution &amp; Yunnan Laboratory \\nof Molecular Biology of Domestic Animals, Kunming Institute of Zoology, Chinese \\nAcademy of Sciences, Kunming, China.\\n(3)School of Software, Yunnan University, Kunming, China.\\n(4)University of Chinese Academy of Sciences, Beijing, China.\\n(5)Department of Anesthesiology, First Affiliated Hospital of Kunming Medical \\nUniversity, Kunming, China.\\n(#)Contributed equally\\n\\nLong-read sequencing technology is a powerful approach with application in \\nvarious genetic and genomic researches. Herein, we developed the pipeline for \\nlong amplicon high-fidelity (HiFi) sequencing and then applied it for sequencing \\nmitochondrial DNA (mtDNA) genomes from pools of 79 Tibetan Mastiffs. We \\namplified the mtDNA genome with long-range PCR using two pairs of primers. Two \\nrounds of circular consensus sequencing (CCS) were conducted and their accuracy \\nwas evaluated. The results indicate that the second round of CCS can improve the \\naccuracy of HiFi reads. In addition, the analysis of 79 high-quality mtDNA \\ngenomes shows the Tibetan Mastiffs from outside of the Tibetan Plateau \\nexperienced hybridization with other dogs. The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.\\n\\nThis article is protected by copyright. All rights reserved.\\n\\nDOI: 10.1111/1755-0998.13765\\nPMID: 36756726\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36756173</td>\n",
       "      <td>2022-10-23</td>\n",
       "      <td>10.1016/j.csbj.2022.10.016</td>\n",
       "      <td>https://github.com/lichen-lab/MTAE</td>\n",
       "      <td>https://github.com/lichen-lab/MTAE</td>\n",
       "      <td>Computational and structural biotechnology journal</td>\n",
       "      <td>Multi-task deep autoencoder to predict Alzheimer's disease progression using temporal DNA methylation data in peripheral blood.</td>\n",
       "      <td>\\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).</td>\n",
       "      <td>Comput Struct Biotechnol J. 2022 Oct 23;20:5761-5774. doi: \\n10.1016/j.csbj.2022.10.016. eCollection 2022.\\n\\nMulti-task deep autoencoder to predict Alzheimer's disease progression using \\ntemporal DNA methylation data in peripheral blood.\\n\\nChen L(1), Saykin AJ(2), Yao B(3), Zhao F(1); Alzheimerâ€™s Disease Neuroimaging \\nInitiative (ADNI).\\n\\nAuthor information:\\n(1)Department of Biostatistics, University of Florida, Gainesville, FL 32603, \\nUnited States.\\n(2)Department of Radiology and Imaging Sciences, Indiana University School of \\nMedicine, Indianapolis, IN 46202, United States.\\n(3)Department of Human Genetics, Emory University, Atlanta, GA 30322, United \\nStates.\\n\\nTraditional approaches for diagnosing Alzheimer's disease (AD) such as brain \\nimaging and cerebrospinal fluid are invasive and expensive. It is desirable to \\ndevelop a useful diagnostic tool by exploiting biomarkers obtained from \\nperipheral tissues due to their noninvasive and easily accessible \\ncharacteristics. However, the capacity of using DNA methylation data in \\nperipheral blood for predicting AD progression is rarely known. It is also \\nchallenging to develop an efficient prediction model considering the complex and \\nhigh-dimensional DNA methylation data in a longitudinal study. Here, we develop \\ntwo multi-task deep autoencoders, which are based on the convolutional \\nautoencoder and long short-term memory autoencoder to learn the compressed \\nfeature representation by jointly minimizing the reconstruction error and \\nmaximizing the prediction accuracy. By benchmarking on longitudinal DNA \\nmethylation data collected from the peripheral blood in Alzheimer's Disease \\nNeuroimaging Initiative, we demonstrate that the proposed multi-task deep \\nautoencoders outperform state-of-the-art machine learning approaches for both \\npredicting AD progression and reconstructing the temporal DNA methylation \\nprofiles. In addition, the proposed multi-task deep autoencoders can predict AD \\nprogression accurately using only the historical DNA methylation data and the \\nperformance is further improved by including all temporal DNA methylation data. \\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).\\n\\nDOI: 10.1016/j.csbj.2022.10.016\\nPMCID: PMC9619306\\nPMID: 36756173\\n\\nConflict of interest statement: The authors declare that they have no known \\ncompeting financial interests or personal relationships that could have appeared \\nto influence the work reported in this paper.\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36711471</td>\n",
       "      <td>2023-01-20</td>\n",
       "      <td>10.1101/2023.01.17.524477</td>\n",
       "      <td>https://github.com/jianlin-cheng/TransFun</td>\n",
       "      <td>https://github.com/jianlin-cheng/TransFun</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Combining protein sequences and structures with transformers and equivariant graph neural networks to predict protein function.</td>\n",
       "      <td>AVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.</td>\n",
       "      <td>bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 10.1101/2023.01.17.524477. \\nPreprint.\\n\\nCombining protein sequences and structures with transformers and equivariant \\ngraph neural networks to predict protein function.\\n\\nBoadu F, Cao H, Cheng J.\\n\\nMOTIVATION: Millions of protein sequences have been generated by numerous genome \\nand transcriptome sequencing projects. However, experimentally determining the \\nfunction of the proteins is still a time consuming, low-throughput, and \\nexpensive process, leading to a large protein sequence-function gap. Therefore, \\nit is important to develop computational methods to accurately predict protein \\nfunction to fill the gap. Even though many methods have been developed to use \\nprotein sequences as input to predict function, much fewer methods leverage \\nprotein structures in protein function prediction because there was lack of \\naccurate protein structures for most proteins until recently.\\nRESULTS: We developed TransFun - a method using a transformer-based protein \\nlanguage model and 3D-equivariant graph neural networks to distill information \\nfrom both protein sequences and structures to predict protein function. It \\nextracts feature embeddings from protein sequences using a pre-trained protein \\nlanguage model (ESM) via transfer learning and combines them with 3D structures \\nof proteins predicted by AlphaFold2 through equivariant graph neural networks. \\nBenchmarked on the CAFA3 test dataset and a new test dataset, TransFun \\noutperforms several state-of-the-art methods, indicating the language model and \\n3D-equivariant graph neural networks are effective methods to leverage protein \\nsequences and structures to improve protein function prediction. Combining \\nTransFun predictions and sequence similarity-based predictions can further \\nincrease prediction accuracy.\\nAVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.\\n\\nDOI: 10.1101/2023.01.17.524477\\nPMCID: PMC9882282\\nPMID: 36711471\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36710930</td>\n",
       "      <td>2023-01-13</td>\n",
       "      <td>10.3389/fnmol.2022.1037565</td>\n",
       "      <td>https://github.com/linxi159/CDSlncR</td>\n",
       "      <td>https://github.com/linxi159/CDSlncR</td>\n",
       "      <td>Frontiers in molecular neuroscience</td>\n",
       "      <td>Inferring cell developmental stage-specific lncRNA regulation in the developing human neocortex with CDSlncR.</td>\n",
       "      <td>CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.</td>\n",
       "      <td>Front Mol Neurosci. 2023 Jan 13;15:1037565. doi: 10.3389/fnmol.2022.1037565. \\neCollection 2022.\\n\\nInferring cell developmental stage-specific lncRNA regulation in the developing \\nhuman neocortex with CDSlncR.\\n\\nHuang M(1)(2), Ma J(1)(3), Zhang J(3).\\n\\nAuthor information:\\n(1)Department of Automation, Xiamen University, Xiamen, China.\\n(2)Department of Computer Science, University of Tsukuba, Tsukuba, Japan.\\n(3)School of Engineering, Dali University, Dali, China.\\n\\nNoncoding RNAs (ncRNAs) occupy ~98% of the transcriptome in human, and are \\nusually not translated into proteins. Among ncRNAs, long non-coding RNAs \\n(lncRNAs, &gt;200 nucleotides) are important regulators to modulate gene \\nexpression, and are involved in many biological processes (e.g., cell \\ndevelopment). To study lncRNA regulation, many computational approaches or tools \\nhave been proposed by using bulk transcriptomics data. Nevertheless, previous \\nbulk data-driven methods are mostly limited to explore the lncRNA regulation \\nregarding all of cells, instead of the lncRNA regulation specific to cell \\ndevelopmental stages. Fortunately, recent advance in single-cell sequencing data \\nhas provided a way to investigate cell developmental stage-specific lncRNA \\nregulation. In this work, we present a novel computational method, CDSlncR (Cell \\nDevelopmental Stage-specific lncRNA regulation), which combines putative \\nlncRNA-target binding information with single-cell transcriptomics data to infer \\ncell developmental stage-specific lncRNA regulation. For each cell developmental \\nstage, CDSlncR constructs a cell developmental stage-specific lncRNA regulatory \\nnetwork in the cell developmental stage. To illustrate the effectiveness of \\nCDSlncR, we apply CDSlncR into single-cell transcriptomics data of the \\ndeveloping human neocortex for exploring lncRNA regulation across different \\nhuman neocortex developmental stages. Network analysis shows that the lncRNA \\nregulation is unique in each developmental stage of human neocortex. As a case \\nstudy, we also perform particular analysis on the cell developmental \\nstage-specific lncRNA regulation related to 18 known lncRNA biomarkers in autism \\nspectrum disorder. Finally, the comparison result indicates that CDSlncR is an \\neffective method for predicting cell developmental stage-specific lncRNA \\ntargets. CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.\\n\\nDOI: 10.3389/fnmol.2022.1037565\\nPMCID: PMC9880432\\nPMID: 36710930\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36710872</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>10.7717/peerj.14706</td>\n",
       "      <td>https://github.com/ningzilan/SSWD</td>\n",
       "      <td>https://github.com/ningzilan/SSWD</td>\n",
       "      <td>PeerJ</td>\n",
       "      <td>A clustering method for small scRNA-seq data based on subspace and weighted distance.</td>\n",
       "      <td>SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.</td>\n",
       "      <td>PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/peerj.14706. eCollection 2023.\\n\\nA clustering method for small scRNA-seq data based on subspace and weighted \\ndistance.\\n\\nNing Z(1)(2), Dai Z(1), Zhang H(2), Chen Y(1), Yuan Z(1).\\n\\nAuthor information:\\n(1)Hunan Engineering &amp; Technology Research Centre for Agricultural Big Data \\nAnalysis &amp; Decision-Making, Hunan Agricultural University, Changsha, Hunan, \\nChina.\\n(2)Hunan Agricultural University, College of Information and Intelligence, \\nChangsha, Hunan, China.\\n\\nBACKGROUND: Identifying the cell types using unsupervised methods is essential \\nfor scRNA-seq research. However, conventional similarity measures introduce \\nchallenges to single-cell data clustering because of the high dimensional, high \\nnoise, and high dropout.\\nMETHODS: We proposed a clustering method for small ScRNA-seq data based on \\nSubspace and Weighted Distance (SSWD), which follows the assumption that the \\nsets of gene subspace composed of similar density-distributing genes can better \\ndistinguish cell groups. To accurately capture the intrinsic relationship among \\ncells or genes, a new distance metric that combines Euclidean and Pearson \\ndistance through a weighting strategy was proposed. The relative \\nCalinski-Harabasz (CH) index was used to estimate the cluster numbers instead of \\nthe CH index because it is comparable across degrees of freedom.\\nRESULTS: We compared SSWD with seven prevailing methods on eight publicly \\nscRNA-seq datasets. The experimental results show that the SSWD has better \\nclustering accuracy and the partitioning ability of cell groups. SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.\\n\\nDOI: 10.7717/peerj.14706\\nPMCID: PMC9879162\\nPMID: 36710872 [Indexed for MEDLINE]\\n\\nConflict of interest statement: The authors declare there are no competing \\ninterests.\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36709790</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>10.1016/j.ymeth.2023.01.006</td>\n",
       "      <td>https://github.com/zhuyuan-cug/VGAE-LCME</td>\n",
       "      <td>https://github.com/zhuyuan-cug/VGAE-LCME</td>\n",
       "      <td>Methods (San Diego, Calif.)</td>\n",
       "      <td>Predicting latent lncRNA and cancer metastatic event associations via variational graph auto-encoder.</td>\n",
       "      <td>The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc.</td>\n",
       "      <td>Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.ymeth.2023.01.006. Online ahead of \\nprint.\\n\\nPredicting latent lncRNA and cancer metastatic event associations via \\nvariational graph auto-encoder.\\n\\nZhu Y(1), Zhang F(2), Zhang S(3), Yi M(4).\\n\\nAuthor information:\\n(1)School of Automation, China University of Geosciences, 388 Lumo Road, \\nHongshan District, 430074, Wuhan, Hubei, China; Hubei Key Laboratory of Advanced \\nControl and Intelligent Automation for Complex Systems, 388 Lumo Road, Hongshan \\nDistrict, 430074, Wuhan, Hubei, China; Engineering Research Center of \\nIntelligent Technology for Geo-Exploration, 388 Lumo Road, Hongshan District, \\n430074, Wuhan, Hubei, China.\\n(2)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China.\\n(3)College of Life Science and Health, Wuhan University of Science and \\nTechnology, 974 Heping Avenue, Qingshan District, 430081, Wuhan, Hubei, China. \\nElectronic address: shihuazhang@wust.edu.cn.\\n(4)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China. Electronic address: \\nmingyi@cug.edu.cn.\\n\\nLong non-coding RNA (lncRNA) are shown to be closely associated with cancer \\nmetastatic events (CME, e.g., cancer cell invasion, intravasation, \\nextravasation, proliferation) that collaboratively accelerate malignant cancer \\nspread and cause high mortality rate in patients. Clinical trials may accurately \\nuncover the relationships between lncRNAs and CMEs; however, it is \\ntime-consuming and expensive. With the accumulation of data, there is an urgent \\nneed to find efficient ways to identify these relationships. Herein, a graph \\nembedding representation-based predictor (VGEA-LCME) for exploring latent \\nlncRNA-CME associations is introduced. In VGEA-LCME, a heterogeneous combined \\nnetwork is constructed by integrating similarity and linkage matrix that can \\nmaintain internal and external characteristics of networks, and a variational \\ngraph auto-encoder serves as a feature generator to represent arbitrary lncRNA \\nand CME pair. The final robustness predicted result is obtained by ensemble \\nclassifier strategy via cross-validation. Experimental comparisons and \\nliterature verification show better remarkable performance of VGEA-LCME, \\nalthough the similarities between CMEs are challenging to calculate. In \\naddition, VGEA-LCME can further identify organ-specific CMEs. To the best of our \\nknowledge, this is the first computational attempt to discover the potential \\nrelationships between lncRNAs and CMEs. It may provide support and new insight \\nfor guiding experimental research of metastatic cancers. The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc. All rights reserved.\\n\\nDOI: 10.1016/j.ymeth.2023.01.006\\nPMID: 36709790\\n\\nConflict of interest statement: Declaration of Competing Interest The authors \\ndeclare that there are no conflict of interests, we do not have any possible \\nconflicts of interest.\\n\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36709557</td>\n",
       "      <td>2023-01-15</td>\n",
       "      <td>10.1016/j.cmpb.2023.107355</td>\n",
       "      <td>https://github.com/cistib/vessel-aneurysm-segmentation</td>\n",
       "      <td>https://github.com/cistib/vessel-aneurysm-segmentation</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>High-throughput 3DRA segmentation of brain vasculature and aneurysms using deep learning.</td>\n",
       "      <td>Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023.</td>\n",
       "      <td>Comput Methods Programs Biomed. 2023 Jan 15;230:107355. doi: \\n10.1016/j.cmpb.2023.107355. Online ahead of print.\\n\\nHigh-throughput 3DRA segmentation of brain vasculature and aneurysms using deep \\nlearning.\\n\\nLin F(1), Xia Y(2), Song S(1), Ravikumar N(1), Frangi AF(3).\\n\\nAuthor information:\\n(1)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK.\\n(2)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK. Electronic address: \\ny.xia@leeds.ac.uk.\\n(3)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK; Leeds Institute for \\nCardiovascular and Metabolic Medicine (LICAMM), School of Medicine, University \\nof Leeds, Leeds LS2 9JT, UK; Medical Imaging Research Center (MIRC), \\nCardiovascular Science and Electronic Engineering Departments, KU Leuven, \\nLeuven, Belgium; Alan Turing Institute, London, UK.\\n\\nBACKGROUND AND OBJECTIVES: Automatic segmentation of the cerebral vasculature \\nand aneurysms facilitates incidental detection of aneurysms. The assessment of \\naneurysm rupture risk assists with pre-operative treatment planning and enables \\nin-silico investigation of cerebral hemodynamics within and in the vicinity of \\naneurysms. However, ensuring precise and robust segmentation of cerebral vessels \\nand aneurysms in neuroimaging modalities such as three-dimensional rotational \\nangiography (3DRA) is challenging. The vasculature constitutes a small \\nproportion of the image volume, resulting in a large class imbalance (relative \\nto surrounding brain tissue). Additionally, aneurysms and vessels have similar \\nimage/appearance characteristics, making it challenging to distinguish the \\naneurysm sac from the vessel lumen.\\nMETHODS: We propose a novel multi-class convolutional neural network to tackle \\nthese challenges and facilitate the automatic segmentation of cerebral vessels \\nand aneurysms in 3DRA images. The proposed model is trained and evaluated on an \\ninternal multi-center dataset and an external publicly available challenge \\ndataset.\\nRESULTS: On the internal clinical dataset, our method consistently outperformed \\nseveral state-of-the-art approaches for vessel and aneurysm segmentation, \\nachieving an average Dice score of 0.81 (0.15 higher than nnUNet) and an average \\nsurface-to-surface error of 0.20Â mm (less than the in-plane resolution \\n(0.35Â mm/pixel)) for aneurysm segmentation; and an average Dice score of 0.91 \\nand average surface-to-surface error of 0.25Â mm for vessel segmentation. In 223 \\ncases of a clinical dataset, our method accurately segmented 190 aneurysm cases.\\nCONCLUSIONS: The proposed approach can help address class imbalance problems and \\ninter-class interference problems in multi-class segmentation. Besides, this \\nmethod performs consistently on clinical datasets from four different sources \\nand the generated results are qualified for hemodynamic simulation. Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023. Published by Elsevier B.V.\\n\\nDOI: 10.1016/j.cmpb.2023.107355\\nPMID: 36709557\\n\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                          DOI  \\\n",
       "0   36760999  2023-01-24   10.3389/fgene.2023.1082032   \n",
       "1   36759692  2023-02-09   10.1038/s41598-023-29320-6   \n",
       "2   36759336  2023-02-09          10.1093/bib/bbad044   \n",
       "3   36756726  2023-02-09      10.1111/1755-0998.13765   \n",
       "4   36756173  2022-10-23   10.1016/j.csbj.2022.10.016   \n",
       "..       ...         ...                          ...   \n",
       "95  36711471  2023-01-20    10.1101/2023.01.17.524477   \n",
       "96  36710930  2023-01-13   10.3389/fnmol.2022.1037565   \n",
       "97  36710872  2023-01-23          10.7717/peerj.14706   \n",
       "98  36709790  2023-01-26  10.1016/j.ymeth.2023.01.006   \n",
       "99  36709557  2023-01-15   10.1016/j.cmpb.2023.107355   \n",
       "\n",
       "                                           GitHub_link_raw  \\\n",
       "0                       https://github.com/free1234hm/CLAM   \n",
       "1                     https://github.com/Socrates023/DCPHA   \n",
       "2            https://github.com/victorykobets/HiConfidence   \n",
       "3                     https://github.com/Caizf-script/HQGR   \n",
       "4                       https://github.com/lichen-lab/MTAE   \n",
       "..                                                     ...   \n",
       "95               https://github.com/jianlin-cheng/TransFun   \n",
       "96                     https://github.com/linxi159/CDSlncR   \n",
       "97                       https://github.com/ningzilan/SSWD   \n",
       "98                https://github.com/zhuyuan-cug/VGAE-LCME   \n",
       "99  https://github.com/cistib/vessel-aneurysm-segmentation   \n",
       "\n",
       "                                         GitHub_link_clean  \\\n",
       "0                       https://github.com/free1234hm/CLAM   \n",
       "1                     https://github.com/Socrates023/DCPHA   \n",
       "2            https://github.com/victorykobets/HiConfidence   \n",
       "3                     https://github.com/Caizf-script/HQGR   \n",
       "4                       https://github.com/lichen-lab/MTAE   \n",
       "..                                                     ...   \n",
       "95               https://github.com/jianlin-cheng/TransFun   \n",
       "96                     https://github.com/linxi159/CDSlncR   \n",
       "97                       https://github.com/ningzilan/SSWD   \n",
       "98                https://github.com/zhuyuan-cug/VGAE-LCME   \n",
       "99  https://github.com/cistib/vessel-aneurysm-segmentation   \n",
       "\n",
       "                                               Journal  \\\n",
       "0                                Frontiers in genetics   \n",
       "1                                   Scientific reports   \n",
       "2                          Briefings in bioinformatics   \n",
       "3                          Molecular ecology resources   \n",
       "4   Computational and structural biotechnology journal   \n",
       "..                                                 ...   \n",
       "95           bioRxiv : the preprint server for biology   \n",
       "96                 Frontiers in molecular neuroscience   \n",
       "97                                               PeerJ   \n",
       "98                         Methods (San Diego, Calif.)   \n",
       "99        Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                                                                                              Title  \\\n",
       "0                       Identification of functional gene modules by integrating multi-omics data and known molecular interactions.   \n",
       "1                                              Deep consistency-preserving hash auto-encoders for neuroimage cross-modal retrieval.   \n",
       "2                        HiConfidence: a novel approach uncovering the biological signal in Hi-C data affected by technical biases.   \n",
       "3                                                                      Long amplicon HiFi sequencing for mitochondrial DNA genomes.   \n",
       "4   Multi-task deep autoencoder to predict Alzheimer's disease progression using temporal DNA methylation data in peripheral blood.   \n",
       "..                                                                                                                              ...   \n",
       "95  Combining protein sequences and structures with transformers and equivariant graph neural networks to predict protein function.   \n",
       "96                    Inferring cell developmental stage-specific lncRNA regulation in the developing human neocortex with CDSlncR.   \n",
       "97                                            A clustering method for small scRNA-seq data based on subspace and weighted distance.   \n",
       "98                            Predicting latent lncRNA and cancer metastatic event associations via variational graph auto-encoder.   \n",
       "99                                        High-throughput 3DRA segmentation of brain vasculature and aneurysms using deep learning.   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                            Phrase  \\\n",
       "0                                                                                                                       We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.   \n",
       "1                                                                                                                                                                                                                                             We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .   \n",
       "2                                                                                                                                                                                                              The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023.   \n",
       "3   The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.   \n",
       "4                                                                                                                                                                                                                                                    \\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).   \n",
       "..                                                                                                                                                                                                                                                                                                                             ...   \n",
       "95                                                                                                                                                                                          AVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.   \n",
       "96                                                                                                                                                                                                                           CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.   \n",
       "97                                                                                                                                                                                                                                             SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.   \n",
       "98                                                                                                                                                                                                         The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc.   \n",
       "99                                                                                                                                                                                                                                Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023.   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Abstract  \n",
       "0                                                                                                                                                                                                             Front Genet. 2023 Jan 24;14:1082032. doi: 10.3389/fgene.2023.1082032. \\neCollection 2023.\\n\\nIdentification of functional gene modules by integrating multi-omics data and \\nknown molecular interactions.\\n\\nChen X(1)(2), Han M(2), Li Y(3), Li X(2), Zhang J(2), Zhu Y(1)(2).\\n\\nAuthor information:\\n(1)Basic Medical School, Anhui Medical University, Hefei, China.\\n(2)National Center for Protein Sciences (Beijing), Beijing Proteome Research \\nCenter, Beijing Institute of Lifeomics, Beijing, China.\\n(3)Central Research Laboratory, Peking Union Medical College Hospital, Chinese \\nAcademy of Medical Sciences and Peking Union Medical College, Beijing, China.\\n\\nMulti-omics data integration has emerged as a promising approach to identify \\npatient subgroups. However, in terms of grouping genes (or gene products) into \\nco-expression modules, data integration methods suffer from two main drawbacks. \\nFirst, most existing methods only consider genes or samples measured in all \\ndifferent datasets. Second, known molecular interactions (e.g., transcriptional \\nregulatory interactions, protein-protein interactions and biological pathways) \\ncannot be utilized to assist in module detection. Herein, we present a novel \\ndata integration framework, Correlation-based Local Approximation of Membership \\n(CLAM), which provides two methodological innovations to address these \\nlimitations: 1) constructing a trans-omics neighborhood matrix by integrating \\nmulti-omics datasets and known molecular interactions, and 2) using a local \\napproximation procedure to define gene modules from the matrix. Applying \\nCorrelation-based Local Approximation of Membership to human colorectal cancer \\n(CRC) and mouse B-cell differentiation multi-omics data obtained from The Cancer \\nGenome Atlas (TCGA), Clinical Proteomics Tumor Analysis Consortium (CPTAC), Gene \\nExpression Omnibus (GEO) and ProteomeXchange database, we demonstrated its \\nsuperior ability to recover biologically relevant modules and gene ontology (GO) \\nterms. Further investigation of the colorectal cancer modules revealed numerous \\ntranscription factors and KEGG pathways that played crucial roles in colorectal \\ncancer progression. Module-based survival analysis constructed four \\nsurvival-related networks in which pairwise gene correlations were significantly \\ncorrelated with colorectal cancer patient survival. Overall, the series of \\nevaluations demonstrated the great potential of Correlation-based Local \\nApproximation of Membership for identifying modular biomarkers for complex \\ndiseases. We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.\\n\\nDOI: 10.3389/fgene.2023.1082032\\nPMCID: PMC9902936\\nPMID: 36760999\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n  \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s41598-023-29320-6.\\n\\nDeep consistency-preserving hash auto-encoders for neuroimage cross-modal \\nretrieval.\\n\\nWang X(1), Zeng X(2).\\n\\nAuthor information:\\n(1)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China.\\n(2)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China. zengxh@cqupt.edu.cn.\\n\\nCross-modal hashing is an efficient method to embed high-dimensional \\nheterogeneous modal feature descriptors into a consistency-preserving Hamming \\nspace with low-dimensional. Most existing cross-modal hashing methods have been \\nable to bridge the heterogeneous modality gap, but there are still two \\nchallenges resulting in limited retrieval accuracy: (1) ignoring the continuous \\nsimilarity of samples on manifold; (2) lack of discriminability of hash codes \\nwith the same semantics. To cope with these problems, we propose a Deep \\nConsistency-Preserving Hash Auto-encoders model, called DCPHA, based on the \\nmulti-manifold property of the feature distribution. Specifically, DCPHA \\nconsists of a pair of asymmetric auto-encoders and two semantics-preserving \\nattention branches working in the encoding and decoding stages, respectively. \\nWhen the number of input medical image modalities is greater than 2, the encoder \\nis a multiple pseudo-Siamese network designed to extract specific modality \\nfeatures of different medical image modalities. In addition, we define the \\ncontinuous similarity of heterogeneous and homogeneous samples on Riemann \\nmanifold from the perspective of multiple sub-manifolds, respectively, and the \\ntwo constraints, i.e., multi-semantic consistency and multi-manifold \\nsimilarity-preserving, are embedded in the learning of hash codes to obtain \\nhigh-quality hash codes with consistency-preserving. The extensive experiments \\nshow that the proposed DCPHA has the most stable and state-of-the-art \\nperformance. We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .\\n\\nÂ© 2023. The Author(s).\\n\\nDOI: 10.1038/s41598-023-29320-6\\nPMID: 36759692\\n\\n  \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1093/bib/bbad044. Online ahead of \\nprint.\\n\\nHiConfidence: a novel approach uncovering the biological signal in Hi-C data \\naffected by technical biases.\\n\\nKobets VA(1), Ulianov SV(2)(3), Galitsyna AA(1)(2)(4), Doronin SA(5), Mikhaleva \\nEA(5), Gelfand MS(1)(4), Shevelyov YY(5), Razin SV(2)(3), Khrameeva EE(1).\\n\\nAuthor information:\\n(1)Skolkovo Institute of Science and Technology, Moscow, 121205, Russia.\\n(2)Institute of Gene Biology, Russian Academy of Sciences, Moscow, 119334, \\nRussia.\\n(3)Faculty of Biology, M.V. Lomonosov Moscow State University, Moscow, 119992, \\nRussia.\\n(4)A.A. Kharkevich Institute for Information Transmission Problems, Russian \\nAcademy of Sciences, Moscow, 127051, Russia.\\n(5)Institute of Molecular Genetics of National Research Centre \"Kurchatov \\nInstitute\", Moscow, 123182, Russia.\\n\\nThe chromatin interaction assays, particularly Hi-C, enable detailed studies of \\ngenome architecture in multiple organisms and model systems, resulting in a \\ndeeper understanding of gene expression regulation mechanisms mediated by \\nepigenetics. However, the analysis and interpretation of Hi-C data remain \\nchallenging due to technical biases, limiting direct comparisons of datasets \\nobtained in different experiments and laboratories. As a result, removing biases \\nfrom Hi-C-generated chromatin contact matrices is a critical data analysis step. \\nOur novel approach, HiConfidence, eliminates biases from the Hi-C data by \\nweighing chromatin contacts according to their consistency between replicates so \\nthat low-quality replicates do not substantially influence the result. The \\nalgorithm is effective for the analysis of global changes in chromatin \\nstructures such as compartments and topologically associating domains. We apply \\nthe HiConfidence approach to several Hi-C datasets with significant technical \\nbiases, that could not be analyzed effectively using existing methods, and \\nobtain meaningful biological conclusions. In particular, HiConfidence aids in \\nthe study of how changes in histone acetylation pattern affect chromatin \\norganization in Drosophila melanogaster S2 cells. The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023. Published by Oxford University Press.\\n\\nDOI: 10.1093/bib/bbad044\\nPMID: 36759336\\n\\n  \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755-0998.13765. Online ahead of \\nprint.\\n\\nLong amplicon HiFi sequencing for mitochondrial DNA genomes.\\n\\nCai ZF(#)(1)(2), Hu JY(#)(3), Yin TT(#)(2), Wang D(3), Shen QK(2), Ma C(2)(4), \\nOu DQ(5), Xu MM(2)(4), Shi X(2)(4), Li QL(1)(2), Wu RN(2), Ajuma L(2)(4), Adeola \\nAC(2), Zhang YP(1)(2)(4), Peng MS(2)(4).\\n\\nAuthor information:\\n(1)State Key Laboratory for Conservation and Utilization of Bio-resources in \\nYunnan, Yunnan University, Kunming, China.\\n(2)State Key Laboratory of Genetic Resources and Evolution & Yunnan Laboratory \\nof Molecular Biology of Domestic Animals, Kunming Institute of Zoology, Chinese \\nAcademy of Sciences, Kunming, China.\\n(3)School of Software, Yunnan University, Kunming, China.\\n(4)University of Chinese Academy of Sciences, Beijing, China.\\n(5)Department of Anesthesiology, First Affiliated Hospital of Kunming Medical \\nUniversity, Kunming, China.\\n(#)Contributed equally\\n\\nLong-read sequencing technology is a powerful approach with application in \\nvarious genetic and genomic researches. Herein, we developed the pipeline for \\nlong amplicon high-fidelity (HiFi) sequencing and then applied it for sequencing \\nmitochondrial DNA (mtDNA) genomes from pools of 79 Tibetan Mastiffs. We \\namplified the mtDNA genome with long-range PCR using two pairs of primers. Two \\nrounds of circular consensus sequencing (CCS) were conducted and their accuracy \\nwas evaluated. The results indicate that the second round of CCS can improve the \\naccuracy of HiFi reads. In addition, the analysis of 79 high-quality mtDNA \\ngenomes shows the Tibetan Mastiffs from outside of the Tibetan Plateau \\nexperienced hybridization with other dogs. The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.\\n\\nThis article is protected by copyright. All rights reserved.\\n\\nDOI: 10.1111/1755-0998.13765\\nPMID: 36756726\\n\\n  \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Comput Struct Biotechnol J. 2022 Oct 23;20:5761-5774. doi: \\n10.1016/j.csbj.2022.10.016. eCollection 2022.\\n\\nMulti-task deep autoencoder to predict Alzheimer's disease progression using \\ntemporal DNA methylation data in peripheral blood.\\n\\nChen L(1), Saykin AJ(2), Yao B(3), Zhao F(1); Alzheimerâ€™s Disease Neuroimaging \\nInitiative (ADNI).\\n\\nAuthor information:\\n(1)Department of Biostatistics, University of Florida, Gainesville, FL 32603, \\nUnited States.\\n(2)Department of Radiology and Imaging Sciences, Indiana University School of \\nMedicine, Indianapolis, IN 46202, United States.\\n(3)Department of Human Genetics, Emory University, Atlanta, GA 30322, United \\nStates.\\n\\nTraditional approaches for diagnosing Alzheimer's disease (AD) such as brain \\nimaging and cerebrospinal fluid are invasive and expensive. It is desirable to \\ndevelop a useful diagnostic tool by exploiting biomarkers obtained from \\nperipheral tissues due to their noninvasive and easily accessible \\ncharacteristics. However, the capacity of using DNA methylation data in \\nperipheral blood for predicting AD progression is rarely known. It is also \\nchallenging to develop an efficient prediction model considering the complex and \\nhigh-dimensional DNA methylation data in a longitudinal study. Here, we develop \\ntwo multi-task deep autoencoders, which are based on the convolutional \\nautoencoder and long short-term memory autoencoder to learn the compressed \\nfeature representation by jointly minimizing the reconstruction error and \\nmaximizing the prediction accuracy. By benchmarking on longitudinal DNA \\nmethylation data collected from the peripheral blood in Alzheimer's Disease \\nNeuroimaging Initiative, we demonstrate that the proposed multi-task deep \\nautoencoders outperform state-of-the-art machine learning approaches for both \\npredicting AD progression and reconstructing the temporal DNA methylation \\nprofiles. In addition, the proposed multi-task deep autoencoders can predict AD \\nprogression accurately using only the historical DNA methylation data and the \\nperformance is further improved by including all temporal DNA methylation data. \\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).\\n\\nDOI: 10.1016/j.csbj.2022.10.016\\nPMCID: PMC9619306\\nPMID: 36756173\\n\\nConflict of interest statement: The authors declare that they have no known \\ncompeting financial interests or personal relationships that could have appeared \\nto influence the work reported in this paper.\\n\\n  \n",
       "..                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              ...  \n",
       "95                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 10.1101/2023.01.17.524477. \\nPreprint.\\n\\nCombining protein sequences and structures with transformers and equivariant \\ngraph neural networks to predict protein function.\\n\\nBoadu F, Cao H, Cheng J.\\n\\nMOTIVATION: Millions of protein sequences have been generated by numerous genome \\nand transcriptome sequencing projects. However, experimentally determining the \\nfunction of the proteins is still a time consuming, low-throughput, and \\nexpensive process, leading to a large protein sequence-function gap. Therefore, \\nit is important to develop computational methods to accurately predict protein \\nfunction to fill the gap. Even though many methods have been developed to use \\nprotein sequences as input to predict function, much fewer methods leverage \\nprotein structures in protein function prediction because there was lack of \\naccurate protein structures for most proteins until recently.\\nRESULTS: We developed TransFun - a method using a transformer-based protein \\nlanguage model and 3D-equivariant graph neural networks to distill information \\nfrom both protein sequences and structures to predict protein function. It \\nextracts feature embeddings from protein sequences using a pre-trained protein \\nlanguage model (ESM) via transfer learning and combines them with 3D structures \\nof proteins predicted by AlphaFold2 through equivariant graph neural networks. \\nBenchmarked on the CAFA3 test dataset and a new test dataset, TransFun \\noutperforms several state-of-the-art methods, indicating the language model and \\n3D-equivariant graph neural networks are effective methods to leverage protein \\nsequences and structures to improve protein function prediction. Combining \\nTransFun predictions and sequence similarity-based predictions can further \\nincrease prediction accuracy.\\nAVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.\\n\\nDOI: 10.1101/2023.01.17.524477\\nPMCID: PMC9882282\\nPMID: 36711471\\n\\n  \n",
       "96                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Front Mol Neurosci. 2023 Jan 13;15:1037565. doi: 10.3389/fnmol.2022.1037565. \\neCollection 2022.\\n\\nInferring cell developmental stage-specific lncRNA regulation in the developing \\nhuman neocortex with CDSlncR.\\n\\nHuang M(1)(2), Ma J(1)(3), Zhang J(3).\\n\\nAuthor information:\\n(1)Department of Automation, Xiamen University, Xiamen, China.\\n(2)Department of Computer Science, University of Tsukuba, Tsukuba, Japan.\\n(3)School of Engineering, Dali University, Dali, China.\\n\\nNoncoding RNAs (ncRNAs) occupy ~98% of the transcriptome in human, and are \\nusually not translated into proteins. Among ncRNAs, long non-coding RNAs \\n(lncRNAs, >200 nucleotides) are important regulators to modulate gene \\nexpression, and are involved in many biological processes (e.g., cell \\ndevelopment). To study lncRNA regulation, many computational approaches or tools \\nhave been proposed by using bulk transcriptomics data. Nevertheless, previous \\nbulk data-driven methods are mostly limited to explore the lncRNA regulation \\nregarding all of cells, instead of the lncRNA regulation specific to cell \\ndevelopmental stages. Fortunately, recent advance in single-cell sequencing data \\nhas provided a way to investigate cell developmental stage-specific lncRNA \\nregulation. In this work, we present a novel computational method, CDSlncR (Cell \\nDevelopmental Stage-specific lncRNA regulation), which combines putative \\nlncRNA-target binding information with single-cell transcriptomics data to infer \\ncell developmental stage-specific lncRNA regulation. For each cell developmental \\nstage, CDSlncR constructs a cell developmental stage-specific lncRNA regulatory \\nnetwork in the cell developmental stage. To illustrate the effectiveness of \\nCDSlncR, we apply CDSlncR into single-cell transcriptomics data of the \\ndeveloping human neocortex for exploring lncRNA regulation across different \\nhuman neocortex developmental stages. Network analysis shows that the lncRNA \\nregulation is unique in each developmental stage of human neocortex. As a case \\nstudy, we also perform particular analysis on the cell developmental \\nstage-specific lncRNA regulation related to 18 known lncRNA biomarkers in autism \\nspectrum disorder. Finally, the comparison result indicates that CDSlncR is an \\neffective method for predicting cell developmental stage-specific lncRNA \\ntargets. CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.\\n\\nDOI: 10.3389/fnmol.2022.1037565\\nPMCID: PMC9880432\\nPMID: 36710930\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n  \n",
       "97                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/peerj.14706. eCollection 2023.\\n\\nA clustering method for small scRNA-seq data based on subspace and weighted \\ndistance.\\n\\nNing Z(1)(2), Dai Z(1), Zhang H(2), Chen Y(1), Yuan Z(1).\\n\\nAuthor information:\\n(1)Hunan Engineering & Technology Research Centre for Agricultural Big Data \\nAnalysis & Decision-Making, Hunan Agricultural University, Changsha, Hunan, \\nChina.\\n(2)Hunan Agricultural University, College of Information and Intelligence, \\nChangsha, Hunan, China.\\n\\nBACKGROUND: Identifying the cell types using unsupervised methods is essential \\nfor scRNA-seq research. However, conventional similarity measures introduce \\nchallenges to single-cell data clustering because of the high dimensional, high \\nnoise, and high dropout.\\nMETHODS: We proposed a clustering method for small ScRNA-seq data based on \\nSubspace and Weighted Distance (SSWD), which follows the assumption that the \\nsets of gene subspace composed of similar density-distributing genes can better \\ndistinguish cell groups. To accurately capture the intrinsic relationship among \\ncells or genes, a new distance metric that combines Euclidean and Pearson \\ndistance through a weighting strategy was proposed. The relative \\nCalinski-Harabasz (CH) index was used to estimate the cluster numbers instead of \\nthe CH index because it is comparable across degrees of freedom.\\nRESULTS: We compared SSWD with seven prevailing methods on eight publicly \\nscRNA-seq datasets. The experimental results show that the SSWD has better \\nclustering accuracy and the partitioning ability of cell groups. SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.\\n\\nDOI: 10.7717/peerj.14706\\nPMCID: PMC9879162\\nPMID: 36710872 [Indexed for MEDLINE]\\n\\nConflict of interest statement: The authors declare there are no competing \\ninterests.\\n\\n  \n",
       "98                                                                                                                                              Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.ymeth.2023.01.006. Online ahead of \\nprint.\\n\\nPredicting latent lncRNA and cancer metastatic event associations via \\nvariational graph auto-encoder.\\n\\nZhu Y(1), Zhang F(2), Zhang S(3), Yi M(4).\\n\\nAuthor information:\\n(1)School of Automation, China University of Geosciences, 388 Lumo Road, \\nHongshan District, 430074, Wuhan, Hubei, China; Hubei Key Laboratory of Advanced \\nControl and Intelligent Automation for Complex Systems, 388 Lumo Road, Hongshan \\nDistrict, 430074, Wuhan, Hubei, China; Engineering Research Center of \\nIntelligent Technology for Geo-Exploration, 388 Lumo Road, Hongshan District, \\n430074, Wuhan, Hubei, China.\\n(2)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China.\\n(3)College of Life Science and Health, Wuhan University of Science and \\nTechnology, 974 Heping Avenue, Qingshan District, 430081, Wuhan, Hubei, China. \\nElectronic address: shihuazhang@wust.edu.cn.\\n(4)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China. Electronic address: \\nmingyi@cug.edu.cn.\\n\\nLong non-coding RNA (lncRNA) are shown to be closely associated with cancer \\nmetastatic events (CME, e.g., cancer cell invasion, intravasation, \\nextravasation, proliferation) that collaboratively accelerate malignant cancer \\nspread and cause high mortality rate in patients. Clinical trials may accurately \\nuncover the relationships between lncRNAs and CMEs; however, it is \\ntime-consuming and expensive. With the accumulation of data, there is an urgent \\nneed to find efficient ways to identify these relationships. Herein, a graph \\nembedding representation-based predictor (VGEA-LCME) for exploring latent \\nlncRNA-CME associations is introduced. In VGEA-LCME, a heterogeneous combined \\nnetwork is constructed by integrating similarity and linkage matrix that can \\nmaintain internal and external characteristics of networks, and a variational \\ngraph auto-encoder serves as a feature generator to represent arbitrary lncRNA \\nand CME pair. The final robustness predicted result is obtained by ensemble \\nclassifier strategy via cross-validation. Experimental comparisons and \\nliterature verification show better remarkable performance of VGEA-LCME, \\nalthough the similarities between CMEs are challenging to calculate. In \\naddition, VGEA-LCME can further identify organ-specific CMEs. To the best of our \\nknowledge, this is the first computational attempt to discover the potential \\nrelationships between lncRNAs and CMEs. It may provide support and new insight \\nfor guiding experimental research of metastatic cancers. The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc. All rights reserved.\\n\\nDOI: 10.1016/j.ymeth.2023.01.006\\nPMID: 36709790\\n\\nConflict of interest statement: Declaration of Competing Interest The authors \\ndeclare that there are no conflict of interests, we do not have any possible \\nconflicts of interest.\\n\\n  \n",
       "99  Comput Methods Programs Biomed. 2023 Jan 15;230:107355. doi: \\n10.1016/j.cmpb.2023.107355. Online ahead of print.\\n\\nHigh-throughput 3DRA segmentation of brain vasculature and aneurysms using deep \\nlearning.\\n\\nLin F(1), Xia Y(2), Song S(1), Ravikumar N(1), Frangi AF(3).\\n\\nAuthor information:\\n(1)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK.\\n(2)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK. Electronic address: \\ny.xia@leeds.ac.uk.\\n(3)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK; Leeds Institute for \\nCardiovascular and Metabolic Medicine (LICAMM), School of Medicine, University \\nof Leeds, Leeds LS2 9JT, UK; Medical Imaging Research Center (MIRC), \\nCardiovascular Science and Electronic Engineering Departments, KU Leuven, \\nLeuven, Belgium; Alan Turing Institute, London, UK.\\n\\nBACKGROUND AND OBJECTIVES: Automatic segmentation of the cerebral vasculature \\nand aneurysms facilitates incidental detection of aneurysms. The assessment of \\naneurysm rupture risk assists with pre-operative treatment planning and enables \\nin-silico investigation of cerebral hemodynamics within and in the vicinity of \\naneurysms. However, ensuring precise and robust segmentation of cerebral vessels \\nand aneurysms in neuroimaging modalities such as three-dimensional rotational \\nangiography (3DRA) is challenging. The vasculature constitutes a small \\nproportion of the image volume, resulting in a large class imbalance (relative \\nto surrounding brain tissue). Additionally, aneurysms and vessels have similar \\nimage/appearance characteristics, making it challenging to distinguish the \\naneurysm sac from the vessel lumen.\\nMETHODS: We propose a novel multi-class convolutional neural network to tackle \\nthese challenges and facilitate the automatic segmentation of cerebral vessels \\nand aneurysms in 3DRA images. The proposed model is trained and evaluated on an \\ninternal multi-center dataset and an external publicly available challenge \\ndataset.\\nRESULTS: On the internal clinical dataset, our method consistently outperformed \\nseveral state-of-the-art approaches for vessel and aneurysm segmentation, \\nachieving an average Dice score of 0.81 (0.15 higher than nnUNet) and an average \\nsurface-to-surface error of 0.20Â mm (less than the in-plane resolution \\n(0.35Â mm/pixel)) for aneurysm segmentation; and an average Dice score of 0.91 \\nand average surface-to-surface error of 0.25Â mm for vessel segmentation. In 223 \\ncases of a clinical dataset, our method accurately segmented 190 aneurysm cases.\\nCONCLUSIONS: The proposed approach can help address class imbalance problems and \\ninter-class interference problems in multi-class segmentation. Besides, this \\nmethod performs consistently on clinical datasets from four different sources \\nand the generated results are qualified for hemodynamic simulation. Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023. Published by Elsevier B.V.\\n\\nDOI: 10.1016/j.cmpb.2023.107355\\nPMID: 36709557\\n\\n  \n",
       "\n",
       "[100 rows x 9 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for id_iter in df1[\"PMID\"]:\n",
    "    \n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_raw\"] = \", \".join(get_link(df1, id_iter))\n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_clean\"] = \", \".join(clean_link(df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_raw\"]))\n",
    "    df1.loc[df1[\"PMID\"] == id_iter, \"Phrase\"] = str(get_phrase_with_link(df1, id_iter))\n",
    "\n",
    "df1 = df1[[\"PMID\", \"PubDate\", \"DOI\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"Journal\", \"Title\", \"Phrase\", \"Abstract\"]]\n",
    "pd.options.display.max_colwidth = None\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3a959e9b-50e3-4af9-b856-9a8f3d9a963b",
   "metadata": {},
   "outputs": [],
   "source": [
    "df1.to_csv('C:/Users/nadia/pubmed_results.tsv', sep='\\t', index=False,columns=df1.columns[:-1].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b9777db-e89f-4c65-b001-0447e04dddc5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('2022-10-27T15:54:53Z', '2022-10-31T19:16:00Z')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#https://github.com/free1234hm/CLAM\n",
    "\n",
    "owner = \"free1234hm\"\n",
    "repo = \"CLAM\"\n",
    "url = f\"https://api.github.com/repos/{owner}/{repo}\"\n",
    "\n",
    "response = requests.get(url)\n",
    "if response.status_code == 200:\n",
    "    repository_info = response.json()\n",
    "    created_at = repository_info[\"created_at\"]\n",
    "    updated_at = repository_info[\"updated_at\"]\n",
    "created_at, updated_at   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e51f9f8-5318-4972-b166-45f609bde3b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA - ????"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8ced97a-04f3-4aa6-be76-bd0e2fa32274",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\nadia\\AppData\\Local\\Temp\\ipykernel_12348\\3653661987.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_created_at\"], df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_updated_at\"] = created_at, updated_at\n",
      "C:\\Users\\nadia\\AppData\\Local\\Temp\\ipykernel_12348\\3653661987.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_created_at\"], df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_updated_at\"] = created_at, updated_at\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(0   NaN\n",
       " Name: Repo_created_at, dtype: float64,\n",
       " 0   NaN\n",
       " Name: Repo_updated_at, dtype: float64)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "link = \"https://github.com/free1234hm/CLAM\"\n",
    "\n",
    "owner = re.findall(\"\\/[^\\/.]+/\", str(link))[0][1:-1]\n",
    "repo = re.findall(\"\\/[^\\/.]+$\", str(link))[0][1:-1]\n",
    "url = f\"https://api.github.com/repos/{owner}/{repo}\"\n",
    "\n",
    "response = requests.get(url)\n",
    "if response.status_code == 200:\n",
    "    repository_info = response.json()\n",
    "    created_at = repository_info[\"created_at\"]\n",
    "    updated_at = repository_info[\"updated_at\"]\n",
    "else:\n",
    "    created_at = None\n",
    "    updated_at = None\n",
    "\n",
    "df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_created_at\"], df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_updated_at\"] = created_at, updated_at\n",
    "\n",
    "df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_created_at\"], df1.loc[df1[\"PMID\"] == \"36760999\",\"Repo_updated_at\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7de1612c-fb1c-4799-b9da-6281b175b0e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_repo_info(link):\n",
    "    \n",
    "    owner = re.findall(\"\\/[^\\/.]+/\", str(link))[0][1:-1]\n",
    "    repo = re.findall(\"\\/[^\\/.]+$\", str(link))[0][1:-1]\n",
    "    url = f\"https://api.github.com/repos/{owner}/{repo}\"\n",
    "    \n",
    "    response = requests.get(url)\n",
    "    if response.status_code == 200:\n",
    "        repository_info = response.json()\n",
    "        created_at = repository_info[\"created_at\"]\n",
    "        updated_at = repository_info[\"updated_at\"]\n",
    "        return created_at, updated_at\n",
    "    else:\n",
    "        return None, None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fa997c1-3f3c-4022-9499-fde77c2e3b63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(None, None)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_repo_info(df1.loc[df1[\"PMID\"] == \"36760999\",\"GitHub_link_clean\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "766c63bb-4a4b-4b24-b8f1-53fa9f1d70db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "      <th>Abstract</th>\n",
       "      <th>Repo_created_at</th>\n",
       "      <th>Repo_updated_at</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36760999</td>\n",
       "      <td>2023-01-24</td>\n",
       "      <td>10.3389/fgene.2023.1082032</td>\n",
       "      <td>https://github.com/free1234hm/CLAM</td>\n",
       "      <td>https://github.com/free1234hm/CLAM</td>\n",
       "      <td>Frontiers in genetics</td>\n",
       "      <td>Identification of functional gene modules by integrating multi-omics data and known molecular interactions.</td>\n",
       "      <td>We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.</td>\n",
       "      <td>Front Genet. 2023 Jan 24;14:1082032. doi: 10.3389/fgene.2023.1082032. \\neCollection 2023.\\n\\nIdentification of functional gene modules by integrating multi-omics data and \\nknown molecular interactions.\\n\\nChen X(1)(2), Han M(2), Li Y(3), Li X(2), Zhang J(2), Zhu Y(1)(2).\\n\\nAuthor information:\\n(1)Basic Medical School, Anhui Medical University, Hefei, China.\\n(2)National Center for Protein Sciences (Beijing), Beijing Proteome Research \\nCenter, Beijing Institute of Lifeomics, Beijing, China.\\n(3)Central Research Laboratory, Peking Union Medical College Hospital, Chinese \\nAcademy of Medical Sciences and Peking Union Medical College, Beijing, China.\\n\\nMulti-omics data integration has emerged as a promising approach to identify \\npatient subgroups. However, in terms of grouping genes (or gene products) into \\nco-expression modules, data integration methods suffer from two main drawbacks. \\nFirst, most existing methods only consider genes or samples measured in all \\ndifferent datasets. Second, known molecular interactions (e.g., transcriptional \\nregulatory interactions, protein-protein interactions and biological pathways) \\ncannot be utilized to assist in module detection. Herein, we present a novel \\ndata integration framework, Correlation-based Local Approximation of Membership \\n(CLAM), which provides two methodological innovations to address these \\nlimitations: 1) constructing a trans-omics neighborhood matrix by integrating \\nmulti-omics datasets and known molecular interactions, and 2) using a local \\napproximation procedure to define gene modules from the matrix. Applying \\nCorrelation-based Local Approximation of Membership to human colorectal cancer \\n(CRC) and mouse B-cell differentiation multi-omics data obtained from The Cancer \\nGenome Atlas (TCGA), Clinical Proteomics Tumor Analysis Consortium (CPTAC), Gene \\nExpression Omnibus (GEO) and ProteomeXchange database, we demonstrated its \\nsuperior ability to recover biologically relevant modules and gene ontology (GO) \\nterms. Further investigation of the colorectal cancer modules revealed numerous \\ntranscription factors and KEGG pathways that played crucial roles in colorectal \\ncancer progression. Module-based survival analysis constructed four \\nsurvival-related networks in which pairwise gene correlations were significantly \\ncorrelated with colorectal cancer patient survival. Overall, the series of \\nevaluations demonstrated the great potential of Correlation-based Local \\nApproximation of Membership for identifying modular biomarkers for complex \\ndiseases. We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.\\n\\nDOI: 10.3389/fgene.2023.1082032\\nPMCID: PMC9902936\\nPMID: 36760999\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36759692</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1038/s41598-023-29320-6</td>\n",
       "      <td>https://github.com/Socrates023/DCPHA</td>\n",
       "      <td>https://github.com/Socrates023/DCPHA</td>\n",
       "      <td>Scientific reports</td>\n",
       "      <td>Deep consistency-preserving hash auto-encoders for neuroimage cross-modal retrieval.</td>\n",
       "      <td>We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .</td>\n",
       "      <td>Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s41598-023-29320-6.\\n\\nDeep consistency-preserving hash auto-encoders for neuroimage cross-modal \\nretrieval.\\n\\nWang X(1), Zeng X(2).\\n\\nAuthor information:\\n(1)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China.\\n(2)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China. zengxh@cqupt.edu.cn.\\n\\nCross-modal hashing is an efficient method to embed high-dimensional \\nheterogeneous modal feature descriptors into a consistency-preserving Hamming \\nspace with low-dimensional. Most existing cross-modal hashing methods have been \\nable to bridge the heterogeneous modality gap, but there are still two \\nchallenges resulting in limited retrieval accuracy: (1) ignoring the continuous \\nsimilarity of samples on manifold; (2) lack of discriminability of hash codes \\nwith the same semantics. To cope with these problems, we propose a Deep \\nConsistency-Preserving Hash Auto-encoders model, called DCPHA, based on the \\nmulti-manifold property of the feature distribution. Specifically, DCPHA \\nconsists of a pair of asymmetric auto-encoders and two semantics-preserving \\nattention branches working in the encoding and decoding stages, respectively. \\nWhen the number of input medical image modalities is greater than 2, the encoder \\nis a multiple pseudo-Siamese network designed to extract specific modality \\nfeatures of different medical image modalities. In addition, we define the \\ncontinuous similarity of heterogeneous and homogeneous samples on Riemann \\nmanifold from the perspective of multiple sub-manifolds, respectively, and the \\ntwo constraints, i.e., multi-semantic consistency and multi-manifold \\nsimilarity-preserving, are embedded in the learning of hash codes to obtain \\nhigh-quality hash codes with consistency-preserving. The extensive experiments \\nshow that the proposed DCPHA has the most stable and state-of-the-art \\nperformance. We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .\\n\\nÂ© 2023. The Author(s).\\n\\nDOI: 10.1038/s41598-023-29320-6\\nPMID: 36759692\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36759336</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1093/bib/bbad044</td>\n",
       "      <td>https://github.com/victorykobets/HiConfidence</td>\n",
       "      <td>https://github.com/victorykobets/HiConfidence</td>\n",
       "      <td>Briefings in bioinformatics</td>\n",
       "      <td>HiConfidence: a novel approach uncovering the biological signal in Hi-C data affected by technical biases.</td>\n",
       "      <td>The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023.</td>\n",
       "      <td>Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1093/bib/bbad044. Online ahead of \\nprint.\\n\\nHiConfidence: a novel approach uncovering the biological signal in Hi-C data \\naffected by technical biases.\\n\\nKobets VA(1), Ulianov SV(2)(3), Galitsyna AA(1)(2)(4), Doronin SA(5), Mikhaleva \\nEA(5), Gelfand MS(1)(4), Shevelyov YY(5), Razin SV(2)(3), Khrameeva EE(1).\\n\\nAuthor information:\\n(1)Skolkovo Institute of Science and Technology, Moscow, 121205, Russia.\\n(2)Institute of Gene Biology, Russian Academy of Sciences, Moscow, 119334, \\nRussia.\\n(3)Faculty of Biology, M.V. Lomonosov Moscow State University, Moscow, 119992, \\nRussia.\\n(4)A.A. Kharkevich Institute for Information Transmission Problems, Russian \\nAcademy of Sciences, Moscow, 127051, Russia.\\n(5)Institute of Molecular Genetics of National Research Centre \"Kurchatov \\nInstitute\", Moscow, 123182, Russia.\\n\\nThe chromatin interaction assays, particularly Hi-C, enable detailed studies of \\ngenome architecture in multiple organisms and model systems, resulting in a \\ndeeper understanding of gene expression regulation mechanisms mediated by \\nepigenetics. However, the analysis and interpretation of Hi-C data remain \\nchallenging due to technical biases, limiting direct comparisons of datasets \\nobtained in different experiments and laboratories. As a result, removing biases \\nfrom Hi-C-generated chromatin contact matrices is a critical data analysis step. \\nOur novel approach, HiConfidence, eliminates biases from the Hi-C data by \\nweighing chromatin contacts according to their consistency between replicates so \\nthat low-quality replicates do not substantially influence the result. The \\nalgorithm is effective for the analysis of global changes in chromatin \\nstructures such as compartments and topologically associating domains. We apply \\nthe HiConfidence approach to several Hi-C datasets with significant technical \\nbiases, that could not be analyzed effectively using existing methods, and \\nobtain meaningful biological conclusions. In particular, HiConfidence aids in \\nthe study of how changes in histone acetylation pattern affect chromatin \\norganization in Drosophila melanogaster S2 cells. The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023. Published by Oxford University Press.\\n\\nDOI: 10.1093/bib/bbad044\\nPMID: 36759336\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36756726</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1111/1755-0998.13765</td>\n",
       "      <td>https://github.com/Caizf-script/HQGR</td>\n",
       "      <td>https://github.com/Caizf-script/HQGR</td>\n",
       "      <td>Molecular ecology resources</td>\n",
       "      <td>Long amplicon HiFi sequencing for mitochondrial DNA genomes.</td>\n",
       "      <td>The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.</td>\n",
       "      <td>Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755-0998.13765. Online ahead of \\nprint.\\n\\nLong amplicon HiFi sequencing for mitochondrial DNA genomes.\\n\\nCai ZF(#)(1)(2), Hu JY(#)(3), Yin TT(#)(2), Wang D(3), Shen QK(2), Ma C(2)(4), \\nOu DQ(5), Xu MM(2)(4), Shi X(2)(4), Li QL(1)(2), Wu RN(2), Ajuma L(2)(4), Adeola \\nAC(2), Zhang YP(1)(2)(4), Peng MS(2)(4).\\n\\nAuthor information:\\n(1)State Key Laboratory for Conservation and Utilization of Bio-resources in \\nYunnan, Yunnan University, Kunming, China.\\n(2)State Key Laboratory of Genetic Resources and Evolution &amp; Yunnan Laboratory \\nof Molecular Biology of Domestic Animals, Kunming Institute of Zoology, Chinese \\nAcademy of Sciences, Kunming, China.\\n(3)School of Software, Yunnan University, Kunming, China.\\n(4)University of Chinese Academy of Sciences, Beijing, China.\\n(5)Department of Anesthesiology, First Affiliated Hospital of Kunming Medical \\nUniversity, Kunming, China.\\n(#)Contributed equally\\n\\nLong-read sequencing technology is a powerful approach with application in \\nvarious genetic and genomic researches. Herein, we developed the pipeline for \\nlong amplicon high-fidelity (HiFi) sequencing and then applied it for sequencing \\nmitochondrial DNA (mtDNA) genomes from pools of 79 Tibetan Mastiffs. We \\namplified the mtDNA genome with long-range PCR using two pairs of primers. Two \\nrounds of circular consensus sequencing (CCS) were conducted and their accuracy \\nwas evaluated. The results indicate that the second round of CCS can improve the \\naccuracy of HiFi reads. In addition, the analysis of 79 high-quality mtDNA \\ngenomes shows the Tibetan Mastiffs from outside of the Tibetan Plateau \\nexperienced hybridization with other dogs. The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.\\n\\nThis article is protected by copyright. All rights reserved.\\n\\nDOI: 10.1111/1755-0998.13765\\nPMID: 36756726\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36756173</td>\n",
       "      <td>2022-10-23</td>\n",
       "      <td>10.1016/j.csbj.2022.10.016</td>\n",
       "      <td>https://github.com/lichen-lab/MTAE</td>\n",
       "      <td>https://github.com/lichen-lab/MTAE</td>\n",
       "      <td>Computational and structural biotechnology journal</td>\n",
       "      <td>Multi-task deep autoencoder to predict Alzheimer's disease progression using temporal DNA methylation data in peripheral blood.</td>\n",
       "      <td>\\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).</td>\n",
       "      <td>Comput Struct Biotechnol J. 2022 Oct 23;20:5761-5774. doi: \\n10.1016/j.csbj.2022.10.016. eCollection 2022.\\n\\nMulti-task deep autoencoder to predict Alzheimer's disease progression using \\ntemporal DNA methylation data in peripheral blood.\\n\\nChen L(1), Saykin AJ(2), Yao B(3), Zhao F(1); Alzheimerâ€™s Disease Neuroimaging \\nInitiative (ADNI).\\n\\nAuthor information:\\n(1)Department of Biostatistics, University of Florida, Gainesville, FL 32603, \\nUnited States.\\n(2)Department of Radiology and Imaging Sciences, Indiana University School of \\nMedicine, Indianapolis, IN 46202, United States.\\n(3)Department of Human Genetics, Emory University, Atlanta, GA 30322, United \\nStates.\\n\\nTraditional approaches for diagnosing Alzheimer's disease (AD) such as brain \\nimaging and cerebrospinal fluid are invasive and expensive. It is desirable to \\ndevelop a useful diagnostic tool by exploiting biomarkers obtained from \\nperipheral tissues due to their noninvasive and easily accessible \\ncharacteristics. However, the capacity of using DNA methylation data in \\nperipheral blood for predicting AD progression is rarely known. It is also \\nchallenging to develop an efficient prediction model considering the complex and \\nhigh-dimensional DNA methylation data in a longitudinal study. Here, we develop \\ntwo multi-task deep autoencoders, which are based on the convolutional \\nautoencoder and long short-term memory autoencoder to learn the compressed \\nfeature representation by jointly minimizing the reconstruction error and \\nmaximizing the prediction accuracy. By benchmarking on longitudinal DNA \\nmethylation data collected from the peripheral blood in Alzheimer's Disease \\nNeuroimaging Initiative, we demonstrate that the proposed multi-task deep \\nautoencoders outperform state-of-the-art machine learning approaches for both \\npredicting AD progression and reconstructing the temporal DNA methylation \\nprofiles. In addition, the proposed multi-task deep autoencoders can predict AD \\nprogression accurately using only the historical DNA methylation data and the \\nperformance is further improved by including all temporal DNA methylation data. \\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).\\n\\nDOI: 10.1016/j.csbj.2022.10.016\\nPMCID: PMC9619306\\nPMID: 36756173\\n\\nConflict of interest statement: The authors declare that they have no known \\ncompeting financial interests or personal relationships that could have appeared \\nto influence the work reported in this paper.\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36711471</td>\n",
       "      <td>2023-01-20</td>\n",
       "      <td>10.1101/2023.01.17.524477</td>\n",
       "      <td>https://github.com/jianlin-cheng/TransFun</td>\n",
       "      <td>https://github.com/jianlin-cheng/TransFun</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Combining protein sequences and structures with transformers and equivariant graph neural networks to predict protein function.</td>\n",
       "      <td>AVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.</td>\n",
       "      <td>bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 10.1101/2023.01.17.524477. \\nPreprint.\\n\\nCombining protein sequences and structures with transformers and equivariant \\ngraph neural networks to predict protein function.\\n\\nBoadu F, Cao H, Cheng J.\\n\\nMOTIVATION: Millions of protein sequences have been generated by numerous genome \\nand transcriptome sequencing projects. However, experimentally determining the \\nfunction of the proteins is still a time consuming, low-throughput, and \\nexpensive process, leading to a large protein sequence-function gap. Therefore, \\nit is important to develop computational methods to accurately predict protein \\nfunction to fill the gap. Even though many methods have been developed to use \\nprotein sequences as input to predict function, much fewer methods leverage \\nprotein structures in protein function prediction because there was lack of \\naccurate protein structures for most proteins until recently.\\nRESULTS: We developed TransFun - a method using a transformer-based protein \\nlanguage model and 3D-equivariant graph neural networks to distill information \\nfrom both protein sequences and structures to predict protein function. It \\nextracts feature embeddings from protein sequences using a pre-trained protein \\nlanguage model (ESM) via transfer learning and combines them with 3D structures \\nof proteins predicted by AlphaFold2 through equivariant graph neural networks. \\nBenchmarked on the CAFA3 test dataset and a new test dataset, TransFun \\noutperforms several state-of-the-art methods, indicating the language model and \\n3D-equivariant graph neural networks are effective methods to leverage protein \\nsequences and structures to improve protein function prediction. Combining \\nTransFun predictions and sequence similarity-based predictions can further \\nincrease prediction accuracy.\\nAVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.\\n\\nDOI: 10.1101/2023.01.17.524477\\nPMCID: PMC9882282\\nPMID: 36711471\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36710930</td>\n",
       "      <td>2023-01-13</td>\n",
       "      <td>10.3389/fnmol.2022.1037565</td>\n",
       "      <td>https://github.com/linxi159/CDSlncR</td>\n",
       "      <td>https://github.com/linxi159/CDSlncR</td>\n",
       "      <td>Frontiers in molecular neuroscience</td>\n",
       "      <td>Inferring cell developmental stage-specific lncRNA regulation in the developing human neocortex with CDSlncR.</td>\n",
       "      <td>CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.</td>\n",
       "      <td>Front Mol Neurosci. 2023 Jan 13;15:1037565. doi: 10.3389/fnmol.2022.1037565. \\neCollection 2022.\\n\\nInferring cell developmental stage-specific lncRNA regulation in the developing \\nhuman neocortex with CDSlncR.\\n\\nHuang M(1)(2), Ma J(1)(3), Zhang J(3).\\n\\nAuthor information:\\n(1)Department of Automation, Xiamen University, Xiamen, China.\\n(2)Department of Computer Science, University of Tsukuba, Tsukuba, Japan.\\n(3)School of Engineering, Dali University, Dali, China.\\n\\nNoncoding RNAs (ncRNAs) occupy ~98% of the transcriptome in human, and are \\nusually not translated into proteins. Among ncRNAs, long non-coding RNAs \\n(lncRNAs, &gt;200 nucleotides) are important regulators to modulate gene \\nexpression, and are involved in many biological processes (e.g., cell \\ndevelopment). To study lncRNA regulation, many computational approaches or tools \\nhave been proposed by using bulk transcriptomics data. Nevertheless, previous \\nbulk data-driven methods are mostly limited to explore the lncRNA regulation \\nregarding all of cells, instead of the lncRNA regulation specific to cell \\ndevelopmental stages. Fortunately, recent advance in single-cell sequencing data \\nhas provided a way to investigate cell developmental stage-specific lncRNA \\nregulation. In this work, we present a novel computational method, CDSlncR (Cell \\nDevelopmental Stage-specific lncRNA regulation), which combines putative \\nlncRNA-target binding information with single-cell transcriptomics data to infer \\ncell developmental stage-specific lncRNA regulation. For each cell developmental \\nstage, CDSlncR constructs a cell developmental stage-specific lncRNA regulatory \\nnetwork in the cell developmental stage. To illustrate the effectiveness of \\nCDSlncR, we apply CDSlncR into single-cell transcriptomics data of the \\ndeveloping human neocortex for exploring lncRNA regulation across different \\nhuman neocortex developmental stages. Network analysis shows that the lncRNA \\nregulation is unique in each developmental stage of human neocortex. As a case \\nstudy, we also perform particular analysis on the cell developmental \\nstage-specific lncRNA regulation related to 18 known lncRNA biomarkers in autism \\nspectrum disorder. Finally, the comparison result indicates that CDSlncR is an \\neffective method for predicting cell developmental stage-specific lncRNA \\ntargets. CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.\\n\\nDOI: 10.3389/fnmol.2022.1037565\\nPMCID: PMC9880432\\nPMID: 36710930\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36710872</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>10.7717/peerj.14706</td>\n",
       "      <td>https://github.com/ningzilan/SSWD</td>\n",
       "      <td>https://github.com/ningzilan/SSWD</td>\n",
       "      <td>PeerJ</td>\n",
       "      <td>A clustering method for small scRNA-seq data based on subspace and weighted distance.</td>\n",
       "      <td>SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.</td>\n",
       "      <td>PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/peerj.14706. eCollection 2023.\\n\\nA clustering method for small scRNA-seq data based on subspace and weighted \\ndistance.\\n\\nNing Z(1)(2), Dai Z(1), Zhang H(2), Chen Y(1), Yuan Z(1).\\n\\nAuthor information:\\n(1)Hunan Engineering &amp; Technology Research Centre for Agricultural Big Data \\nAnalysis &amp; Decision-Making, Hunan Agricultural University, Changsha, Hunan, \\nChina.\\n(2)Hunan Agricultural University, College of Information and Intelligence, \\nChangsha, Hunan, China.\\n\\nBACKGROUND: Identifying the cell types using unsupervised methods is essential \\nfor scRNA-seq research. However, conventional similarity measures introduce \\nchallenges to single-cell data clustering because of the high dimensional, high \\nnoise, and high dropout.\\nMETHODS: We proposed a clustering method for small ScRNA-seq data based on \\nSubspace and Weighted Distance (SSWD), which follows the assumption that the \\nsets of gene subspace composed of similar density-distributing genes can better \\ndistinguish cell groups. To accurately capture the intrinsic relationship among \\ncells or genes, a new distance metric that combines Euclidean and Pearson \\ndistance through a weighting strategy was proposed. The relative \\nCalinski-Harabasz (CH) index was used to estimate the cluster numbers instead of \\nthe CH index because it is comparable across degrees of freedom.\\nRESULTS: We compared SSWD with seven prevailing methods on eight publicly \\nscRNA-seq datasets. The experimental results show that the SSWD has better \\nclustering accuracy and the partitioning ability of cell groups. SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.\\n\\nDOI: 10.7717/peerj.14706\\nPMCID: PMC9879162\\nPMID: 36710872 [Indexed for MEDLINE]\\n\\nConflict of interest statement: The authors declare there are no competing \\ninterests.\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36709790</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>10.1016/j.ymeth.2023.01.006</td>\n",
       "      <td>https://github.com/zhuyuan-cug/VGAE-LCME</td>\n",
       "      <td>https://github.com/zhuyuan-cug/VGAE-LCME</td>\n",
       "      <td>Methods (San Diego, Calif.)</td>\n",
       "      <td>Predicting latent lncRNA and cancer metastatic event associations via variational graph auto-encoder.</td>\n",
       "      <td>The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc.</td>\n",
       "      <td>Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.ymeth.2023.01.006. Online ahead of \\nprint.\\n\\nPredicting latent lncRNA and cancer metastatic event associations via \\nvariational graph auto-encoder.\\n\\nZhu Y(1), Zhang F(2), Zhang S(3), Yi M(4).\\n\\nAuthor information:\\n(1)School of Automation, China University of Geosciences, 388 Lumo Road, \\nHongshan District, 430074, Wuhan, Hubei, China; Hubei Key Laboratory of Advanced \\nControl and Intelligent Automation for Complex Systems, 388 Lumo Road, Hongshan \\nDistrict, 430074, Wuhan, Hubei, China; Engineering Research Center of \\nIntelligent Technology for Geo-Exploration, 388 Lumo Road, Hongshan District, \\n430074, Wuhan, Hubei, China.\\n(2)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China.\\n(3)College of Life Science and Health, Wuhan University of Science and \\nTechnology, 974 Heping Avenue, Qingshan District, 430081, Wuhan, Hubei, China. \\nElectronic address: shihuazhang@wust.edu.cn.\\n(4)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China. Electronic address: \\nmingyi@cug.edu.cn.\\n\\nLong non-coding RNA (lncRNA) are shown to be closely associated with cancer \\nmetastatic events (CME, e.g., cancer cell invasion, intravasation, \\nextravasation, proliferation) that collaboratively accelerate malignant cancer \\nspread and cause high mortality rate in patients. Clinical trials may accurately \\nuncover the relationships between lncRNAs and CMEs; however, it is \\ntime-consuming and expensive. With the accumulation of data, there is an urgent \\nneed to find efficient ways to identify these relationships. Herein, a graph \\nembedding representation-based predictor (VGEA-LCME) for exploring latent \\nlncRNA-CME associations is introduced. In VGEA-LCME, a heterogeneous combined \\nnetwork is constructed by integrating similarity and linkage matrix that can \\nmaintain internal and external characteristics of networks, and a variational \\ngraph auto-encoder serves as a feature generator to represent arbitrary lncRNA \\nand CME pair. The final robustness predicted result is obtained by ensemble \\nclassifier strategy via cross-validation. Experimental comparisons and \\nliterature verification show better remarkable performance of VGEA-LCME, \\nalthough the similarities between CMEs are challenging to calculate. In \\naddition, VGEA-LCME can further identify organ-specific CMEs. To the best of our \\nknowledge, this is the first computational attempt to discover the potential \\nrelationships between lncRNAs and CMEs. It may provide support and new insight \\nfor guiding experimental research of metastatic cancers. The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc. All rights reserved.\\n\\nDOI: 10.1016/j.ymeth.2023.01.006\\nPMID: 36709790\\n\\nConflict of interest statement: Declaration of Competing Interest The authors \\ndeclare that there are no conflict of interests, we do not have any possible \\nconflicts of interest.\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36709557</td>\n",
       "      <td>2023-01-15</td>\n",
       "      <td>10.1016/j.cmpb.2023.107355</td>\n",
       "      <td>https://github.com/cistib/vessel-aneurysm-segmentation</td>\n",
       "      <td>https://github.com/cistib/vessel-aneurysm-segmentation</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>High-throughput 3DRA segmentation of brain vasculature and aneurysms using deep learning.</td>\n",
       "      <td>Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023.</td>\n",
       "      <td>Comput Methods Programs Biomed. 2023 Jan 15;230:107355. doi: \\n10.1016/j.cmpb.2023.107355. Online ahead of print.\\n\\nHigh-throughput 3DRA segmentation of brain vasculature and aneurysms using deep \\nlearning.\\n\\nLin F(1), Xia Y(2), Song S(1), Ravikumar N(1), Frangi AF(3).\\n\\nAuthor information:\\n(1)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK.\\n(2)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK. Electronic address: \\ny.xia@leeds.ac.uk.\\n(3)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK; Leeds Institute for \\nCardiovascular and Metabolic Medicine (LICAMM), School of Medicine, University \\nof Leeds, Leeds LS2 9JT, UK; Medical Imaging Research Center (MIRC), \\nCardiovascular Science and Electronic Engineering Departments, KU Leuven, \\nLeuven, Belgium; Alan Turing Institute, London, UK.\\n\\nBACKGROUND AND OBJECTIVES: Automatic segmentation of the cerebral vasculature \\nand aneurysms facilitates incidental detection of aneurysms. The assessment of \\naneurysm rupture risk assists with pre-operative treatment planning and enables \\nin-silico investigation of cerebral hemodynamics within and in the vicinity of \\naneurysms. However, ensuring precise and robust segmentation of cerebral vessels \\nand aneurysms in neuroimaging modalities such as three-dimensional rotational \\nangiography (3DRA) is challenging. The vasculature constitutes a small \\nproportion of the image volume, resulting in a large class imbalance (relative \\nto surrounding brain tissue). Additionally, aneurysms and vessels have similar \\nimage/appearance characteristics, making it challenging to distinguish the \\naneurysm sac from the vessel lumen.\\nMETHODS: We propose a novel multi-class convolutional neural network to tackle \\nthese challenges and facilitate the automatic segmentation of cerebral vessels \\nand aneurysms in 3DRA images. The proposed model is trained and evaluated on an \\ninternal multi-center dataset and an external publicly available challenge \\ndataset.\\nRESULTS: On the internal clinical dataset, our method consistently outperformed \\nseveral state-of-the-art approaches for vessel and aneurysm segmentation, \\nachieving an average Dice score of 0.81 (0.15 higher than nnUNet) and an average \\nsurface-to-surface error of 0.20Â mm (less than the in-plane resolution \\n(0.35Â mm/pixel)) for aneurysm segmentation; and an average Dice score of 0.91 \\nand average surface-to-surface error of 0.25Â mm for vessel segmentation. In 223 \\ncases of a clinical dataset, our method accurately segmented 190 aneurysm cases.\\nCONCLUSIONS: The proposed approach can help address class imbalance problems and \\ninter-class interference problems in multi-class segmentation. Besides, this \\nmethod performs consistently on clinical datasets from four different sources \\nand the generated results are qualified for hemodynamic simulation. Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023. Published by Elsevier B.V.\\n\\nDOI: 10.1016/j.cmpb.2023.107355\\nPMID: 36709557\\n\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                          DOI  \\\n",
       "0   36760999  2023-01-24   10.3389/fgene.2023.1082032   \n",
       "1   36759692  2023-02-09   10.1038/s41598-023-29320-6   \n",
       "2   36759336  2023-02-09          10.1093/bib/bbad044   \n",
       "3   36756726  2023-02-09      10.1111/1755-0998.13765   \n",
       "4   36756173  2022-10-23   10.1016/j.csbj.2022.10.016   \n",
       "..       ...         ...                          ...   \n",
       "95  36711471  2023-01-20    10.1101/2023.01.17.524477   \n",
       "96  36710930  2023-01-13   10.3389/fnmol.2022.1037565   \n",
       "97  36710872  2023-01-23          10.7717/peerj.14706   \n",
       "98  36709790  2023-01-26  10.1016/j.ymeth.2023.01.006   \n",
       "99  36709557  2023-01-15   10.1016/j.cmpb.2023.107355   \n",
       "\n",
       "                                           GitHub_link_raw  \\\n",
       "0                       https://github.com/free1234hm/CLAM   \n",
       "1                     https://github.com/Socrates023/DCPHA   \n",
       "2            https://github.com/victorykobets/HiConfidence   \n",
       "3                     https://github.com/Caizf-script/HQGR   \n",
       "4                       https://github.com/lichen-lab/MTAE   \n",
       "..                                                     ...   \n",
       "95               https://github.com/jianlin-cheng/TransFun   \n",
       "96                     https://github.com/linxi159/CDSlncR   \n",
       "97                       https://github.com/ningzilan/SSWD   \n",
       "98                https://github.com/zhuyuan-cug/VGAE-LCME   \n",
       "99  https://github.com/cistib/vessel-aneurysm-segmentation   \n",
       "\n",
       "                                         GitHub_link_clean  \\\n",
       "0                       https://github.com/free1234hm/CLAM   \n",
       "1                     https://github.com/Socrates023/DCPHA   \n",
       "2            https://github.com/victorykobets/HiConfidence   \n",
       "3                     https://github.com/Caizf-script/HQGR   \n",
       "4                       https://github.com/lichen-lab/MTAE   \n",
       "..                                                     ...   \n",
       "95               https://github.com/jianlin-cheng/TransFun   \n",
       "96                     https://github.com/linxi159/CDSlncR   \n",
       "97                       https://github.com/ningzilan/SSWD   \n",
       "98                https://github.com/zhuyuan-cug/VGAE-LCME   \n",
       "99  https://github.com/cistib/vessel-aneurysm-segmentation   \n",
       "\n",
       "                                               Journal  \\\n",
       "0                                Frontiers in genetics   \n",
       "1                                   Scientific reports   \n",
       "2                          Briefings in bioinformatics   \n",
       "3                          Molecular ecology resources   \n",
       "4   Computational and structural biotechnology journal   \n",
       "..                                                 ...   \n",
       "95           bioRxiv : the preprint server for biology   \n",
       "96                 Frontiers in molecular neuroscience   \n",
       "97                                               PeerJ   \n",
       "98                         Methods (San Diego, Calif.)   \n",
       "99        Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                                                                                              Title  \\\n",
       "0                       Identification of functional gene modules by integrating multi-omics data and known molecular interactions.   \n",
       "1                                              Deep consistency-preserving hash auto-encoders for neuroimage cross-modal retrieval.   \n",
       "2                        HiConfidence: a novel approach uncovering the biological signal in Hi-C data affected by technical biases.   \n",
       "3                                                                      Long amplicon HiFi sequencing for mitochondrial DNA genomes.   \n",
       "4   Multi-task deep autoencoder to predict Alzheimer's disease progression using temporal DNA methylation data in peripheral blood.   \n",
       "..                                                                                                                              ...   \n",
       "95  Combining protein sequences and structures with transformers and equivariant graph neural networks to predict protein function.   \n",
       "96                    Inferring cell developmental stage-specific lncRNA regulation in the developing human neocortex with CDSlncR.   \n",
       "97                                            A clustering method for small scRNA-seq data based on subspace and weighted distance.   \n",
       "98                            Predicting latent lncRNA and cancer metastatic event associations via variational graph auto-encoder.   \n",
       "99                                        High-throughput 3DRA segmentation of brain vasculature and aneurysms using deep learning.   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                            Phrase  \\\n",
       "0                                                                                                                       We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.   \n",
       "1                                                                                                                                                                                                                                             We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .   \n",
       "2                                                                                                                                                                                                              The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023.   \n",
       "3   The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.   \n",
       "4                                                                                                                                                                                                                                                    \\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).   \n",
       "..                                                                                                                                                                                                                                                                                                                             ...   \n",
       "95                                                                                                                                                                                          AVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.   \n",
       "96                                                                                                                                                                                                                           CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.   \n",
       "97                                                                                                                                                                                                                                             SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.   \n",
       "98                                                                                                                                                                                                         The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc.   \n",
       "99                                                                                                                                                                                                                                Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023.   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Abstract  \\\n",
       "0                                                                                                                                                                                                             Front Genet. 2023 Jan 24;14:1082032. doi: 10.3389/fgene.2023.1082032. \\neCollection 2023.\\n\\nIdentification of functional gene modules by integrating multi-omics data and \\nknown molecular interactions.\\n\\nChen X(1)(2), Han M(2), Li Y(3), Li X(2), Zhang J(2), Zhu Y(1)(2).\\n\\nAuthor information:\\n(1)Basic Medical School, Anhui Medical University, Hefei, China.\\n(2)National Center for Protein Sciences (Beijing), Beijing Proteome Research \\nCenter, Beijing Institute of Lifeomics, Beijing, China.\\n(3)Central Research Laboratory, Peking Union Medical College Hospital, Chinese \\nAcademy of Medical Sciences and Peking Union Medical College, Beijing, China.\\n\\nMulti-omics data integration has emerged as a promising approach to identify \\npatient subgroups. However, in terms of grouping genes (or gene products) into \\nco-expression modules, data integration methods suffer from two main drawbacks. \\nFirst, most existing methods only consider genes or samples measured in all \\ndifferent datasets. Second, known molecular interactions (e.g., transcriptional \\nregulatory interactions, protein-protein interactions and biological pathways) \\ncannot be utilized to assist in module detection. Herein, we present a novel \\ndata integration framework, Correlation-based Local Approximation of Membership \\n(CLAM), which provides two methodological innovations to address these \\nlimitations: 1) constructing a trans-omics neighborhood matrix by integrating \\nmulti-omics datasets and known molecular interactions, and 2) using a local \\napproximation procedure to define gene modules from the matrix. Applying \\nCorrelation-based Local Approximation of Membership to human colorectal cancer \\n(CRC) and mouse B-cell differentiation multi-omics data obtained from The Cancer \\nGenome Atlas (TCGA), Clinical Proteomics Tumor Analysis Consortium (CPTAC), Gene \\nExpression Omnibus (GEO) and ProteomeXchange database, we demonstrated its \\nsuperior ability to recover biologically relevant modules and gene ontology (GO) \\nterms. Further investigation of the colorectal cancer modules revealed numerous \\ntranscription factors and KEGG pathways that played crucial roles in colorectal \\ncancer progression. Module-based survival analysis constructed four \\nsurvival-related networks in which pairwise gene correlations were significantly \\ncorrelated with colorectal cancer patient survival. Overall, the series of \\nevaluations demonstrated the great potential of Correlation-based Local \\nApproximation of Membership for identifying modular biomarkers for complex \\ndiseases. We implemented Correlation-based Local Approximation of Membership as \\na user-friendly application available at https://github.com/free1234hm/CLAM.\\n\\nCopyright Â© 2023 Chen, Han, Li, Li, Zhang and Zhu.\\n\\nDOI: 10.3389/fgene.2023.1082032\\nPMCID: PMC9902936\\nPMID: 36760999\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Sci Rep. 2023 Feb 9;13(1):2316. doi: 10.1038/s41598-023-29320-6.\\n\\nDeep consistency-preserving hash auto-encoders for neuroimage cross-modal \\nretrieval.\\n\\nWang X(1), Zeng X(2).\\n\\nAuthor information:\\n(1)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China.\\n(2)College of Computer Science and Technology, Chongqing University of Posts and \\nTelecommunications, Chongqing, 400065, China. zengxh@cqupt.edu.cn.\\n\\nCross-modal hashing is an efficient method to embed high-dimensional \\nheterogeneous modal feature descriptors into a consistency-preserving Hamming \\nspace with low-dimensional. Most existing cross-modal hashing methods have been \\nable to bridge the heterogeneous modality gap, but there are still two \\nchallenges resulting in limited retrieval accuracy: (1) ignoring the continuous \\nsimilarity of samples on manifold; (2) lack of discriminability of hash codes \\nwith the same semantics. To cope with these problems, we propose a Deep \\nConsistency-Preserving Hash Auto-encoders model, called DCPHA, based on the \\nmulti-manifold property of the feature distribution. Specifically, DCPHA \\nconsists of a pair of asymmetric auto-encoders and two semantics-preserving \\nattention branches working in the encoding and decoding stages, respectively. \\nWhen the number of input medical image modalities is greater than 2, the encoder \\nis a multiple pseudo-Siamese network designed to extract specific modality \\nfeatures of different medical image modalities. In addition, we define the \\ncontinuous similarity of heterogeneous and homogeneous samples on Riemann \\nmanifold from the perspective of multiple sub-manifolds, respectively, and the \\ntwo constraints, i.e., multi-semantic consistency and multi-manifold \\nsimilarity-preserving, are embedded in the learning of hash codes to obtain \\nhigh-quality hash codes with consistency-preserving. The extensive experiments \\nshow that the proposed DCPHA has the most stable and state-of-the-art \\nperformance. We make code and models publicly available: \\nhttps://github.com/Socrates023/DCPHA .\\n\\nÂ© 2023. The Author(s).\\n\\nDOI: 10.1038/s41598-023-29320-6\\nPMID: 36759692\\n\\n   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Brief Bioinform. 2023 Feb 9:bbad044. doi: 10.1093/bib/bbad044. Online ahead of \\nprint.\\n\\nHiConfidence: a novel approach uncovering the biological signal in Hi-C data \\naffected by technical biases.\\n\\nKobets VA(1), Ulianov SV(2)(3), Galitsyna AA(1)(2)(4), Doronin SA(5), Mikhaleva \\nEA(5), Gelfand MS(1)(4), Shevelyov YY(5), Razin SV(2)(3), Khrameeva EE(1).\\n\\nAuthor information:\\n(1)Skolkovo Institute of Science and Technology, Moscow, 121205, Russia.\\n(2)Institute of Gene Biology, Russian Academy of Sciences, Moscow, 119334, \\nRussia.\\n(3)Faculty of Biology, M.V. Lomonosov Moscow State University, Moscow, 119992, \\nRussia.\\n(4)A.A. Kharkevich Institute for Information Transmission Problems, Russian \\nAcademy of Sciences, Moscow, 127051, Russia.\\n(5)Institute of Molecular Genetics of National Research Centre \"Kurchatov \\nInstitute\", Moscow, 123182, Russia.\\n\\nThe chromatin interaction assays, particularly Hi-C, enable detailed studies of \\ngenome architecture in multiple organisms and model systems, resulting in a \\ndeeper understanding of gene expression regulation mechanisms mediated by \\nepigenetics. However, the analysis and interpretation of Hi-C data remain \\nchallenging due to technical biases, limiting direct comparisons of datasets \\nobtained in different experiments and laboratories. As a result, removing biases \\nfrom Hi-C-generated chromatin contact matrices is a critical data analysis step. \\nOur novel approach, HiConfidence, eliminates biases from the Hi-C data by \\nweighing chromatin contacts according to their consistency between replicates so \\nthat low-quality replicates do not substantially influence the result. The \\nalgorithm is effective for the analysis of global changes in chromatin \\nstructures such as compartments and topologically associating domains. We apply \\nthe HiConfidence approach to several Hi-C datasets with significant technical \\nbiases, that could not be analyzed effectively using existing methods, and \\nobtain meaningful biological conclusions. In particular, HiConfidence aids in \\nthe study of how changes in histone acetylation pattern affect chromatin \\norganization in Drosophila melanogaster S2 cells. The method is freely available \\nat GitHub: https://github.com/victorykobets/HiConfidence.\\n\\nÂ© The Author(s) 2023. Published by Oxford University Press.\\n\\nDOI: 10.1093/bib/bbad044\\nPMID: 36759336\\n\\n   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Mol Ecol Resour. 2023 Feb 9. doi: 10.1111/1755-0998.13765. Online ahead of \\nprint.\\n\\nLong amplicon HiFi sequencing for mitochondrial DNA genomes.\\n\\nCai ZF(#)(1)(2), Hu JY(#)(3), Yin TT(#)(2), Wang D(3), Shen QK(2), Ma C(2)(4), \\nOu DQ(5), Xu MM(2)(4), Shi X(2)(4), Li QL(1)(2), Wu RN(2), Ajuma L(2)(4), Adeola \\nAC(2), Zhang YP(1)(2)(4), Peng MS(2)(4).\\n\\nAuthor information:\\n(1)State Key Laboratory for Conservation and Utilization of Bio-resources in \\nYunnan, Yunnan University, Kunming, China.\\n(2)State Key Laboratory of Genetic Resources and Evolution & Yunnan Laboratory \\nof Molecular Biology of Domestic Animals, Kunming Institute of Zoology, Chinese \\nAcademy of Sciences, Kunming, China.\\n(3)School of Software, Yunnan University, Kunming, China.\\n(4)University of Chinese Academy of Sciences, Beijing, China.\\n(5)Department of Anesthesiology, First Affiliated Hospital of Kunming Medical \\nUniversity, Kunming, China.\\n(#)Contributed equally\\n\\nLong-read sequencing technology is a powerful approach with application in \\nvarious genetic and genomic researches. Herein, we developed the pipeline for \\nlong amplicon high-fidelity (HiFi) sequencing and then applied it for sequencing \\nmitochondrial DNA (mtDNA) genomes from pools of 79 Tibetan Mastiffs. We \\namplified the mtDNA genome with long-range PCR using two pairs of primers. Two \\nrounds of circular consensus sequencing (CCS) were conducted and their accuracy \\nwas evaluated. The results indicate that the second round of CCS can improve the \\naccuracy of HiFi reads. In addition, the analysis of 79 high-quality mtDNA \\ngenomes shows the Tibetan Mastiffs from outside of the Tibetan Plateau \\nexperienced hybridization with other dogs. The High Quality Reads Generator \\n(HQGR) software is provided to facilitate data analyses, which is publicly \\naccessible on GitHub (https://github.com/Caizf-script/HQGR). Our long amplicon \\nHiFi sequencing pipeline can also be applied in various target enrichment \\nstrategies for small genomes and candidate genes.\\n\\nThis article is protected by copyright. All rights reserved.\\n\\nDOI: 10.1111/1755-0998.13765\\nPMID: 36756726\\n\\n   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Comput Struct Biotechnol J. 2022 Oct 23;20:5761-5774. doi: \\n10.1016/j.csbj.2022.10.016. eCollection 2022.\\n\\nMulti-task deep autoencoder to predict Alzheimer's disease progression using \\ntemporal DNA methylation data in peripheral blood.\\n\\nChen L(1), Saykin AJ(2), Yao B(3), Zhao F(1); Alzheimerâ€™s Disease Neuroimaging \\nInitiative (ADNI).\\n\\nAuthor information:\\n(1)Department of Biostatistics, University of Florida, Gainesville, FL 32603, \\nUnited States.\\n(2)Department of Radiology and Imaging Sciences, Indiana University School of \\nMedicine, Indianapolis, IN 46202, United States.\\n(3)Department of Human Genetics, Emory University, Atlanta, GA 30322, United \\nStates.\\n\\nTraditional approaches for diagnosing Alzheimer's disease (AD) such as brain \\nimaging and cerebrospinal fluid are invasive and expensive. It is desirable to \\ndevelop a useful diagnostic tool by exploiting biomarkers obtained from \\nperipheral tissues due to their noninvasive and easily accessible \\ncharacteristics. However, the capacity of using DNA methylation data in \\nperipheral blood for predicting AD progression is rarely known. It is also \\nchallenging to develop an efficient prediction model considering the complex and \\nhigh-dimensional DNA methylation data in a longitudinal study. Here, we develop \\ntwo multi-task deep autoencoders, which are based on the convolutional \\nautoencoder and long short-term memory autoencoder to learn the compressed \\nfeature representation by jointly minimizing the reconstruction error and \\nmaximizing the prediction accuracy. By benchmarking on longitudinal DNA \\nmethylation data collected from the peripheral blood in Alzheimer's Disease \\nNeuroimaging Initiative, we demonstrate that the proposed multi-task deep \\nautoencoders outperform state-of-the-art machine learning approaches for both \\npredicting AD progression and reconstructing the temporal DNA methylation \\nprofiles. In addition, the proposed multi-task deep autoencoders can predict AD \\nprogression accurately using only the historical DNA methylation data and the \\nperformance is further improved by including all temporal DNA methylation data. \\nAvailability::Â https://github.com/lichen-lab/MTAE.\\n\\nÂ© 2022 The Author(s).\\n\\nDOI: 10.1016/j.csbj.2022.10.016\\nPMCID: PMC9619306\\nPMID: 36756173\\n\\nConflict of interest statement: The authors declare that they have no known \\ncompeting financial interests or personal relationships that could have appeared \\nto influence the work reported in this paper.\\n\\n   \n",
       "..                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              ...   \n",
       "95                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         bioRxiv. 2023 Jan 20:2023.01.17.524477. doi: 10.1101/2023.01.17.524477. \\nPreprint.\\n\\nCombining protein sequences and structures with transformers and equivariant \\ngraph neural networks to predict protein function.\\n\\nBoadu F, Cao H, Cheng J.\\n\\nMOTIVATION: Millions of protein sequences have been generated by numerous genome \\nand transcriptome sequencing projects. However, experimentally determining the \\nfunction of the proteins is still a time consuming, low-throughput, and \\nexpensive process, leading to a large protein sequence-function gap. Therefore, \\nit is important to develop computational methods to accurately predict protein \\nfunction to fill the gap. Even though many methods have been developed to use \\nprotein sequences as input to predict function, much fewer methods leverage \\nprotein structures in protein function prediction because there was lack of \\naccurate protein structures for most proteins until recently.\\nRESULTS: We developed TransFun - a method using a transformer-based protein \\nlanguage model and 3D-equivariant graph neural networks to distill information \\nfrom both protein sequences and structures to predict protein function. It \\nextracts feature embeddings from protein sequences using a pre-trained protein \\nlanguage model (ESM) via transfer learning and combines them with 3D structures \\nof proteins predicted by AlphaFold2 through equivariant graph neural networks. \\nBenchmarked on the CAFA3 test dataset and a new test dataset, TransFun \\noutperforms several state-of-the-art methods, indicating the language model and \\n3D-equivariant graph neural networks are effective methods to leverage protein \\nsequences and structures to improve protein function prediction. Combining \\nTransFun predictions and sequence similarity-based predictions can further \\nincrease prediction accuracy.\\nAVAILABILITY: The source code of TransFun is available at \\nhttps://github.com/jianlin-cheng/TransFun.\\nCONTACT: chengji@missouri.edu.\\n\\nDOI: 10.1101/2023.01.17.524477\\nPMCID: PMC9882282\\nPMID: 36711471\\n\\n   \n",
       "96                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Front Mol Neurosci. 2023 Jan 13;15:1037565. doi: 10.3389/fnmol.2022.1037565. \\neCollection 2022.\\n\\nInferring cell developmental stage-specific lncRNA regulation in the developing \\nhuman neocortex with CDSlncR.\\n\\nHuang M(1)(2), Ma J(1)(3), Zhang J(3).\\n\\nAuthor information:\\n(1)Department of Automation, Xiamen University, Xiamen, China.\\n(2)Department of Computer Science, University of Tsukuba, Tsukuba, Japan.\\n(3)School of Engineering, Dali University, Dali, China.\\n\\nNoncoding RNAs (ncRNAs) occupy ~98% of the transcriptome in human, and are \\nusually not translated into proteins. Among ncRNAs, long non-coding RNAs \\n(lncRNAs, >200 nucleotides) are important regulators to modulate gene \\nexpression, and are involved in many biological processes (e.g., cell \\ndevelopment). To study lncRNA regulation, many computational approaches or tools \\nhave been proposed by using bulk transcriptomics data. Nevertheless, previous \\nbulk data-driven methods are mostly limited to explore the lncRNA regulation \\nregarding all of cells, instead of the lncRNA regulation specific to cell \\ndevelopmental stages. Fortunately, recent advance in single-cell sequencing data \\nhas provided a way to investigate cell developmental stage-specific lncRNA \\nregulation. In this work, we present a novel computational method, CDSlncR (Cell \\nDevelopmental Stage-specific lncRNA regulation), which combines putative \\nlncRNA-target binding information with single-cell transcriptomics data to infer \\ncell developmental stage-specific lncRNA regulation. For each cell developmental \\nstage, CDSlncR constructs a cell developmental stage-specific lncRNA regulatory \\nnetwork in the cell developmental stage. To illustrate the effectiveness of \\nCDSlncR, we apply CDSlncR into single-cell transcriptomics data of the \\ndeveloping human neocortex for exploring lncRNA regulation across different \\nhuman neocortex developmental stages. Network analysis shows that the lncRNA \\nregulation is unique in each developmental stage of human neocortex. As a case \\nstudy, we also perform particular analysis on the cell developmental \\nstage-specific lncRNA regulation related to 18 known lncRNA biomarkers in autism \\nspectrum disorder. Finally, the comparison result indicates that CDSlncR is an \\neffective method for predicting cell developmental stage-specific lncRNA \\ntargets. CDSlncR is available at https://github.com/linxi159/CDSlncR.\\n\\nCopyright Â© 2023 Huang, Ma and Zhang.\\n\\nDOI: 10.3389/fnmol.2022.1037565\\nPMCID: PMC9880432\\nPMID: 36710930\\n\\nConflict of interest statement: The authors declare that the research was \\nconducted in the absence of any commercial or financial relationships that could \\nbe construed as a potential conflict of interest.\\n\\n   \n",
       "97                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           PeerJ. 2023 Jan 23;11:e14706. doi: 10.7717/peerj.14706. eCollection 2023.\\n\\nA clustering method for small scRNA-seq data based on subspace and weighted \\ndistance.\\n\\nNing Z(1)(2), Dai Z(1), Zhang H(2), Chen Y(1), Yuan Z(1).\\n\\nAuthor information:\\n(1)Hunan Engineering & Technology Research Centre for Agricultural Big Data \\nAnalysis & Decision-Making, Hunan Agricultural University, Changsha, Hunan, \\nChina.\\n(2)Hunan Agricultural University, College of Information and Intelligence, \\nChangsha, Hunan, China.\\n\\nBACKGROUND: Identifying the cell types using unsupervised methods is essential \\nfor scRNA-seq research. However, conventional similarity measures introduce \\nchallenges to single-cell data clustering because of the high dimensional, high \\nnoise, and high dropout.\\nMETHODS: We proposed a clustering method for small ScRNA-seq data based on \\nSubspace and Weighted Distance (SSWD), which follows the assumption that the \\nsets of gene subspace composed of similar density-distributing genes can better \\ndistinguish cell groups. To accurately capture the intrinsic relationship among \\ncells or genes, a new distance metric that combines Euclidean and Pearson \\ndistance through a weighting strategy was proposed. The relative \\nCalinski-Harabasz (CH) index was used to estimate the cluster numbers instead of \\nthe CH index because it is comparable across degrees of freedom.\\nRESULTS: We compared SSWD with seven prevailing methods on eight publicly \\nscRNA-seq datasets. The experimental results show that the SSWD has better \\nclustering accuracy and the partitioning ability of cell groups. SSWD can be \\ndownloaded at https://github.com/ningzilan/SSWD.\\n\\nÂ©2023 Ning et al.\\n\\nDOI: 10.7717/peerj.14706\\nPMCID: PMC9879162\\nPMID: 36710872 [Indexed for MEDLINE]\\n\\nConflict of interest statement: The authors declare there are no competing \\ninterests.\\n\\n   \n",
       "98                                                                                                                                              Methods. 2023 Jan 26;211:1-9. doi: 10.1016/j.ymeth.2023.01.006. Online ahead of \\nprint.\\n\\nPredicting latent lncRNA and cancer metastatic event associations via \\nvariational graph auto-encoder.\\n\\nZhu Y(1), Zhang F(2), Zhang S(3), Yi M(4).\\n\\nAuthor information:\\n(1)School of Automation, China University of Geosciences, 388 Lumo Road, \\nHongshan District, 430074, Wuhan, Hubei, China; Hubei Key Laboratory of Advanced \\nControl and Intelligent Automation for Complex Systems, 388 Lumo Road, Hongshan \\nDistrict, 430074, Wuhan, Hubei, China; Engineering Research Center of \\nIntelligent Technology for Geo-Exploration, 388 Lumo Road, Hongshan District, \\n430074, Wuhan, Hubei, China.\\n(2)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China.\\n(3)College of Life Science and Health, Wuhan University of Science and \\nTechnology, 974 Heping Avenue, Qingshan District, 430081, Wuhan, Hubei, China. \\nElectronic address: shihuazhang@wust.edu.cn.\\n(4)School of Mathematics and Physics, China University of Geosciences, 388 Lumo \\nRoad, Hongshan District, 430074, Wuhan, Hubei, China. Electronic address: \\nmingyi@cug.edu.cn.\\n\\nLong non-coding RNA (lncRNA) are shown to be closely associated with cancer \\nmetastatic events (CME, e.g., cancer cell invasion, intravasation, \\nextravasation, proliferation) that collaboratively accelerate malignant cancer \\nspread and cause high mortality rate in patients. Clinical trials may accurately \\nuncover the relationships between lncRNAs and CMEs; however, it is \\ntime-consuming and expensive. With the accumulation of data, there is an urgent \\nneed to find efficient ways to identify these relationships. Herein, a graph \\nembedding representation-based predictor (VGEA-LCME) for exploring latent \\nlncRNA-CME associations is introduced. In VGEA-LCME, a heterogeneous combined \\nnetwork is constructed by integrating similarity and linkage matrix that can \\nmaintain internal and external characteristics of networks, and a variational \\ngraph auto-encoder serves as a feature generator to represent arbitrary lncRNA \\nand CME pair. The final robustness predicted result is obtained by ensemble \\nclassifier strategy via cross-validation. Experimental comparisons and \\nliterature verification show better remarkable performance of VGEA-LCME, \\nalthough the similarities between CMEs are challenging to calculate. In \\naddition, VGEA-LCME can further identify organ-specific CMEs. To the best of our \\nknowledge, this is the first computational attempt to discover the potential \\nrelationships between lncRNAs and CMEs. It may provide support and new insight \\nfor guiding experimental research of metastatic cancers. The source code and \\ndata are available at https://github.com/zhuyuan-cug/VGAE-LCME.\\n\\nCopyright Â© 2023 Elsevier Inc. All rights reserved.\\n\\nDOI: 10.1016/j.ymeth.2023.01.006\\nPMID: 36709790\\n\\nConflict of interest statement: Declaration of Competing Interest The authors \\ndeclare that there are no conflict of interests, we do not have any possible \\nconflicts of interest.\\n\\n   \n",
       "99  Comput Methods Programs Biomed. 2023 Jan 15;230:107355. doi: \\n10.1016/j.cmpb.2023.107355. Online ahead of print.\\n\\nHigh-throughput 3DRA segmentation of brain vasculature and aneurysms using deep \\nlearning.\\n\\nLin F(1), Xia Y(2), Song S(1), Ravikumar N(1), Frangi AF(3).\\n\\nAuthor information:\\n(1)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK.\\n(2)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK. Electronic address: \\ny.xia@leeds.ac.uk.\\n(3)Centre for Computational Imaging and Simulation Technologies in Biomedicine \\n(CISTIB), The University of Leeds, Leeds LS2 9JT, UK; Leeds Institute for \\nCardiovascular and Metabolic Medicine (LICAMM), School of Medicine, University \\nof Leeds, Leeds LS2 9JT, UK; Medical Imaging Research Center (MIRC), \\nCardiovascular Science and Electronic Engineering Departments, KU Leuven, \\nLeuven, Belgium; Alan Turing Institute, London, UK.\\n\\nBACKGROUND AND OBJECTIVES: Automatic segmentation of the cerebral vasculature \\nand aneurysms facilitates incidental detection of aneurysms. The assessment of \\naneurysm rupture risk assists with pre-operative treatment planning and enables \\nin-silico investigation of cerebral hemodynamics within and in the vicinity of \\naneurysms. However, ensuring precise and robust segmentation of cerebral vessels \\nand aneurysms in neuroimaging modalities such as three-dimensional rotational \\nangiography (3DRA) is challenging. The vasculature constitutes a small \\nproportion of the image volume, resulting in a large class imbalance (relative \\nto surrounding brain tissue). Additionally, aneurysms and vessels have similar \\nimage/appearance characteristics, making it challenging to distinguish the \\naneurysm sac from the vessel lumen.\\nMETHODS: We propose a novel multi-class convolutional neural network to tackle \\nthese challenges and facilitate the automatic segmentation of cerebral vessels \\nand aneurysms in 3DRA images. The proposed model is trained and evaluated on an \\ninternal multi-center dataset and an external publicly available challenge \\ndataset.\\nRESULTS: On the internal clinical dataset, our method consistently outperformed \\nseveral state-of-the-art approaches for vessel and aneurysm segmentation, \\nachieving an average Dice score of 0.81 (0.15 higher than nnUNet) and an average \\nsurface-to-surface error of 0.20Â mm (less than the in-plane resolution \\n(0.35Â mm/pixel)) for aneurysm segmentation; and an average Dice score of 0.91 \\nand average surface-to-surface error of 0.25Â mm for vessel segmentation. In 223 \\ncases of a clinical dataset, our method accurately segmented 190 aneurysm cases.\\nCONCLUSIONS: The proposed approach can help address class imbalance problems and \\ninter-class interference problems in multi-class segmentation. Besides, this \\nmethod performs consistently on clinical datasets from four different sources \\nand the generated results are qualified for hemodynamic simulation. Code \\navailable at https://github.com/cistib/vessel-aneurysm-segmentation.\\n\\nCopyright Â© 2023. Published by Elsevier B.V.\\n\\nDOI: 10.1016/j.cmpb.2023.107355\\nPMID: 36709557\\n\\n   \n",
       "\n",
       "    Repo_created_at  Repo_updated_at  \n",
       "0               NaN              NaN  \n",
       "1               NaN              NaN  \n",
       "2               NaN              NaN  \n",
       "3               NaN              NaN  \n",
       "4               NaN              NaN  \n",
       "..              ...              ...  \n",
       "95              NaN              NaN  \n",
       "96              NaN              NaN  \n",
       "97              NaN              NaN  \n",
       "98              NaN              NaN  \n",
       "99              NaN              NaN  \n",
       "\n",
       "[100 rows x 11 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for id_iter in df1[\"PMID\"]:\n",
    "    \n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"Repo_created_at\"], df1.loc[df1[\"PMID\"] == id_iter,\"Repo_updated_at\"] = get_repo_info(df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_clean\"])\n",
    "df1"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
