{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "2092ef92-81df-4128-a1fa-f728c558bac0",
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime\n",
    "import json\n",
    "import re\n",
    "import time\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import tqdm as tqdm_module\n",
    "import requests\n",
    "\n",
    "%reload_ext autoreload\n",
    "%autoreload 2\n",
    "import pbmd_tools as pbmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d85646ad-ce38-4cb4-803a-85959aaf6d9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 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  !!!!!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "1e66c109-89bd-4f6f-8d99-89ed18a4be3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "db = 'pubmed'\n",
    "domain = 'https://www.ncbi.nlm.nih.gov/entrez/eutils'\n",
    "nresults = 100\n",
    "query = '\"github.com\"[Title/Abstract] NOT \"github.com\"[Title]'\n",
    "retmode = 'json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "e774752a-5e60-4e94-9242-b8a1015a9767",
   "metadata": {},
   "outputs": [],
   "source": [
    "queryLinkSearch = f'{domain}/esearch.fcgi?db={db}&retmax={nresults}&retmode={retmode}&term={query}'\n",
    "response = requests.get(queryLinkSearch)\n",
    "pubmedJson = response.json()\n",
    "results = []\n",
    "for paperId in pubmedJson[\"esearchresult\"][\"idlist\"]:\n",
    "    queryLinkSummary = f'{domain}/esummary.fcgi?db={db}&id={paperId}&retmode={retmode}'\n",
    "    results.append({'paperId': paperId, 'metadata': requests.get(queryLinkSummary).json()})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "5b3727b0-68a6-4591-bfec-26764dd2fcfd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36798707</td>\n",
       "      <td>2023-01-31</td>\n",
       "      <td>10.3389/fpls.2023.1109314</td>\n",
       "      <td>Frontiers in plant science</td>\n",
       "      <td>DFSP: A fast and automatic distance field-base...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36798459</td>\n",
       "      <td>2023-02-08</td>\n",
       "      <td>None</td>\n",
       "      <td>ArXiv</td>\n",
       "      <td>Geometry-Complete Diffusion for 3D Molecule Ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36798308</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>10.1101/2023.02.09.527886</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>IDSL.CSA: Composite Spectra Analysis for Chemi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36798258</td>\n",
       "      <td>2023-02-12</td>\n",
       "      <td>10.1101/2023.02.11.528027</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Foreign RNA spike-ins enable accurate allele-s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36798154</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>10.1101/2023.02.09.527740</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Diffsig: Associating Risk Factors With Mutatio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36740003</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>10.1016/j.ab.2023.115075</td>\n",
       "      <td>Analytical biochemistry</td>\n",
       "      <td>DapNet-HLA: Adaptive dual-attention mechanism ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36739821</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>10.1016/j.compbiomed.2023.106615</td>\n",
       "      <td>Computers in biology and medicine</td>\n",
       "      <td>BX2S-Net: Learning to reconstruct 3D spinal st...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36739616</td>\n",
       "      <td>2023-02-05</td>\n",
       "      <td>10.1002/gepi.22517</td>\n",
       "      <td>Genetic epidemiology</td>\n",
       "      <td>Bayesian multivariant fine mapping using the L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36739333</td>\n",
       "      <td>2023-02-04</td>\n",
       "      <td>10.1038/s41437-023-00595-9</td>\n",
       "      <td>Heredity</td>\n",
       "      <td>Meiosis at three loci in autotetraploids: Prob...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36738608</td>\n",
       "      <td>2023-01-28</td>\n",
       "      <td>10.1016/j.cmpb.2023.107374</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>PyRaDiSe: A Python package for DICOM-RT-based ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                               DOI  \\\n",
       "0   36798707  2023-01-31         10.3389/fpls.2023.1109314   \n",
       "1   36798459  2023-02-08                              None   \n",
       "2   36798308  2023-02-10         10.1101/2023.02.09.527886   \n",
       "3   36798258  2023-02-12         10.1101/2023.02.11.528027   \n",
       "4   36798154  2023-02-10         10.1101/2023.02.09.527740   \n",
       "..       ...         ...                               ...   \n",
       "95  36740003  2023-02-03          10.1016/j.ab.2023.115075   \n",
       "96  36739821  2023-02-02  10.1016/j.compbiomed.2023.106615   \n",
       "97  36739616  2023-02-05                10.1002/gepi.22517   \n",
       "98  36739333  2023-02-04        10.1038/s41437-023-00595-9   \n",
       "99  36738608  2023-01-28        10.1016/j.cmpb.2023.107374   \n",
       "\n",
       "                                         Journal  \\\n",
       "0                     Frontiers in plant science   \n",
       "1                                          ArXiv   \n",
       "2      bioRxiv : the preprint server for biology   \n",
       "3      bioRxiv : the preprint server for biology   \n",
       "4      bioRxiv : the preprint server for biology   \n",
       "..                                           ...   \n",
       "95                       Analytical biochemistry   \n",
       "96             Computers in biology and medicine   \n",
       "97                          Genetic epidemiology   \n",
       "98                                      Heredity   \n",
       "99  Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                Title  \n",
       "0   DFSP: A fast and automatic distance field-base...  \n",
       "1   Geometry-Complete Diffusion for 3D Molecule Ge...  \n",
       "2   IDSL.CSA: Composite Spectra Analysis for Chemi...  \n",
       "3   Foreign RNA spike-ins enable accurate allele-s...  \n",
       "4   Diffsig: Associating Risk Factors With Mutatio...  \n",
       "..                                                ...  \n",
       "95  DapNet-HLA: Adaptive dual-attention mechanism ...  \n",
       "96  BX2S-Net: Learning to reconstruct 3D spinal st...  \n",
       "97  Bayesian multivariant fine mapping using the L...  \n",
       "98  Meiosis at three loci in autotetraploids: Prob...  \n",
       "99  PyRaDiSe: A Python package for DICOM-RT-based ...  \n",
       "\n",
       "[100 rows x 5 columns]"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = pd.DataFrame.from_records(results)\n",
    "df1[\"result\"] = df1[\"metadata\"].apply(lambda x: x[\"result\"])\n",
    "df1[\"PMID\"] = df1[\"paperId\"]\n",
    "df1[\"PubDate\"] = [d[k][\"sortpubdate\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\"]\n",
    "df1[\"PubDate\"] = df1[\"PubDate\"].apply(lambda x: datetime.strptime(x, \"%Y/%m/%d %H:%M\").strftime(\"%Y-%m-%d\") if x else None)\n",
    "\n",
    "id_list = pubmedJson[\"esearchresult\"][\"idlist\"]\n",
    "df1[\"DOI\"] = None\n",
    "for i in range(len(id_list)):\n",
    "    articleid_list = results[i]['metadata']['result'][id_list[i]][\"articleids\"]\n",
    "    for dic in articleid_list:\n",
    "        if 'doi' in dic.values():\n",
    "            df1[\"DOI\"].iloc[i] = dic['value']\n",
    "            \n",
    "df1[\"Journal\"] = [d[k][\"fulljournalname\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\"]\n",
    "df1[\"Title\"] = [d[k][\"title\"] for d in df1[\"result\"] for k in d.keys() if k != \"uids\"]\n",
    "df1 = df1.drop([\"paperId\", \"result\", \"metadata\"], axis=1)\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "cb9e5ea9-f932-41aa-8a6e-db1628a84b8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "rate_limit = 3\n",
    "request_count = 0\n",
    "\n",
    "for id_iter in df1[\"PMID\"]:  \n",
    "    \n",
    "    if request_count >= rate_limit:\n",
    "        \n",
    "        time.sleep(1)\n",
    "        request_count = 0\n",
    "        \n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"Abstract\"] = pbmd.get_abstract(id_iter)\n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_raw\"] = \", \".join(pbmd.get_link(df1, id_iter))\n",
    "    df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_clean\"] = \", \".join(pbmd.clean_link(df1.loc[df1[\"PMID\"] == id_iter,\"GitHub_link_raw\"]))\n",
    "    df1.loc[df1[\"PMID\"] == id_iter, \"Phrase\"] = str(pbmd.get_phrase_with_link(df1, id_iter))\n",
    "    \n",
    "    request_count += 1\n",
    "    \n",
    "df1 = df1[[\"PMID\", \"PubDate\", \"DOI\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"Journal\", \"Title\", \"Phrase\", \"Abstract\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19beac84-c93f-4259-96c4-ac422913bfbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PMID = 36789260 - 2 links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "baaeface-4d91-46c1-8ee4-f972180a1fd1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>DOI</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "      <th>Abstract</th>\n",
       "      <th>Repo_created_at</th>\n",
       "      <th>Repo_updated_at</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36798707</td>\n",
       "      <td>2023-01-31</td>\n",
       "      <td>10.3389/fpls.2023.1109314</td>\n",
       "      <td>github.com/syau-miao/DFSP</td>\n",
       "      <td>https://github.com/syau-miao/DFSP/</td>\n",
       "      <td>Frontiers in plant science</td>\n",
       "      <td>DFSP: A fast and automatic distance field-base...</td>\n",
       "      <td>The source code can be found at \\nhttps://gith...</td>\n",
       "      <td>Front Plant Sci. 2023 Jan 31;14:1109314. doi: ...</td>\n",
       "      <td>2022-09-11</td>\n",
       "      <td>2023-02-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36798459</td>\n",
       "      <td>2023-02-08</td>\n",
       "      <td>None</td>\n",
       "      <td>github.com/BioinfoMachineLearning/bio-diffusion</td>\n",
       "      <td>https://github.com/BioinfoMachineLearning/bio-...</td>\n",
       "      <td>ArXiv</td>\n",
       "      <td>Geometry-Complete Diffusion for 3D Molecule Ge...</td>\n",
       "      <td>The \\nsource code, data, and instructions to t...</td>\n",
       "      <td>ArXiv. 2023 Feb 8:arXiv:2302.04313v1. Preprint...</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>2023-02-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36798308</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>10.1101/2023.02.09.527886</td>\n",
       "      <td>github.com/idslme/IDSL</td>\n",
       "      <td>https://github.com/idslme/IDSL/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>IDSL.CSA: Composite Spectra Analysis for Chemi...</td>\n",
       "      <td>Detailed \\ndocumentation and tutorials are pro...</td>\n",
       "      <td>bioRxiv. 2023 Feb 10:2023.02.09.527886. doi: 1...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36798258</td>\n",
       "      <td>2023-02-12</td>\n",
       "      <td>10.1101/2023.02.11.528027</td>\n",
       "      <td>github.com/gimelbrantlab/controlFreq</td>\n",
       "      <td>https://github.com/gimelbrantlab/controlFreq/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Foreign RNA spike-ins enable accurate allele-s...</td>\n",
       "      <td>VAILABILITY: Analysis pipeline for this approa...</td>\n",
       "      <td>bioRxiv. 2023 Feb 12:2023.02.11.528027. doi: 1...</td>\n",
       "      <td>2022-06-17</td>\n",
       "      <td>2023-01-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36798154</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>10.1101/2023.02.09.527740</td>\n",
       "      <td>github.com/jennprk/diffsig</td>\n",
       "      <td>https://github.com/jennprk/diffsig/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Diffsig: Associating Risk Factors With Mutatio...</td>\n",
       "      <td>Diffsig is implemented as an R package availab...</td>\n",
       "      <td>bioRxiv. 2023 Feb 10:2023.02.09.527740. doi: 1...</td>\n",
       "      <td>2022-04-18</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>36794927</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.1093/bioinformatics/btad093</td>\n",
       "      <td>github.com/GreyGuoweiChen/RNA_virus_detector</td>\n",
       "      <td>https://github.com/GreyGuoweiChen/RNA_virus_de...</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>VirBot: an RNA viral contig detector for metag...</td>\n",
       "      <td>AVAILABILITY AND IMPLEMENTATION: \\nhttps://git...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 16:btad093. doi: 10.1...</td>\n",
       "      <td>2022-07-12</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>36794924</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.1093/bioinformatics/btad092</td>\n",
       "      <td>github.com/G2Lab/ldmat</td>\n",
       "      <td>https://github.com/G2Lab/ldmat/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>LDmat: Efficiently Queryable Compression of Li...</td>\n",
       "      <td>It can also be \\naccessed through https://gith...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 16:btad092. doi: 10.1...</td>\n",
       "      <td>2022-08-01</td>\n",
       "      <td>2022-09-26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>36794913</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.1093/bioinformatics/btad094</td>\n",
       "      <td>github.com/BioColLab/PFresGO</td>\n",
       "      <td>https://github.com/BioColLab/PFresGO/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>PFresGO: an attention mechanism-based deep-lea...</td>\n",
       "      <td>AVAILABILITY: PFresGO is available for academi...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 16:btad094. doi: 10.1...</td>\n",
       "      <td>2022-12-20</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>36794911</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.1093/bioinformatics/btad091</td>\n",
       "      <td>github.com/ccb-hms/BioPlexAnalysis</td>\n",
       "      <td>https://github.com/ccb-hms/BioPlexAnalysis/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>BioPlexR and BioPlexPy: integrated data produc...</td>\n",
       "      <td>pplications and downstream analyses are \\navai...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 16:btad091. doi: 10.1...</td>\n",
       "      <td>2021-04-13</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>36794706</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.1002/mp.16287</td>\n",
       "      <td>github.com/corcor27/BUS-Set</td>\n",
       "      <td>https://github.com/corcor27/BUS-Set/</td>\n",
       "      <td>Medical physics</td>\n",
       "      <td>BUS-Set: A benchmark for quantitative evaluati...</td>\n",
       "      <td>All dataset and \\narchitecture details are ava...</td>\n",
       "      <td>Med Phys. 2023 Feb 16. doi: 10.1002/mp.16287. ...</td>\n",
       "      <td>2020-09-27</td>\n",
       "      <td>2023-02-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>36793016</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1186/s12874-023-01847-2</td>\n",
       "      <td>github.com/ljwa2323/MTNN</td>\n",
       "      <td>https://github.com/ljwa2323/MTNN/</td>\n",
       "      <td>BMC medical research methodology</td>\n",
       "      <td>Propensity score analysis with missing data us...</td>\n",
       "      <td>Our code is publicly available at https://gith...</td>\n",
       "      <td>BMC Med Res Methodol. 2023 Feb 15;23(1):41. do...</td>\n",
       "      <td>2022-05-08</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>36792366</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.1093/gbe/evad020</td>\n",
       "      <td>github.com/isugifNF/polishCLR</td>\n",
       "      <td>https://github.com/isugifNF/polishCLR/</td>\n",
       "      <td>Genome biology and evolution</td>\n",
       "      <td>\"polishCLR: a Nextflow workflow for polishing ...</td>\n",
       "      <td>\\nHere, we describe polishCLR (https://github....</td>\n",
       "      <td>Genome Biol Evol. 2023 Feb 16:evad020. doi: 10...</td>\n",
       "      <td>2021-06-08</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>36792093</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.5534/wjmh.220231</td>\n",
       "      <td>github.com/cgab-ncc/TarGA</td>\n",
       "      <td>https://github.com/cgab-ncc/TarGA/</td>\n",
       "      <td>The world journal of men's health</td>\n",
       "      <td>OGT and FLAD1 Genes Had Significant Prognostic...</td>\n",
       "      <td>Twenty metabolic genes were \\nidentified by ow...</td>\n",
       "      <td>World J Mens Health. 2023 Feb 9. doi: 10.5534/...</td>\n",
       "      <td>2018-01-11</td>\n",
       "      <td>2022-09-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>36791547</td>\n",
       "      <td>2023-02-07</td>\n",
       "      <td>10.1016/j.compbiomed.2023.106635</td>\n",
       "      <td>github.com/Leejyuan/MGNet</td>\n",
       "      <td>https://github.com/Leejyuan/MGNet/</td>\n",
       "      <td>Computers in biology and medicine</td>\n",
       "      <td>MG-Net: Multi-level global-aware network for t...</td>\n",
       "      <td>The code is \\navailable at: https://github.com...</td>\n",
       "      <td>Comput Biol Med. 2023 Feb 7;155:106635. doi: 1...</td>\n",
       "      <td>2022-06-24</td>\n",
       "      <td>2022-11-25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>36791465</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1002/sim.9684</td>\n",
       "      <td>github.com/umich-biostatistics/zGPS</td>\n",
       "      <td>https://github.com/umich-biostatistics/zGPS/</td>\n",
       "      <td>Statistics in medicine</td>\n",
       "      <td>Mining adverse events in large frequency table...</td>\n",
       "      <td>The proposed method \\nis implemented in an R p...</td>\n",
       "      <td>Stat Med. 2023 Feb 15. doi: 10.1002/sim.9684. ...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>36791286</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1021/acs.jcim.2c01498</td>\n",
       "      <td>github.com/GOMC-WSU/MoSDeF-GOMC</td>\n",
       "      <td>https://github.com/GOMC-WSU/MoSDeF-GOMC/</td>\n",
       "      <td>Journal of chemical information and modeling</td>\n",
       "      <td>MoSDeF-GOMC: Python Software for the Creation ...</td>\n",
       "      <td>The MoSDeF-GOMC software is available on GitHu...</td>\n",
       "      <td>J Chem Inf Model. 2023 Feb 15. doi: 10.1021/ac...</td>\n",
       "      <td>2022-03-21</td>\n",
       "      <td>2023-02-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>36790845</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1093/bib/bbad057</td>\n",
       "      <td>github.com/HuiLiuCode/MDBuilder</td>\n",
       "      <td>https://github.com/HuiLiuCode/MDBuilder/</td>\n",
       "      <td>Briefings in bioinformatics</td>\n",
       "      <td>MDBuilder: a PyMOL plugin for the preparation ...</td>\n",
       "      <td>The source code \\nand documentation of MDBuild...</td>\n",
       "      <td>Brief Bioinform. 2023 Feb 15:bbad057. doi: 10....</td>\n",
       "      <td>2017-10-04</td>\n",
       "      <td>2022-10-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>36790067</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1093/bioinformatics/btad089</td>\n",
       "      <td>github.com/HassounLab/GNN-SOM</td>\n",
       "      <td>https://github.com/HassounLab/GNN-SOM/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>Using Graph Neural Networks for Site-of-Metabo...</td>\n",
       "      <td>AVAILABILITY: A python implementation of the t...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 15:btad089. doi: 10.1...</td>\n",
       "      <td>2022-07-19</td>\n",
       "      <td>2023-02-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>36790056</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1093/bioinformatics/btad087</td>\n",
       "      <td>github.com/meidanis-lab/rank-indel</td>\n",
       "      <td>https://github.com/meidanis-lab/rank-indel/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>Generalizations of the Genomic Rank Distance t...</td>\n",
       "      <td>AVAILABILITY: Code and instructions available ...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 15:btad087. doi: 10.1...</td>\n",
       "      <td>2022-08-12</td>\n",
       "      <td>2023-02-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>36790055</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1093/bioinformatics/btad072</td>\n",
       "      <td>github.com/CompBioIPM/wpLogicNet</td>\n",
       "      <td>https://github.com/CompBioIPM/wpLogicNet/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>wpLogicNet: logic gate and structure inference...</td>\n",
       "      <td>AVAILABILITY AND IMPLEMENTATION: The datasets ...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 15:btad072. doi: 10.1...</td>\n",
       "      <td>2022-07-23</td>\n",
       "      <td>2023-02-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>36789431</td>\n",
       "      <td>2023-01-29</td>\n",
       "      <td>10.1101/2023.01.27.525971</td>\n",
       "      <td>github.com/chrishil1/ChromDL</td>\n",
       "      <td>https://github.com/chrishil1/ChromDL/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>ChromDL: A Next-Generation Regulatory DNA Clas...</td>\n",
       "      <td>AVAILABILITY: The ChromDL source code can be f...</td>\n",
       "      <td>bioRxiv. 2023 Jan 29:2023.01.27.525971. doi: 1...</td>\n",
       "      <td>2021-08-02</td>\n",
       "      <td>2023-02-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>36789427</td>\n",
       "      <td>2023-01-20</td>\n",
       "      <td>10.1101/2023.01.19.524495</td>\n",
       "      <td>github.com/biomedbigdata/circRNA-sponging</td>\n",
       "      <td>https://github.com/biomedbigdata/circRNA-spong...</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>circRNA-sponging: a pipeline for extensive ana...</td>\n",
       "      <td>AVAILABILITY: https://github.com/biomedbigdata...</td>\n",
       "      <td>bioRxiv. 2023 Jan 20:2023.01.19.524495. doi: 1...</td>\n",
       "      <td>2021-01-08</td>\n",
       "      <td>2023-01-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>36789411</td>\n",
       "      <td>2023-01-28</td>\n",
       "      <td>10.1101/2023.01.27.525885</td>\n",
       "      <td>github.com/KULL-Centre/FRETpredict</td>\n",
       "      <td>https://github.com/KULL-Centre/FRETpredict/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>FRETpredict: A Python package for FRET efficie...</td>\n",
       "      <td>RETpredict is open \\nsource (GPLv3) and is ava...</td>\n",
       "      <td>bioRxiv. 2023 Jan 28:2023.01.27.525885. doi: 1...</td>\n",
       "      <td>2020-08-28</td>\n",
       "      <td>2023-02-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>36789294</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>10.1093/bioadv/vbad005</td>\n",
       "      <td>github.com/PNNL-CompBio/Snekmer</td>\n",
       "      <td>https://github.com/PNNL-CompBio/Snekmer/</td>\n",
       "      <td>Bioinformatics advances</td>\n",
       "      <td>Snekmer: a scalable pipeline for protein seque...</td>\n",
       "      <td>\\nCode and data used in this article, along wi...</td>\n",
       "      <td>Bioinform Adv. 2023 Feb 2;3(1):vbad005. doi: 1...</td>\n",
       "      <td>2020-11-16</td>\n",
       "      <td>2023-02-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>36789260</td>\n",
       "      <td>2023-01-27</td>\n",
       "      <td>10.1016/j.csbj.2023.01.034</td>\n",
       "      <td>github.com/aitgon/vtam, github.com/aitgon/vtam...</td>\n",
       "      <td>https://github.com/aitgon/vtam, github.com/ait...</td>\n",
       "      <td>Computational and structural biotechnology jou...</td>\n",
       "      <td>VTAM: A robust pipeline for validating metabar...</td>\n",
       "      <td>VTAM is implemented in Python and available fr...</td>\n",
       "      <td>Comput Struct Biotechnol J. 2023 Jan 27;21:115...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>36788560</td>\n",
       "      <td>2023-02-14</td>\n",
       "      <td>10.1186/s12911-023-02129-z</td>\n",
       "      <td>github.com/1152545264/SwinUnet3D</td>\n",
       "      <td>https://github.com/1152545264/SwinUnet3D/</td>\n",
       "      <td>BMC medical informatics and decision making</td>\n",
       "      <td>Swin Unet3D: a three-dimensional medical image...</td>\n",
       "      <td>The code can be found at \\nhttps://github.com/...</td>\n",
       "      <td>BMC Med Inform Decis Mak. 2023 Feb 14;23(1):33...</td>\n",
       "      <td>2021-12-21</td>\n",
       "      <td>2022-12-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>36788531</td>\n",
       "      <td>2023-02-14</td>\n",
       "      <td>10.1186/s12859-023-05166-7</td>\n",
       "      <td>github.com/kramerlab/Multi-Omics_analysis</td>\n",
       "      <td>https://github.com/kramerlab/Multi-Omics_analy...</td>\n",
       "      <td>BMC bioinformatics</td>\n",
       "      <td>A fair experimental comparison of neural netwo...</td>\n",
       "      <td>The source code of all experiments is availabl...</td>\n",
       "      <td>BMC Bioinformatics. 2023 Feb 14;24(1):45. doi:...</td>\n",
       "      <td>2022-07-20</td>\n",
       "      <td>2023-02-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>36787234</td>\n",
       "      <td>2023-02-14</td>\n",
       "      <td>10.1093/bfgp/elad006</td>\n",
       "      <td>github.com/YanhaoHuo/dysregulated-DNBs</td>\n",
       "      <td>https://github.com/YanhaoHuo/dysregulated-DNBs/</td>\n",
       "      <td>Briefings in functional genomics</td>\n",
       "      <td>Detecting early-warning signals for influenza ...</td>\n",
       "      <td>The source code of this paper can be \\nfreely ...</td>\n",
       "      <td>Brief Funct Genomics. 2023 Feb 14:elad006. doi...</td>\n",
       "      <td>2022-11-24</td>\n",
       "      <td>2022-11-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>36786404</td>\n",
       "      <td>2023-02-14</td>\n",
       "      <td>10.1093/bioinformatics/btad081</td>\n",
       "      <td>github.com/mofradlab</td>\n",
       "      <td>https://github.com/mofradlab/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>GO Bench: Shared-hub for Universal Benchmarkin...</td>\n",
       "      <td>ode is hosted at github.com/mofradlab,</td>\n",
       "      <td>Bioinformatics. 2023 Feb 14:btad081. doi: 10.1...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>36786403</td>\n",
       "      <td>2023-02-14</td>\n",
       "      <td>10.1093/bioinformatics/btad088</td>\n",
       "      <td>github.com/leopold-weidner/DBDIpy</td>\n",
       "      <td>https://github.com/leopold-weidner/DBDIpy/</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>DBDIpy: a Python library for processing of unt...</td>\n",
       "      <td>rg/project/DBDIpy) or from GitHub \\n(https://g...</td>\n",
       "      <td>Bioinformatics. 2023 Feb 14:btad088. doi: 10.1...</td>\n",
       "      <td>2022-10-18</td>\n",
       "      <td>2023-02-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>36785708</td>\n",
       "      <td>2023-02-08</td>\n",
       "      <td>10.7717/peerj.14779</td>\n",
       "      <td>github.com/EESI/Complet-Plus</td>\n",
       "      <td>https://github.com/EESI/Complet-Plus/</td>\n",
       "      <td>PeerJ</td>\n",
       "      <td>Complet+: a computationally scalable method to...</td>\n",
       "      <td>Code and \\nsupplementary information is availa...</td>\n",
       "      <td>PeerJ. 2023 Feb 8;11:e14779. doi: 10.7717/peer...</td>\n",
       "      <td>2022-05-24</td>\n",
       "      <td>2022-05-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>36781633</td>\n",
       "      <td>2023-01-01</td>\n",
       "      <td>10.1007/978-1-0716-3024-2_3</td>\n",
       "      <td>github.com/wendelljpereira/DArTseqMet</td>\n",
       "      <td>https://github.com/wendelljpereira/DArTseqMet/</td>\n",
       "      <td>Methods in molecular biology (Clifton, N.J.)</td>\n",
       "      <td>Computational Protocol for DNA Methylation Pro...</td>\n",
       "      <td>The software is available at \\nhttps://github....</td>\n",
       "      <td>Methods Mol Biol. 2023;2638:23-36. doi: 10.100...</td>\n",
       "      <td>2022-06-17</td>\n",
       "      <td>2022-09-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>36780251</td>\n",
       "      <td>2022-09-01</td>\n",
       "      <td>10.1007/978-3-031-21014-3_18</td>\n",
       "      <td>github.com/freshman97/LinTransUNet</td>\n",
       "      <td>https://github.com/freshman97/LinTransUNet/</td>\n",
       "      <td>Machine learning in medical imaging. MLMI (Wor...</td>\n",
       "      <td>Dynamic Linear Transformer for 3D Biomedical I...</td>\n",
       "      <td>Code is available \\nhttps://github.com/freshma...</td>\n",
       "      <td>Mach Learn Med Imaging. 2022 Sep;13583:171-180...</td>\n",
       "      <td>2022-06-01</td>\n",
       "      <td>2022-10-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>36780239</td>\n",
       "      <td>2022-09-01</td>\n",
       "      <td>10.1007/978-3-031-16437-8_15</td>\n",
       "      <td>github.com/nikhilroxtomar/TGANet</td>\n",
       "      <td>https://github.com/nikhilroxtomar/TGANet/</td>\n",
       "      <td>Medical image computing and computer-assisted ...</td>\n",
       "      <td>TGANet: Text-guided attention for improved pol...</td>\n",
       "      <td>Codes are available at \\nhttps://github.com/ni...</td>\n",
       "      <td>Med Image Comput Comput Assist Interv. 2022 Se...</td>\n",
       "      <td>2022-05-09</td>\n",
       "      <td>2023-02-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>36778474</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>10.1101/2023.02.02.526809</td>\n",
       "      <td>github.com/SlavovLab/DO-MS</td>\n",
       "      <td>https://github.com/SlavovLab/DO-MS/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Data-Driven Optimization of DIA Mass-Spectrome...</td>\n",
       "      <td>he source code is \\navailable at: github.com/S...</td>\n",
       "      <td>bioRxiv. 2023 Feb 3:2023.02.02.526809. doi: 10...</td>\n",
       "      <td>2018-07-20</td>\n",
       "      <td>2023-02-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36778452</td>\n",
       "      <td>2023-02-04</td>\n",
       "      <td>10.1101/2023.02.03.527015</td>\n",
       "      <td>github.com/AlexandrovLab/SigProfilerMatrixGene...</td>\n",
       "      <td>https://github.com/AlexandrovLab/SigProfilerMa...</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Visualizing and exploring patterns of large mu...</td>\n",
       "      <td>The tool is freely available at \\nhttps://gith...</td>\n",
       "      <td>bioRxiv. 2023 Feb 4:2023.02.03.527015. doi: 10...</td>\n",
       "      <td>2018-08-27</td>\n",
       "      <td>2023-02-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>36778443</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>10.1101/2023.02.02.526869</td>\n",
       "      <td>github.com/sandybioteck/RTCpredictor</td>\n",
       "      <td>https://github.com/sandybioteck/RTCpredictor/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>RTCpredictor: Identification of Read-Through C...</td>\n",
       "      <td>In addition, \\nRTCpredictor ( https://github.c...</td>\n",
       "      <td>bioRxiv. 2023 Feb 3:2023.02.02.526869. doi: 10...</td>\n",
       "      <td>2021-11-09</td>\n",
       "      <td>2022-02-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>36778393</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>10.1101/2023.02.02.526837</td>\n",
       "      <td>github.com/jessicabonnie/dandd</td>\n",
       "      <td>https://github.com/jessicabonnie/dandd/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>DandD: efficient measurement of sequence growt...</td>\n",
       "      <td>DandD is open source software available at: \\n...</td>\n",
       "      <td>bioRxiv. 2023 Feb 3:2023.02.02.526837. doi: 10...</td>\n",
       "      <td>2022-04-25</td>\n",
       "      <td>2023-02-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>36778280</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>10.1101/2023.01.30.526317</td>\n",
       "      <td>github.com/AnantharamanLab/ViWrap</td>\n",
       "      <td>https://github.com/AnantharamanLab/ViWrap/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>ViWrap: A modular pipeline to identify, bin, c...</td>\n",
       "      <td>ViWrap is publicly available via GitHub ( \\nht...</td>\n",
       "      <td>bioRxiv. 2023 Feb 2:2023.01.30.526317. doi: 10...</td>\n",
       "      <td>2022-09-08</td>\n",
       "      <td>2023-02-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>36778272</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>10.1101/2023.01.29.23285172</td>\n",
       "      <td>github.com/Zhizhen-Zhao/IPTW-TBI</td>\n",
       "      <td>https://github.com/Zhizhen-Zhao/IPTW-TBI/</td>\n",
       "      <td>medRxiv : the preprint server for health sciences</td>\n",
       "      <td>Data Imputation for Clinical Trial Emulation: ...</td>\n",
       "      <td>The \\nexperiment details and code are availabl...</td>\n",
       "      <td>medRxiv. 2023 Feb 2:2023.01.29.23285172. doi: ...</td>\n",
       "      <td>2022-03-07</td>\n",
       "      <td>2022-03-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>36778267</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>10.1101/2023.02.02.526854</td>\n",
       "      <td>github.com/KoslickiLab/L2-UniFrac</td>\n",
       "      <td>https://github.com/KoslickiLab/L2-UniFrac/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Finding phylogeny-aware and biologically meani...</td>\n",
       "      <td>A \\nprototype implementation is provided at: \\...</td>\n",
       "      <td>bioRxiv. 2023 Feb 3:2023.02.02.526854. doi: 10...</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-01-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>36778023</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>10.3389/fphar.2023.1008330</td>\n",
       "      <td>github.com/PreMedKB/PAnno</td>\n",
       "      <td>https://github.com/PreMedKB/PAnno/</td>\n",
       "      <td>Frontiers in pharmacology</td>\n",
       "      <td>PAnno: A pharmacogenomics annotation tool for ...</td>\n",
       "      <td>PAnno is available at \\nhttps://github.com/Pre...</td>\n",
       "      <td>Front Pharmacol. 2023 Jan 26;14:1008330. doi: ...</td>\n",
       "      <td>2022-07-26</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>36777556</td>\n",
       "      <td>2023-05-01</td>\n",
       "      <td>10.1016/j.bspc.2023.104672</td>\n",
       "      <td>github.com/YishuLiu/SA-COVID</td>\n",
       "      <td>https://github.com/YishuLiu/SA-COVID/</td>\n",
       "      <td>Biomedical signal processing and control</td>\n",
       "      <td>Self-paced Multi-view Learning for CT-based se...</td>\n",
       "      <td>The SA-COVID dataset is publicly released at \\...</td>\n",
       "      <td>Biomed Signal Process Control. 2023 May;83:104...</td>\n",
       "      <td>2022-03-23</td>\n",
       "      <td>2022-03-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>36777398</td>\n",
       "      <td>2022-07-01</td>\n",
       "      <td>10.1109/CBMS55023.2022.00063</td>\n",
       "      <td>github.com/nikhilroxtomar/MKDCNet</td>\n",
       "      <td>https://github.com/nikhilroxtomar/MKDCNet/</td>\n",
       "      <td>Proceedings. IEEE International Symposium on C...</td>\n",
       "      <td>Automatic Polyp Segmentation with Multiple Ker...</td>\n",
       "      <td>The code of the proposed \\nMKDCNet is availabl...</td>\n",
       "      <td>Proc IEEE Int Symp Comput Based Med Syst. 2022...</td>\n",
       "      <td>2022-05-12</td>\n",
       "      <td>2022-10-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>36777397</td>\n",
       "      <td>2022-07-01</td>\n",
       "      <td>10.1109/CBMS55023.2022.00064</td>\n",
       "      <td>github.com/NoviceMAn-prog/FocalConvNet</td>\n",
       "      <td>https://github.com/NoviceMAn-prog/FocalConvNet/</td>\n",
       "      <td>Proceedings. IEEE International Symposium on C...</td>\n",
       "      <td>Video Capsule Endoscopy Classification using F...</td>\n",
       "      <td>The code of the proposed FocalConvNet is \\nava...</td>\n",
       "      <td>Proc IEEE Int Symp Comput Based Med Syst. 2022...</td>\n",
       "      <td>2022-05-15</td>\n",
       "      <td>2022-08-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>36776914</td>\n",
       "      <td>2023-02-01</td>\n",
       "      <td>10.1016/j.heliyon.2023.e13545</td>\n",
       "      <td>github.com/peterman65</td>\n",
       "      <td>https://github.com/peterman65/</td>\n",
       "      <td>Heliyon</td>\n",
       "      <td>A multistate model and its standalone tool to ...</td>\n",
       "      <td>CONCLUSIONS: Our easy-to-use and freely access...</td>\n",
       "      <td>Heliyon. 2023 Feb;9(2):e13545. doi: 10.1016/j....</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>36776820</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>None</td>\n",
       "      <td>github.com/rohdelab/nuclear_morphometry</td>\n",
       "      <td>https://github.com/rohdelab/nuclear_morphometry/</td>\n",
       "      <td>ArXiv</td>\n",
       "      <td>Transport-based morphometry of nuclear structu...</td>\n",
       "      <td>The source codes implementing our method is av...</td>\n",
       "      <td>ArXiv. 2023 Feb 2:arXiv:2302.01449v1. Preprint...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>36775347</td>\n",
       "      <td>2022-10-01</td>\n",
       "      <td>10.1038/s41374-022-00801-y</td>\n",
       "      <td>github.com/wangwangCCChen/CNN-Cox</td>\n",
       "      <td>https://github.com/wangwangCCChen/CNN-Cox/</td>\n",
       "      <td>Laboratory investigation; a journal of technic...</td>\n",
       "      <td>A convolutional neural network model for survi...</td>\n",
       "      <td>The codes are available \\nat the GitHub websit...</td>\n",
       "      <td>Lab Invest. 2022 Oct;102(10):1064-1074. doi: 1...</td>\n",
       "      <td>2022-03-23</td>\n",
       "      <td>2022-12-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>36775056</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>10.1016/j.gpb.2023.01.006</td>\n",
       "      <td>github.com/ddhostallero/tindl</td>\n",
       "      <td>https://github.com/ddhostallero/tindl/</td>\n",
       "      <td>Genomics, proteomics &amp; bioinformatics</td>\n",
       "      <td>Preclinical-to-clinical Anti-cancer Drug Respo...</td>\n",
       "      <td>The code can be accessed at https://github.com...</td>\n",
       "      <td>Genomics Proteomics Bioinformatics. 2023 Feb 1...</td>\n",
       "      <td>2021-06-11</td>\n",
       "      <td>2022-06-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>36774895</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>10.1016/j.compbiomed.2023.106634</td>\n",
       "      <td>github.com/doganlab/scmags</td>\n",
       "      <td>https://github.com/doganlab/scmags/</td>\n",
       "      <td>Computers in biology and medicine</td>\n",
       "      <td>scMAGS: Marker gene selection from scRNA-seq d...</td>\n",
       "      <td>scMAGS is \\nmade freely available at https://g...</td>\n",
       "      <td>Comput Biol Med. 2023 Feb 9;155:106634. doi: 1...</td>\n",
       "      <td>2022-03-05</td>\n",
       "      <td>2022-03-05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate                               DOI  \\\n",
       "0   36798707  2023-01-31         10.3389/fpls.2023.1109314   \n",
       "1   36798459  2023-02-08                              None   \n",
       "2   36798308  2023-02-10         10.1101/2023.02.09.527886   \n",
       "3   36798258  2023-02-12         10.1101/2023.02.11.528027   \n",
       "4   36798154  2023-02-10         10.1101/2023.02.09.527740   \n",
       "5   36794927  2023-02-16    10.1093/bioinformatics/btad093   \n",
       "6   36794924  2023-02-16    10.1093/bioinformatics/btad092   \n",
       "7   36794913  2023-02-16    10.1093/bioinformatics/btad094   \n",
       "8   36794911  2023-02-16    10.1093/bioinformatics/btad091   \n",
       "9   36794706  2023-02-16                  10.1002/mp.16287   \n",
       "10  36793016  2023-02-15        10.1186/s12874-023-01847-2   \n",
       "11  36792366  2023-02-16               10.1093/gbe/evad020   \n",
       "12  36792093  2023-02-09               10.5534/wjmh.220231   \n",
       "13  36791547  2023-02-07  10.1016/j.compbiomed.2023.106635   \n",
       "14  36791465  2023-02-15                  10.1002/sim.9684   \n",
       "15  36791286  2023-02-15          10.1021/acs.jcim.2c01498   \n",
       "16  36790845  2023-02-15               10.1093/bib/bbad057   \n",
       "17  36790067  2023-02-15    10.1093/bioinformatics/btad089   \n",
       "18  36790056  2023-02-15    10.1093/bioinformatics/btad087   \n",
       "19  36790055  2023-02-15    10.1093/bioinformatics/btad072   \n",
       "20  36789431  2023-01-29         10.1101/2023.01.27.525971   \n",
       "21  36789427  2023-01-20         10.1101/2023.01.19.524495   \n",
       "22  36789411  2023-01-28         10.1101/2023.01.27.525885   \n",
       "23  36789294  2023-02-02            10.1093/bioadv/vbad005   \n",
       "24  36789260  2023-01-27        10.1016/j.csbj.2023.01.034   \n",
       "25  36788560  2023-02-14        10.1186/s12911-023-02129-z   \n",
       "26  36788531  2023-02-14        10.1186/s12859-023-05166-7   \n",
       "27  36787234  2023-02-14              10.1093/bfgp/elad006   \n",
       "28  36786404  2023-02-14    10.1093/bioinformatics/btad081   \n",
       "29  36786403  2023-02-14    10.1093/bioinformatics/btad088   \n",
       "30  36785708  2023-02-08               10.7717/peerj.14779   \n",
       "31  36781633  2023-01-01       10.1007/978-1-0716-3024-2_3   \n",
       "32  36780251  2022-09-01      10.1007/978-3-031-21014-3_18   \n",
       "33  36780239  2022-09-01      10.1007/978-3-031-16437-8_15   \n",
       "34  36778474  2023-02-03         10.1101/2023.02.02.526809   \n",
       "35  36778452  2023-02-04         10.1101/2023.02.03.527015   \n",
       "36  36778443  2023-02-03         10.1101/2023.02.02.526869   \n",
       "37  36778393  2023-02-03         10.1101/2023.02.02.526837   \n",
       "38  36778280  2023-02-02         10.1101/2023.01.30.526317   \n",
       "39  36778272  2023-02-02       10.1101/2023.01.29.23285172   \n",
       "40  36778267  2023-02-03         10.1101/2023.02.02.526854   \n",
       "41  36778023  2023-01-26        10.3389/fphar.2023.1008330   \n",
       "42  36777556  2023-05-01        10.1016/j.bspc.2023.104672   \n",
       "43  36777398  2022-07-01      10.1109/CBMS55023.2022.00063   \n",
       "44  36777397  2022-07-01      10.1109/CBMS55023.2022.00064   \n",
       "45  36776914  2023-02-01     10.1016/j.heliyon.2023.e13545   \n",
       "46  36776820  2023-02-02                              None   \n",
       "47  36775347  2022-10-01        10.1038/s41374-022-00801-y   \n",
       "48  36775056  2023-02-10         10.1016/j.gpb.2023.01.006   \n",
       "49  36774895  2023-02-09  10.1016/j.compbiomed.2023.106634   \n",
       "\n",
       "                                      GitHub_link_raw  \\\n",
       "0                           github.com/syau-miao/DFSP   \n",
       "1     github.com/BioinfoMachineLearning/bio-diffusion   \n",
       "2                              github.com/idslme/IDSL   \n",
       "3                github.com/gimelbrantlab/controlFreq   \n",
       "4                          github.com/jennprk/diffsig   \n",
       "5        github.com/GreyGuoweiChen/RNA_virus_detector   \n",
       "6                              github.com/G2Lab/ldmat   \n",
       "7                        github.com/BioColLab/PFresGO   \n",
       "8                  github.com/ccb-hms/BioPlexAnalysis   \n",
       "9                         github.com/corcor27/BUS-Set   \n",
       "10                           github.com/ljwa2323/MTNN   \n",
       "11                      github.com/isugifNF/polishCLR   \n",
       "12                          github.com/cgab-ncc/TarGA   \n",
       "13                          github.com/Leejyuan/MGNet   \n",
       "14                github.com/umich-biostatistics/zGPS   \n",
       "15                    github.com/GOMC-WSU/MoSDeF-GOMC   \n",
       "16                    github.com/HuiLiuCode/MDBuilder   \n",
       "17                      github.com/HassounLab/GNN-SOM   \n",
       "18                 github.com/meidanis-lab/rank-indel   \n",
       "19                   github.com/CompBioIPM/wpLogicNet   \n",
       "20                       github.com/chrishil1/ChromDL   \n",
       "21          github.com/biomedbigdata/circRNA-sponging   \n",
       "22                 github.com/KULL-Centre/FRETpredict   \n",
       "23                    github.com/PNNL-CompBio/Snekmer   \n",
       "24  github.com/aitgon/vtam, github.com/aitgon/vtam...   \n",
       "25                   github.com/1152545264/SwinUnet3D   \n",
       "26          github.com/kramerlab/Multi-Omics_analysis   \n",
       "27             github.com/YanhaoHuo/dysregulated-DNBs   \n",
       "28                               github.com/mofradlab   \n",
       "29                  github.com/leopold-weidner/DBDIpy   \n",
       "30                       github.com/EESI/Complet-Plus   \n",
       "31              github.com/wendelljpereira/DArTseqMet   \n",
       "32                 github.com/freshman97/LinTransUNet   \n",
       "33                   github.com/nikhilroxtomar/TGANet   \n",
       "34                         github.com/SlavovLab/DO-MS   \n",
       "35  github.com/AlexandrovLab/SigProfilerMatrixGene...   \n",
       "36               github.com/sandybioteck/RTCpredictor   \n",
       "37                     github.com/jessicabonnie/dandd   \n",
       "38                  github.com/AnantharamanLab/ViWrap   \n",
       "39                   github.com/Zhizhen-Zhao/IPTW-TBI   \n",
       "40                  github.com/KoslickiLab/L2-UniFrac   \n",
       "41                          github.com/PreMedKB/PAnno   \n",
       "42                       github.com/YishuLiu/SA-COVID   \n",
       "43                  github.com/nikhilroxtomar/MKDCNet   \n",
       "44             github.com/NoviceMAn-prog/FocalConvNet   \n",
       "45                              github.com/peterman65   \n",
       "46            github.com/rohdelab/nuclear_morphometry   \n",
       "47                  github.com/wangwangCCChen/CNN-Cox   \n",
       "48                      github.com/ddhostallero/tindl   \n",
       "49                         github.com/doganlab/scmags   \n",
       "\n",
       "                                    GitHub_link_clean  \\\n",
       "0                  https://github.com/syau-miao/DFSP/   \n",
       "1   https://github.com/BioinfoMachineLearning/bio-...   \n",
       "2                     https://github.com/idslme/IDSL/   \n",
       "3       https://github.com/gimelbrantlab/controlFreq/   \n",
       "4                 https://github.com/jennprk/diffsig/   \n",
       "5   https://github.com/GreyGuoweiChen/RNA_virus_de...   \n",
       "6                     https://github.com/G2Lab/ldmat/   \n",
       "7               https://github.com/BioColLab/PFresGO/   \n",
       "8         https://github.com/ccb-hms/BioPlexAnalysis/   \n",
       "9                https://github.com/corcor27/BUS-Set/   \n",
       "10                  https://github.com/ljwa2323/MTNN/   \n",
       "11             https://github.com/isugifNF/polishCLR/   \n",
       "12                 https://github.com/cgab-ncc/TarGA/   \n",
       "13                 https://github.com/Leejyuan/MGNet/   \n",
       "14       https://github.com/umich-biostatistics/zGPS/   \n",
       "15           https://github.com/GOMC-WSU/MoSDeF-GOMC/   \n",
       "16           https://github.com/HuiLiuCode/MDBuilder/   \n",
       "17             https://github.com/HassounLab/GNN-SOM/   \n",
       "18        https://github.com/meidanis-lab/rank-indel/   \n",
       "19          https://github.com/CompBioIPM/wpLogicNet/   \n",
       "20              https://github.com/chrishil1/ChromDL/   \n",
       "21  https://github.com/biomedbigdata/circRNA-spong...   \n",
       "22        https://github.com/KULL-Centre/FRETpredict/   \n",
       "23           https://github.com/PNNL-CompBio/Snekmer/   \n",
       "24  https://github.com/aitgon/vtam, github.com/ait...   \n",
       "25          https://github.com/1152545264/SwinUnet3D/   \n",
       "26  https://github.com/kramerlab/Multi-Omics_analy...   \n",
       "27    https://github.com/YanhaoHuo/dysregulated-DNBs/   \n",
       "28                      https://github.com/mofradlab/   \n",
       "29         https://github.com/leopold-weidner/DBDIpy/   \n",
       "30              https://github.com/EESI/Complet-Plus/   \n",
       "31     https://github.com/wendelljpereira/DArTseqMet/   \n",
       "32        https://github.com/freshman97/LinTransUNet/   \n",
       "33          https://github.com/nikhilroxtomar/TGANet/   \n",
       "34                https://github.com/SlavovLab/DO-MS/   \n",
       "35  https://github.com/AlexandrovLab/SigProfilerMa...   \n",
       "36      https://github.com/sandybioteck/RTCpredictor/   \n",
       "37            https://github.com/jessicabonnie/dandd/   \n",
       "38         https://github.com/AnantharamanLab/ViWrap/   \n",
       "39          https://github.com/Zhizhen-Zhao/IPTW-TBI/   \n",
       "40         https://github.com/KoslickiLab/L2-UniFrac/   \n",
       "41                 https://github.com/PreMedKB/PAnno/   \n",
       "42              https://github.com/YishuLiu/SA-COVID/   \n",
       "43         https://github.com/nikhilroxtomar/MKDCNet/   \n",
       "44    https://github.com/NoviceMAn-prog/FocalConvNet/   \n",
       "45                     https://github.com/peterman65/   \n",
       "46   https://github.com/rohdelab/nuclear_morphometry/   \n",
       "47         https://github.com/wangwangCCChen/CNN-Cox/   \n",
       "48             https://github.com/ddhostallero/tindl/   \n",
       "49                https://github.com/doganlab/scmags/   \n",
       "\n",
       "                                              Journal  \\\n",
       "0                          Frontiers in plant science   \n",
       "1                                               ArXiv   \n",
       "2           bioRxiv : the preprint server for biology   \n",
       "3           bioRxiv : the preprint server for biology   \n",
       "4           bioRxiv : the preprint server for biology   \n",
       "5                    Bioinformatics (Oxford, England)   \n",
       "6                    Bioinformatics (Oxford, England)   \n",
       "7                    Bioinformatics (Oxford, England)   \n",
       "8                    Bioinformatics (Oxford, England)   \n",
       "9                                     Medical physics   \n",
       "10                   BMC medical research methodology   \n",
       "11                       Genome biology and evolution   \n",
       "12                  The world journal of men's health   \n",
       "13                  Computers in biology and medicine   \n",
       "14                             Statistics in medicine   \n",
       "15       Journal of chemical information and modeling   \n",
       "16                        Briefings in bioinformatics   \n",
       "17                   Bioinformatics (Oxford, England)   \n",
       "18                   Bioinformatics (Oxford, England)   \n",
       "19                   Bioinformatics (Oxford, England)   \n",
       "20          bioRxiv : the preprint server for biology   \n",
       "21          bioRxiv : the preprint server for biology   \n",
       "22          bioRxiv : the preprint server for biology   \n",
       "23                            Bioinformatics advances   \n",
       "24  Computational and structural biotechnology jou...   \n",
       "25        BMC medical informatics and decision making   \n",
       "26                                 BMC bioinformatics   \n",
       "27                   Briefings in functional genomics   \n",
       "28                   Bioinformatics (Oxford, England)   \n",
       "29                   Bioinformatics (Oxford, England)   \n",
       "30                                              PeerJ   \n",
       "31       Methods in molecular biology (Clifton, N.J.)   \n",
       "32  Machine learning in medical imaging. MLMI (Wor...   \n",
       "33  Medical image computing and computer-assisted ...   \n",
       "34          bioRxiv : the preprint server for biology   \n",
       "35          bioRxiv : the preprint server for biology   \n",
       "36          bioRxiv : the preprint server for biology   \n",
       "37          bioRxiv : the preprint server for biology   \n",
       "38          bioRxiv : the preprint server for biology   \n",
       "39  medRxiv : the preprint server for health sciences   \n",
       "40          bioRxiv : the preprint server for biology   \n",
       "41                          Frontiers in pharmacology   \n",
       "42           Biomedical signal processing and control   \n",
       "43  Proceedings. IEEE International Symposium on C...   \n",
       "44  Proceedings. IEEE International Symposium on C...   \n",
       "45                                            Heliyon   \n",
       "46                                              ArXiv   \n",
       "47  Laboratory investigation; a journal of technic...   \n",
       "48              Genomics, proteomics & bioinformatics   \n",
       "49                  Computers in biology and medicine   \n",
       "\n",
       "                                                Title  \\\n",
       "0   DFSP: A fast and automatic distance field-base...   \n",
       "1   Geometry-Complete Diffusion for 3D Molecule Ge...   \n",
       "2   IDSL.CSA: Composite Spectra Analysis for Chemi...   \n",
       "3   Foreign RNA spike-ins enable accurate allele-s...   \n",
       "4   Diffsig: Associating Risk Factors With Mutatio...   \n",
       "5   VirBot: an RNA viral contig detector for metag...   \n",
       "6   LDmat: Efficiently Queryable Compression of Li...   \n",
       "7   PFresGO: an attention mechanism-based deep-lea...   \n",
       "8   BioPlexR and BioPlexPy: integrated data produc...   \n",
       "9   BUS-Set: A benchmark for quantitative evaluati...   \n",
       "10  Propensity score analysis with missing data us...   \n",
       "11  \"polishCLR: a Nextflow workflow for polishing ...   \n",
       "12  OGT and FLAD1 Genes Had Significant Prognostic...   \n",
       "13  MG-Net: Multi-level global-aware network for t...   \n",
       "14  Mining adverse events in large frequency table...   \n",
       "15  MoSDeF-GOMC: Python Software for the Creation ...   \n",
       "16  MDBuilder: a PyMOL plugin for the preparation ...   \n",
       "17  Using Graph Neural Networks for Site-of-Metabo...   \n",
       "18  Generalizations of the Genomic Rank Distance t...   \n",
       "19  wpLogicNet: logic gate and structure inference...   \n",
       "20  ChromDL: A Next-Generation Regulatory DNA Clas...   \n",
       "21  circRNA-sponging: a pipeline for extensive ana...   \n",
       "22  FRETpredict: A Python package for FRET efficie...   \n",
       "23  Snekmer: a scalable pipeline for protein seque...   \n",
       "24  VTAM: A robust pipeline for validating metabar...   \n",
       "25  Swin Unet3D: a three-dimensional medical image...   \n",
       "26  A fair experimental comparison of neural netwo...   \n",
       "27  Detecting early-warning signals for influenza ...   \n",
       "28  GO Bench: Shared-hub for Universal Benchmarkin...   \n",
       "29  DBDIpy: a Python library for processing of unt...   \n",
       "30  Complet+: a computationally scalable method to...   \n",
       "31  Computational Protocol for DNA Methylation Pro...   \n",
       "32  Dynamic Linear Transformer for 3D Biomedical I...   \n",
       "33  TGANet: Text-guided attention for improved pol...   \n",
       "34  Data-Driven Optimization of DIA Mass-Spectrome...   \n",
       "35  Visualizing and exploring patterns of large mu...   \n",
       "36  RTCpredictor: Identification of Read-Through C...   \n",
       "37  DandD: efficient measurement of sequence growt...   \n",
       "38  ViWrap: A modular pipeline to identify, bin, c...   \n",
       "39  Data Imputation for Clinical Trial Emulation: ...   \n",
       "40  Finding phylogeny-aware and biologically meani...   \n",
       "41  PAnno: A pharmacogenomics annotation tool for ...   \n",
       "42  Self-paced Multi-view Learning for CT-based se...   \n",
       "43  Automatic Polyp Segmentation with Multiple Ker...   \n",
       "44  Video Capsule Endoscopy Classification using F...   \n",
       "45  A multistate model and its standalone tool to ...   \n",
       "46  Transport-based morphometry of nuclear structu...   \n",
       "47  A convolutional neural network model for survi...   \n",
       "48  Preclinical-to-clinical Anti-cancer Drug Respo...   \n",
       "49  scMAGS: Marker gene selection from scRNA-seq d...   \n",
       "\n",
       "                                               Phrase  \\\n",
       "0   The source code can be found at \\nhttps://gith...   \n",
       "1   The \\nsource code, data, and instructions to t...   \n",
       "2   Detailed \\ndocumentation and tutorials are pro...   \n",
       "3   VAILABILITY: Analysis pipeline for this approa...   \n",
       "4   Diffsig is implemented as an R package availab...   \n",
       "5   AVAILABILITY AND IMPLEMENTATION: \\nhttps://git...   \n",
       "6   It can also be \\naccessed through https://gith...   \n",
       "7   AVAILABILITY: PFresGO is available for academi...   \n",
       "8   pplications and downstream analyses are \\navai...   \n",
       "9   All dataset and \\narchitecture details are ava...   \n",
       "10  Our code is publicly available at https://gith...   \n",
       "11  \\nHere, we describe polishCLR (https://github....   \n",
       "12  Twenty metabolic genes were \\nidentified by ow...   \n",
       "13  The code is \\navailable at: https://github.com...   \n",
       "14  The proposed method \\nis implemented in an R p...   \n",
       "15  The MoSDeF-GOMC software is available on GitHu...   \n",
       "16  The source code \\nand documentation of MDBuild...   \n",
       "17  AVAILABILITY: A python implementation of the t...   \n",
       "18  AVAILABILITY: Code and instructions available ...   \n",
       "19  AVAILABILITY AND IMPLEMENTATION: The datasets ...   \n",
       "20  AVAILABILITY: The ChromDL source code can be f...   \n",
       "21  AVAILABILITY: https://github.com/biomedbigdata...   \n",
       "22  RETpredict is open \\nsource (GPLv3) and is ava...   \n",
       "23  \\nCode and data used in this article, along wi...   \n",
       "24  VTAM is implemented in Python and available fr...   \n",
       "25  The code can be found at \\nhttps://github.com/...   \n",
       "26  The source code of all experiments is availabl...   \n",
       "27  The source code of this paper can be \\nfreely ...   \n",
       "28             ode is hosted at github.com/mofradlab,   \n",
       "29  rg/project/DBDIpy) or from GitHub \\n(https://g...   \n",
       "30  Code and \\nsupplementary information is availa...   \n",
       "31  The software is available at \\nhttps://github....   \n",
       "32  Code is available \\nhttps://github.com/freshma...   \n",
       "33  Codes are available at \\nhttps://github.com/ni...   \n",
       "34  he source code is \\navailable at: github.com/S...   \n",
       "35  The tool is freely available at \\nhttps://gith...   \n",
       "36  In addition, \\nRTCpredictor ( https://github.c...   \n",
       "37  DandD is open source software available at: \\n...   \n",
       "38  ViWrap is publicly available via GitHub ( \\nht...   \n",
       "39  The \\nexperiment details and code are availabl...   \n",
       "40  A \\nprototype implementation is provided at: \\...   \n",
       "41  PAnno is available at \\nhttps://github.com/Pre...   \n",
       "42  The SA-COVID dataset is publicly released at \\...   \n",
       "43  The code of the proposed \\nMKDCNet is availabl...   \n",
       "44  The code of the proposed FocalConvNet is \\nava...   \n",
       "45  CONCLUSIONS: Our easy-to-use and freely access...   \n",
       "46  The source codes implementing our method is av...   \n",
       "47  The codes are available \\nat the GitHub websit...   \n",
       "48  The code can be accessed at https://github.com...   \n",
       "49  scMAGS is \\nmade freely available at https://g...   \n",
       "\n",
       "                                             Abstract Repo_created_at  \\\n",
       "0   Front Plant Sci. 2023 Jan 31;14:1109314. doi: ...      2022-09-11   \n",
       "1   ArXiv. 2023 Feb 8:arXiv:2302.04313v1. Preprint...      2023-02-09   \n",
       "2   bioRxiv. 2023 Feb 10:2023.02.09.527886. doi: 1...            None   \n",
       "3   bioRxiv. 2023 Feb 12:2023.02.11.528027. doi: 1...      2022-06-17   \n",
       "4   bioRxiv. 2023 Feb 10:2023.02.09.527740. doi: 1...      2022-04-18   \n",
       "5   Bioinformatics. 2023 Feb 16:btad093. doi: 10.1...      2022-07-12   \n",
       "6   Bioinformatics. 2023 Feb 16:btad092. doi: 10.1...      2022-08-01   \n",
       "7   Bioinformatics. 2023 Feb 16:btad094. doi: 10.1...      2022-12-20   \n",
       "8   Bioinformatics. 2023 Feb 16:btad091. doi: 10.1...      2021-04-13   \n",
       "9   Med Phys. 2023 Feb 16. doi: 10.1002/mp.16287. ...      2020-09-27   \n",
       "10  BMC Med Res Methodol. 2023 Feb 15;23(1):41. do...      2022-05-08   \n",
       "11  Genome Biol Evol. 2023 Feb 16:evad020. doi: 10...      2021-06-08   \n",
       "12  World J Mens Health. 2023 Feb 9. doi: 10.5534/...      2018-01-11   \n",
       "13  Comput Biol Med. 2023 Feb 7;155:106635. doi: 1...      2022-06-24   \n",
       "14  Stat Med. 2023 Feb 15. doi: 10.1002/sim.9684. ...            None   \n",
       "15  J Chem Inf Model. 2023 Feb 15. doi: 10.1021/ac...      2022-03-21   \n",
       "16  Brief Bioinform. 2023 Feb 15:bbad057. doi: 10....      2017-10-04   \n",
       "17  Bioinformatics. 2023 Feb 15:btad089. doi: 10.1...      2022-07-19   \n",
       "18  Bioinformatics. 2023 Feb 15:btad087. doi: 10.1...      2022-08-12   \n",
       "19  Bioinformatics. 2023 Feb 15:btad072. doi: 10.1...      2022-07-23   \n",
       "20  bioRxiv. 2023 Jan 29:2023.01.27.525971. doi: 1...      2021-08-02   \n",
       "21  bioRxiv. 2023 Jan 20:2023.01.19.524495. doi: 1...      2021-01-08   \n",
       "22  bioRxiv. 2023 Jan 28:2023.01.27.525885. doi: 1...      2020-08-28   \n",
       "23  Bioinform Adv. 2023 Feb 2;3(1):vbad005. doi: 1...      2020-11-16   \n",
       "24  Comput Struct Biotechnol J. 2023 Jan 27;21:115...            None   \n",
       "25  BMC Med Inform Decis Mak. 2023 Feb 14;23(1):33...      2021-12-21   \n",
       "26  BMC Bioinformatics. 2023 Feb 14;24(1):45. doi:...      2022-07-20   \n",
       "27  Brief Funct Genomics. 2023 Feb 14:elad006. doi...      2022-11-24   \n",
       "28  Bioinformatics. 2023 Feb 14:btad081. doi: 10.1...            None   \n",
       "29  Bioinformatics. 2023 Feb 14:btad088. doi: 10.1...      2022-10-18   \n",
       "30  PeerJ. 2023 Feb 8;11:e14779. doi: 10.7717/peer...      2022-05-24   \n",
       "31  Methods Mol Biol. 2023;2638:23-36. doi: 10.100...      2022-06-17   \n",
       "32  Mach Learn Med Imaging. 2022 Sep;13583:171-180...      2022-06-01   \n",
       "33  Med Image Comput Comput Assist Interv. 2022 Se...      2022-05-09   \n",
       "34  bioRxiv. 2023 Feb 3:2023.02.02.526809. doi: 10...      2018-07-20   \n",
       "35  bioRxiv. 2023 Feb 4:2023.02.03.527015. doi: 10...      2018-08-27   \n",
       "36  bioRxiv. 2023 Feb 3:2023.02.02.526869. doi: 10...      2021-11-09   \n",
       "37  bioRxiv. 2023 Feb 3:2023.02.02.526837. doi: 10...      2022-04-25   \n",
       "38  bioRxiv. 2023 Feb 2:2023.01.30.526317. doi: 10...      2022-09-08   \n",
       "39  medRxiv. 2023 Feb 2:2023.01.29.23285172. doi: ...      2022-03-07   \n",
       "40  bioRxiv. 2023 Feb 3:2023.02.02.526854. doi: 10...      2021-02-17   \n",
       "41  Front Pharmacol. 2023 Jan 26;14:1008330. doi: ...      2022-07-26   \n",
       "42  Biomed Signal Process Control. 2023 May;83:104...      2022-03-23   \n",
       "43  Proc IEEE Int Symp Comput Based Med Syst. 2022...      2022-05-12   \n",
       "44  Proc IEEE Int Symp Comput Based Med Syst. 2022...      2022-05-15   \n",
       "45  Heliyon. 2023 Feb;9(2):e13545. doi: 10.1016/j....            None   \n",
       "46  ArXiv. 2023 Feb 2:arXiv:2302.01449v1. Preprint...            None   \n",
       "47  Lab Invest. 2022 Oct;102(10):1064-1074. doi: 1...      2022-03-23   \n",
       "48  Genomics Proteomics Bioinformatics. 2023 Feb 1...      2021-06-11   \n",
       "49  Comput Biol Med. 2023 Feb 9;155:106634. doi: 1...      2022-03-05   \n",
       "\n",
       "   Repo_updated_at  \n",
       "0       2023-02-16  \n",
       "1       2023-02-15  \n",
       "2             None  \n",
       "3       2023-01-31  \n",
       "4       2023-02-17  \n",
       "5       2023-02-17  \n",
       "6       2022-09-26  \n",
       "7       2023-02-17  \n",
       "8       2023-02-17  \n",
       "9       2023-02-06  \n",
       "10      2023-02-17  \n",
       "11      2023-02-17  \n",
       "12      2022-09-04  \n",
       "13      2022-11-25  \n",
       "14            None  \n",
       "15      2023-02-16  \n",
       "16      2022-10-09  \n",
       "17      2023-02-15  \n",
       "18      2023-02-16  \n",
       "19      2023-02-17  \n",
       "20      2023-02-16  \n",
       "21      2023-01-15  \n",
       "22      2023-02-13  \n",
       "23      2023-02-06  \n",
       "24            None  \n",
       "25      2022-12-15  \n",
       "26      2023-02-13  \n",
       "27      2022-11-24  \n",
       "28            None  \n",
       "29      2023-02-15  \n",
       "30      2022-05-24  \n",
       "31      2022-09-09  \n",
       "32      2022-10-28  \n",
       "33      2023-02-15  \n",
       "34      2023-02-08  \n",
       "35      2023-02-09  \n",
       "36      2022-02-01  \n",
       "37      2023-02-09  \n",
       "38      2023-02-07  \n",
       "39      2022-03-15  \n",
       "40      2022-01-11  \n",
       "41      2022-12-21  \n",
       "42      2022-03-23  \n",
       "43      2022-10-15  \n",
       "44      2022-08-29  \n",
       "45            None  \n",
       "46            None  \n",
       "47      2022-12-01  \n",
       "48      2022-06-29  \n",
       "49      2022-03-05  "
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#pd.options.display.max_colwidth = None\n",
    "df1.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e51f9f8-5318-4972-b166-45f609bde3b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA - ????"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b86bd12e-6d10-4a41-b6c1-45cf0f18cc93",
   "metadata": {},
   "outputs": [],
   "source": [
    "#https://github.com/mofradlab - ?????? (PMID 36786404)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "766c63bb-4a4b-4b24-b8f1-53fa9f1d70db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error with URL: https://api.github.com/repos/idslme/IDSL\n",
      "Status code: 404\n",
      "Error with URL: https://api.github.com/repos/umich-biostatistics/zGPS\n",
      "Status code: 404\n",
      "Error with URL: https://api.github.com/repos/aitgon/vtam, github.com\n",
      "Status code: 404\n",
      "Error with URL: https://api.github.com/repos/mofradlab/\n",
      "Status code: 404\n",
      "Error with URL: https://api.github.com/repos/peterman65/\n",
      "Status code: 404\n",
      "Error with URL: https://api.github.com/repos/rohdelab/nuclear_morphometry\n",
      "Status code: 404\n",
      "Error with URL: https://api.github.com/repos/CTC5/\n",
      "Status code: 404\n"
     ]
    }
   ],
   "source": [
    "df1[\"Repo_created_at\"] = None\n",
    "df1[\"Repo_updated_at\"] = None\n",
    "for i in range(len(df1[\"PMID\"])):\n",
    "    try:\n",
    "        df1[\"Repo_created_at\"].iloc[i], df1[\"Repo_updated_at\"].iloc[i] = pbmd.get_repo_info(str(df1[\"GitHub_link_clean\"].iloc[i]))\n",
    "    except:\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "9c96ac2a-8606-4207-b31c-3110b32b1174",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>Repo_created_at</th>\n",
       "      <th>Repo_updated_at</th>\n",
       "      <th>DOI</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36798707</td>\n",
       "      <td>2023-01-31</td>\n",
       "      <td>2022-09-11</td>\n",
       "      <td>2023-02-16</td>\n",
       "      <td>10.3389/fpls.2023.1109314</td>\n",
       "      <td>https://github.com/syau-miao/DFSP/</td>\n",
       "      <td>Frontiers in plant science</td>\n",
       "      <td>DFSP: A fast and automatic distance field-base...</td>\n",
       "      <td>The source code can be found at \\nhttps://gith...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36798459</td>\n",
       "      <td>2023-02-08</td>\n",
       "      <td>2023-02-09</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>None</td>\n",
       "      <td>https://github.com/BioinfoMachineLearning/bio-...</td>\n",
       "      <td>ArXiv</td>\n",
       "      <td>Geometry-Complete Diffusion for 3D Molecule Ge...</td>\n",
       "      <td>The \\nsource code, data, and instructions to t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36798308</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>10.1101/2023.02.09.527886</td>\n",
       "      <td>https://github.com/idslme/IDSL/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>IDSL.CSA: Composite Spectra Analysis for Chemi...</td>\n",
       "      <td>Detailed \\ndocumentation and tutorials are pro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36798258</td>\n",
       "      <td>2023-02-12</td>\n",
       "      <td>2022-06-17</td>\n",
       "      <td>2023-01-31</td>\n",
       "      <td>10.1101/2023.02.11.528027</td>\n",
       "      <td>https://github.com/gimelbrantlab/controlFreq/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Foreign RNA spike-ins enable accurate allele-s...</td>\n",
       "      <td>VAILABILITY: Analysis pipeline for this approa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36798154</td>\n",
       "      <td>2023-02-10</td>\n",
       "      <td>2022-04-18</td>\n",
       "      <td>2023-02-17</td>\n",
       "      <td>10.1101/2023.02.09.527740</td>\n",
       "      <td>https://github.com/jennprk/diffsig/</td>\n",
       "      <td>bioRxiv : the preprint server for biology</td>\n",
       "      <td>Diffsig: Associating Risk Factors With Mutatio...</td>\n",
       "      <td>Diffsig is implemented as an R package availab...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36740003</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>2022-10-26</td>\n",
       "      <td>2022-10-26</td>\n",
       "      <td>10.1016/j.ab.2023.115075</td>\n",
       "      <td>https://github.com/JYY625/DapNet-HLA/</td>\n",
       "      <td>Analytical biochemistry</td>\n",
       "      <td>DapNet-HLA: Adaptive dual-attention mechanism ...</td>\n",
       "      <td>The codes and \\ndatasets are accessible at htt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36739821</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>2023-01-23</td>\n",
       "      <td>10.1016/j.compbiomed.2023.106615</td>\n",
       "      <td>https://github.com/NBU-CVMI/bx2s-net/</td>\n",
       "      <td>Computers in biology and medicine</td>\n",
       "      <td>BX2S-Net: Learning to reconstruct 3D spinal st...</td>\n",
       "      <td>\\nThe code is available at https://github.com/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36739616</td>\n",
       "      <td>2023-02-05</td>\n",
       "      <td>2022-06-24</td>\n",
       "      <td>2022-07-04</td>\n",
       "      <td>10.1002/gepi.22517</td>\n",
       "      <td>https://github.com/Kevin-walters/lapmapr/</td>\n",
       "      <td>Genetic epidemiology</td>\n",
       "      <td>Bayesian multivariant fine mapping using the L...</td>\n",
       "      <td>R code to implement the Laplace (and Gaussian)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36739333</td>\n",
       "      <td>2023-02-04</td>\n",
       "      <td>2022-07-06</td>\n",
       "      <td>2022-10-21</td>\n",
       "      <td>10.1038/s41437-023-00595-9</td>\n",
       "      <td>https://github.com/ckgriswold/3-locus-autotetr...</td>\n",
       "      <td>Heredity</td>\n",
       "      <td>Meiosis at three loci in autotetraploids: Prob...</td>\n",
       "      <td>All Supplementary Information is \\navailable a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36738608</td>\n",
       "      <td>2023-01-28</td>\n",
       "      <td>2022-07-27</td>\n",
       "      <td>2023-02-15</td>\n",
       "      <td>10.1016/j.cmpb.2023.107374</td>\n",
       "      <td>https://github.com/ubern-mia/pyradise/</td>\n",
       "      <td>Computer methods and programs in biomedicine</td>\n",
       "      <td>PyRaDiSe: A Python package for DICOM-RT-based ...</td>\n",
       "      <td>PyRaDiSe is available on \\nhttps://github.com/...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID     PubDate Repo_created_at Repo_updated_at  \\\n",
       "0   36798707  2023-01-31      2022-09-11      2023-02-16   \n",
       "1   36798459  2023-02-08      2023-02-09      2023-02-15   \n",
       "2   36798308  2023-02-10            None            None   \n",
       "3   36798258  2023-02-12      2022-06-17      2023-01-31   \n",
       "4   36798154  2023-02-10      2022-04-18      2023-02-17   \n",
       "..       ...         ...             ...             ...   \n",
       "95  36740003  2023-02-03      2022-10-26      2022-10-26   \n",
       "96  36739821  2023-02-02      2023-01-23      2023-01-23   \n",
       "97  36739616  2023-02-05      2022-06-24      2022-07-04   \n",
       "98  36739333  2023-02-04      2022-07-06      2022-10-21   \n",
       "99  36738608  2023-01-28      2022-07-27      2023-02-15   \n",
       "\n",
       "                                 DOI  \\\n",
       "0          10.3389/fpls.2023.1109314   \n",
       "1                               None   \n",
       "2          10.1101/2023.02.09.527886   \n",
       "3          10.1101/2023.02.11.528027   \n",
       "4          10.1101/2023.02.09.527740   \n",
       "..                               ...   \n",
       "95          10.1016/j.ab.2023.115075   \n",
       "96  10.1016/j.compbiomed.2023.106615   \n",
       "97                10.1002/gepi.22517   \n",
       "98        10.1038/s41437-023-00595-9   \n",
       "99        10.1016/j.cmpb.2023.107374   \n",
       "\n",
       "                                    GitHub_link_clean  \\\n",
       "0                  https://github.com/syau-miao/DFSP/   \n",
       "1   https://github.com/BioinfoMachineLearning/bio-...   \n",
       "2                     https://github.com/idslme/IDSL/   \n",
       "3       https://github.com/gimelbrantlab/controlFreq/   \n",
       "4                 https://github.com/jennprk/diffsig/   \n",
       "..                                                ...   \n",
       "95              https://github.com/JYY625/DapNet-HLA/   \n",
       "96              https://github.com/NBU-CVMI/bx2s-net/   \n",
       "97          https://github.com/Kevin-walters/lapmapr/   \n",
       "98  https://github.com/ckgriswold/3-locus-autotetr...   \n",
       "99             https://github.com/ubern-mia/pyradise/   \n",
       "\n",
       "                                         Journal  \\\n",
       "0                     Frontiers in plant science   \n",
       "1                                          ArXiv   \n",
       "2      bioRxiv : the preprint server for biology   \n",
       "3      bioRxiv : the preprint server for biology   \n",
       "4      bioRxiv : the preprint server for biology   \n",
       "..                                           ...   \n",
       "95                       Analytical biochemistry   \n",
       "96             Computers in biology and medicine   \n",
       "97                          Genetic epidemiology   \n",
       "98                                      Heredity   \n",
       "99  Computer methods and programs in biomedicine   \n",
       "\n",
       "                                                Title  \\\n",
       "0   DFSP: A fast and automatic distance field-base...   \n",
       "1   Geometry-Complete Diffusion for 3D Molecule Ge...   \n",
       "2   IDSL.CSA: Composite Spectra Analysis for Chemi...   \n",
       "3   Foreign RNA spike-ins enable accurate allele-s...   \n",
       "4   Diffsig: Associating Risk Factors With Mutatio...   \n",
       "..                                                ...   \n",
       "95  DapNet-HLA: Adaptive dual-attention mechanism ...   \n",
       "96  BX2S-Net: Learning to reconstruct 3D spinal st...   \n",
       "97  Bayesian multivariant fine mapping using the L...   \n",
       "98  Meiosis at three loci in autotetraploids: Prob...   \n",
       "99  PyRaDiSe: A Python package for DICOM-RT-based ...   \n",
       "\n",
       "                                               Phrase  \n",
       "0   The source code can be found at \\nhttps://gith...  \n",
       "1   The \\nsource code, data, and instructions to t...  \n",
       "2   Detailed \\ndocumentation and tutorials are pro...  \n",
       "3   VAILABILITY: Analysis pipeline for this approa...  \n",
       "4   Diffsig is implemented as an R package availab...  \n",
       "..                                                ...  \n",
       "95  The codes and \\ndatasets are accessible at htt...  \n",
       "96  \\nThe code is available at https://github.com/...  \n",
       "97  R code to implement the Laplace (and Gaussian)...  \n",
       "98  All Supplementary Information is \\navailable a...  \n",
       "99  PyRaDiSe is available on \\nhttps://github.com/...  \n",
       "\n",
       "[100 rows x 9 columns]"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = df1[[\"PMID\", \"PubDate\", \"Repo_created_at\", \"Repo_updated_at\", \"DOI\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"Journal\", \"Title\", \"Phrase\"]]\n",
    "df1[[\"PMID\", \"PubDate\", \"Repo_created_at\", \"Repo_updated_at\", \"DOI\", \"GitHub_link_clean\", \"Journal\", \"Title\", \"Phrase\"]]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
