{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "from datetime import datetime\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import tqdm as tqdm_module\n",
    "import pbmd_tools as pbmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = \"nadiajuckova@gmail.com\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "IncompleteRead",
     "evalue": "IncompleteRead(1321 bytes read)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_get_chunk_left\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    559\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 560\u001b[1;33m                 \u001b[0mchunk_left\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_read_next_chunk_size\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    561\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_read_next_chunk_size\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    526\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 527\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mline\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m16\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    528\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: invalid literal for int() with base 16: b''",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mIncompleteRead\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_readinto_chunked\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    591\u001b[0m             \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 592\u001b[1;33m                 \u001b[0mchunk_left\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_chunk_left\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    593\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0mchunk_left\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_get_chunk_left\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    561\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 562\u001b[1;33m                 \u001b[1;32mraise\u001b[0m \u001b[0mIncompleteRead\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mb''\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    563\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mchunk_left\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mIncompleteRead\u001b[0m: IncompleteRead(0 bytes read)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mIncompleteRead\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_18460\\2828441492.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mres_esearch\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mreq_esearch\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mreq_esummary\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mesummary\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdb\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"pubmed\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mid\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\",\"\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mres_esearch\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"IdList\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 4\u001b[1;33m \u001b[0mres_esummary\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mreq_esummary\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      5\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mres_esummary\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\Entrez\\__init__.py\u001b[0m in \u001b[0;36mread\u001b[1;34m(handle, validate, escape)\u001b[0m\n\u001b[0;32m    497\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    498\u001b[0m     \u001b[0mhandler\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mDataHandler\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mescape\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 499\u001b[1;33m     \u001b[0mrecord\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mhandler\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    500\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mrecord\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    501\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\Entrez\\Parser.py\u001b[0m in \u001b[0;36mread\u001b[1;34m(self, handle)\u001b[0m\n\u001b[0;32m    369\u001b[0m             \u001b[1;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"file should be opened in binary mode\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    370\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 371\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mParseFile\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    372\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mexpat\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mExpatError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    373\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mStartElementHandler\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36mread\u001b[1;34m(self, amt)\u001b[0m\n\u001b[0;32m    461\u001b[0m             \u001b[1;31m# Amount is given, implement using readinto\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    462\u001b[0m             \u001b[0mb\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbytearray\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 463\u001b[1;33m             \u001b[0mn\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadinto\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    464\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mmemoryview\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mn\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtobytes\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    465\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36mreadinto\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    495\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    496\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchunked\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 497\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_readinto_chunked\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    498\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    499\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mlength\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_readinto_chunked\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    606\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    607\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mIncompleteRead\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 608\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mIncompleteRead\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mbytes\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mtotal_bytes\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    609\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    610\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_safe_read\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mamt\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mIncompleteRead\u001b[0m: IncompleteRead(1321 bytes read)"
     ]
    }
   ],
   "source": [
    "req_esearch = Entrez.esearch(db = \"pubmed\", term = \"github\", retmax = 15000)\n",
    "res_esearch = Entrez.read(req_esearch)\n",
    "req_esummary = Entrez.esummary(db = \"pubmed\", id = \",\".join(res_esearch[\"IdList\"]))\n",
    "res_esummary = Entrez.read(req_esummary)\n",
    "len(res_esummary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame.from_records(res_esummary)\n",
    "df = df[[\"Id\", \"DOI\", \"History\", \"FullJournalName\", \"Title\"]]\n",
    "df[\"PubDate\"] = df[\"History\"].apply(lambda x: x[\"entrez\"] if \"entrez\" in x else None)\n",
    "df[\"PubDate\"] = df[\"PubDate\"].apply(lambda x: datetime.strptime(x, \"%Y/%m/%d %H:%M\").strftime(\"%Y-%m-%d\") if x else None)\n",
    "df = df[[\"Id\", \"DOI\", \"PubDate\", \"FullJournalName\", \"Title\"]]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "module 'pbmd_tools' has no attribute 'get_abstract'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-23-71badf0bda8c>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mid_iter\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m     \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m\"Abstract\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpbmd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_abstract\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mid_iter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m: module 'pbmd_tools' has no attribute 'get_abstract'"
     ]
    }
   ],
   "source": [
    "for id_iter in df[\"Id\"]:  \n",
    "    df.loc[df[\"Id\"] == id_iter,\"Abstract\"] = pbmd.get_abstract(id_iter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Id</th>\n",
       "      <th>DOI</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>FullJournalName</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "      <th>Abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1681</th>\n",
       "      <td>35811294</td>\n",
       "      <td>10.1016/S2589-7500(22)00127-3</td>\n",
       "      <td>2022-07-10</td>\n",
       "      <td></td>\n",
       "      <td>The Lancet. Digital health</td>\n",
       "      <td>Early human judgment forecasts of human monkey...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2522</th>\n",
       "      <td>35337638</td>\n",
       "      <td>10.1016/S2589-7500(22)00032-2</td>\n",
       "      <td>2022-03-26</td>\n",
       "      <td></td>\n",
       "      <td>The Lancet. Digital health</td>\n",
       "      <td>An interactive dashboard to track themes, deve...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3669</th>\n",
       "      <td>34627498</td>\n",
       "      <td>10.1016/S1473-3099(21)00620-4</td>\n",
       "      <td>2021-10-10</td>\n",
       "      <td></td>\n",
       "      <td>The Lancet. Infectious diseases</td>\n",
       "      <td>Implications of suboptimal COVID-19 vaccinatio...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4379</th>\n",
       "      <td>34197809</td>\n",
       "      <td>10.1016/S0140-6736(21)01462-8</td>\n",
       "      <td>2021-07-01</td>\n",
       "      <td></td>\n",
       "      <td>Lancet (London, England)</td>\n",
       "      <td>AZD1222-induced neutralising antibody activity...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4564</th>\n",
       "      <td>34090624</td>\n",
       "      <td>10.1016/S0140-6736(21)01290-3</td>\n",
       "      <td>2021-06-06</td>\n",
       "      <td></td>\n",
       "      <td>Lancet (London, England)</td>\n",
       "      <td>Neutralising antibody activity against SARS-Co...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8479</th>\n",
       "      <td>33937797</td>\n",
       "      <td>10.1148/ryai.2019190072</td>\n",
       "      <td>2021-05-03</td>\n",
       "      <td></td>\n",
       "      <td>Radiology. Artificial intelligence</td>\n",
       "      <td>Magician's Corner: How to Start Learning about...</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8576</th>\n",
       "      <td>31281945</td>\n",
       "      <td>10.1186/s13321-019-0365-4</td>\n",
       "      <td>2019-07-09</td>\n",
       "      <td></td>\n",
       "      <td>Journal of cheminformatics</td>\n",
       "      <td>Journal of Cheminformatics, ORCID, and GitHub.</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Id                            DOI     PubDate GitHub_link_raw  \\\n",
       "1681  35811294  10.1016/S2589-7500(22)00127-3  2022-07-10                   \n",
       "2522  35337638  10.1016/S2589-7500(22)00032-2  2022-03-26                   \n",
       "3669  34627498  10.1016/S1473-3099(21)00620-4  2021-10-10                   \n",
       "4379  34197809  10.1016/S0140-6736(21)01462-8  2021-07-01                   \n",
       "4564  34090624  10.1016/S0140-6736(21)01290-3  2021-06-06                   \n",
       "8479  33937797        10.1148/ryai.2019190072  2021-05-03                   \n",
       "8576  31281945      10.1186/s13321-019-0365-4  2019-07-09                   \n",
       "\n",
       "                         FullJournalName  \\\n",
       "1681          The Lancet. Digital health   \n",
       "2522          The Lancet. Digital health   \n",
       "3669     The Lancet. Infectious diseases   \n",
       "4379            Lancet (London, England)   \n",
       "4564            Lancet (London, England)   \n",
       "8479  Radiology. Artificial intelligence   \n",
       "8576          Journal of cheminformatics   \n",
       "\n",
       "                                                  Title Phrase Abstract  \n",
       "1681  Early human judgment forecasts of human monkey...             NaN  \n",
       "2522  An interactive dashboard to track themes, deve...             NaN  \n",
       "3669  Implications of suboptimal COVID-19 vaccinatio...             NaN  \n",
       "4379  AZD1222-induced neutralising antibody activity...             NaN  \n",
       "4564  Neutralising antibody activity against SARS-Co...             NaN  \n",
       "8479  Magician's Corner: How to Start Learning about...             NaN  \n",
       "8576     Journal of Cheminformatics, ORCID, and GitHub.             NaN  "
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['Abstract'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "module 'pbmd_tools' has no attribute 'get_link_github'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-19-5dbdd4438b22>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mid_iter\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m     \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m\"GitHub_link_raw\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\", \"\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpbmd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_link_github\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m     \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m\"GitHub_link_clean\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\", \"\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpbmd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclean_link\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m\"GitHub_link_raw\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"Id\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"Phrase\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpbmd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_phrase_with_link\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mid_iter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: module 'pbmd_tools' has no attribute 'get_link_github'"
     ]
    }
   ],
   "source": [
    "for id_iter in df[\"Id\"]:\n",
    "    \n",
    "    df.loc[df[\"Id\"] == id_iter,\"GitHub_link_raw\"] = \", \".join(pbmd.get_link_github(df, id_iter))\n",
    "    df.loc[df[\"Id\"] == id_iter,\"GitHub_link_clean\"] = \", \".join(pbmd.clean_link(df.loc[df[\"Id\"] == id_iter,\"GitHub_link_raw\"]))\n",
    "    df.loc[df[\"Id\"] == id_iter, \"Phrase\"] = str(pbmd.get_phrase_with_link(df, id_iter))\n",
    "\n",
    "df = df[[\"Id\", \"DOI\", \"PubDate\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"FullJournalName\", \"Title\", \"Phrase\", \"Abstract\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Id</th>\n",
       "      <th>DOI</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>FullJournalName</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36732776</td>\n",
       "      <td>10.1186/s13040-023-00321-5</td>\n",
       "      <td>2023-02-03</td>\n",
       "      <td>https://github.com/CirculatoryHealth/LoFTK</td>\n",
       "      <td>https://github.com/CirculatoryHealth/LoFTK</td>\n",
       "      <td>BioData mining</td>\n",
       "      <td>LoFTK: a framework for fully automated calculation of predicted Loss-of-Function variants and genes.</td>\n",
       "      <td>LoFTK is an open software and is freely available to non-commercial users at https://github.com/CirculatoryHealth/LoFTK .</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>36732704</td>\n",
       "      <td>10.1186/s12859-023-05158-7</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA</td>\n",
       "      <td>https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA</td>\n",
       "      <td>BMC bioinformatics</td>\n",
       "      <td>GACNNMDA: a computational model for predicting potential human microbe-drug associations based on graph attention network and CNN-based classifier.</td>\n",
       "      <td>Source codes and supplementary materials are available at: https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>36732214</td>\n",
       "      <td>10.1121/10.0016850</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>https://ucrbraingamecenter.github.io/PART_Utilities/</td>\n",
       "      <td>https://ucrbraingamecenter.github.io/PART_Utilities/</td>\n",
       "      <td>The Journal of the Acoustical Society of America</td>\n",
       "      <td>Development and validation of a Spanish-language spatial release from masking task in a Mexican population.</td>\n",
       "      <td>To present the Spanish-language SRM test, we created new recordings of the CRM with Spanish-language Translations and updated the freely available app (PART; https://ucrbraingamecenter.github.io/PART_Utilities/) to present materials in Spanish.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>36731618</td>\n",
       "      <td>10.1016/j.gene.2023.147231</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>https://github.com/almeidasilvaf/SoyPestGCN</td>\n",
       "      <td>https://github.com/almeidasilvaf/SoyPestGCN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>Discovering and prioritizing candidate resistance genes against soybean pests by integrating GWAS and gene coexpression networks.</td>\n",
       "      <td>r/) and an R/Shiny package (https://github.com/almeidasilvaf/SoyPestGCN) that serve as a public resource to explore soybean-pest interactions at the transcriptional level.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36731360</td>\n",
       "      <td>10.1016/j.compbiomed.2023.106585</td>\n",
       "      <td>2023-02-02</td>\n",
       "      <td>https://github.com/CloneRob/ECMT</td>\n",
       "      <td>https://github.com/CloneRob/ECMT</td>\n",
       "      <td>Computers in biology and medicine</td>\n",
       "      <td>Error-Correcting Mean-Teacher: Corrections instead of consistency-targets applied to semi-supervised medical image segmentation.</td>\n",
       "      <td>Code is available at https://github.com/CloneRob/ECMT.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>36699385</td>\n",
       "      <td>10.1093/bioadv/vbac016</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>https://github.com/saezlab/decoupler-py, https://github.com/saezlab/decoupleR_manuscript</td>\n",
       "      <td>https://github.com/saezlab/decoupler-py, https://github.com/saezlab/decoupleR_manuscript</td>\n",
       "      <td>Bioinformatics advances</td>\n",
       "      <td>decoupleR: ensemble of computational methods to infer biological activities from omics data.</td>\n",
       "      <td>tml) for R and in GitHub (https://github.com/saezlab/decoupler-py) for Python.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>36699384</td>\n",
       "      <td>10.1093/bioadv/vbac005</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>https://github.com/AdamaJava/adamajava,</td>\n",
       "      <td>https://github.com/AdamaJava/adamajava</td>\n",
       "      <td>Bioinformatics advances</td>\n",
       "      <td>qmotif: determination of telomere content from whole-genome sequence data.</td>\n",
       "      <td>qmotif is implemented in Java and source code is available at https://github.com/AdamaJava/adamajava, with instructions on how to build and use the application available from https://adamajava.readthedocs.io/en/latest/.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>36699383</td>\n",
       "      <td>10.1093/bioadv/vbac025</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>https://github.com/jchow32/MAGI-MS</td>\n",
       "      <td>https://github.com/jchow32/MAGI-MS</td>\n",
       "      <td>Bioinformatics advances</td>\n",
       "      <td>MAGI-MS: multiple seed-centric module discovery.</td>\n",
       "      <td>MAGI-MS is free and available at https://github.com/jchow32/MAGI-MS. Supplementary data are available at &lt;i&gt;Bioinformatics Advances&lt;/i&gt; online.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>36699382</td>\n",
       "      <td>10.1093/bioadv/vbac056</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>github.com/pipolinlab/ExplorePipolin</td>\n",
       "      <td>https://github.com/pipolinlab/ExplorePipolin</td>\n",
       "      <td>Bioinformatics advances</td>\n",
       "      <td>ExplorePipolin: reconstruction and annotation of piPolB-encoding bacterial mobile elements from draft genomes.</td>\n",
       "      <td>ll code is available and can be accessed here: github.com/pipolinlab/ExplorePipolin.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>36699381</td>\n",
       "      <td>10.1093/bioadv/vbac021</td>\n",
       "      <td>2023-01-26</td>\n",
       "      <td>https://github.com/williamdee1/LMPred_AMP_Prediction</td>\n",
       "      <td>https://github.com/williamdee1/LMPred_AMP_Prediction</td>\n",
       "      <td>Bioinformatics advances</td>\n",
       "      <td>LMPred: predicting antimicrobial peptides using pre-trained language models and deep learning.</td>\n",
       "      <td>All codes are available and can be accessed here: https://github.com/williamdee1/LMPred_AMP_Prediction. Supplementary data are available at &lt;i&gt;Bioinformatics Advances&lt;/i&gt; online.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Id                               DOI     PubDate  \\\n",
       "0   36732776        10.1186/s13040-023-00321-5  2023-02-03   \n",
       "1   36732704        10.1186/s12859-023-05158-7  2023-02-02   \n",
       "2   36732214                10.1121/10.0016850  2023-02-02   \n",
       "3   36731618        10.1016/j.gene.2023.147231  2023-02-02   \n",
       "4   36731360  10.1016/j.compbiomed.2023.106585  2023-02-02   \n",
       "..       ...                               ...         ...   \n",
       "95  36699385            10.1093/bioadv/vbac016  2023-01-26   \n",
       "96  36699384            10.1093/bioadv/vbac005  2023-01-26   \n",
       "97  36699383            10.1093/bioadv/vbac025  2023-01-26   \n",
       "98  36699382            10.1093/bioadv/vbac056  2023-01-26   \n",
       "99  36699381            10.1093/bioadv/vbac021  2023-01-26   \n",
       "\n",
       "                                                                             GitHub_link_raw  \\\n",
       "0                                                 https://github.com/CirculatoryHealth/LoFTK   \n",
       "1                                      https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA   \n",
       "2                                       https://ucrbraingamecenter.github.io/PART_Utilities/   \n",
       "3                                                https://github.com/almeidasilvaf/SoyPestGCN   \n",
       "4                                                           https://github.com/CloneRob/ECMT   \n",
       "..                                                                                       ...   \n",
       "95  https://github.com/saezlab/decoupler-py, https://github.com/saezlab/decoupleR_manuscript   \n",
       "96                                                   https://github.com/AdamaJava/adamajava,   \n",
       "97                                                        https://github.com/jchow32/MAGI-MS   \n",
       "98                                                      github.com/pipolinlab/ExplorePipolin   \n",
       "99                                      https://github.com/williamdee1/LMPred_AMP_Prediction   \n",
       "\n",
       "                                                                           GitHub_link_clean  \\\n",
       "0                                                 https://github.com/CirculatoryHealth/LoFTK   \n",
       "1                                      https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA   \n",
       "2                                       https://ucrbraingamecenter.github.io/PART_Utilities/   \n",
       "3                                                https://github.com/almeidasilvaf/SoyPestGCN   \n",
       "4                                                           https://github.com/CloneRob/ECMT   \n",
       "..                                                                                       ...   \n",
       "95  https://github.com/saezlab/decoupler-py, https://github.com/saezlab/decoupleR_manuscript   \n",
       "96                                                    https://github.com/AdamaJava/adamajava   \n",
       "97                                                        https://github.com/jchow32/MAGI-MS   \n",
       "98                                              https://github.com/pipolinlab/ExplorePipolin   \n",
       "99                                      https://github.com/williamdee1/LMPred_AMP_Prediction   \n",
       "\n",
       "                                     FullJournalName  \\\n",
       "0                                     BioData mining   \n",
       "1                                 BMC bioinformatics   \n",
       "2   The Journal of the Acoustical Society of America   \n",
       "3                                               Gene   \n",
       "4                  Computers in biology and medicine   \n",
       "..                                               ...   \n",
       "95                           Bioinformatics advances   \n",
       "96                           Bioinformatics advances   \n",
       "97                           Bioinformatics advances   \n",
       "98                           Bioinformatics advances   \n",
       "99                           Bioinformatics advances   \n",
       "\n",
       "                                                                                                                                                  Title  \\\n",
       "0                                                  LoFTK: a framework for fully automated calculation of predicted Loss-of-Function variants and genes.   \n",
       "1   GACNNMDA: a computational model for predicting potential human microbe-drug associations based on graph attention network and CNN-based classifier.   \n",
       "2                                           Development and validation of a Spanish-language spatial release from masking task in a Mexican population.   \n",
       "3                     Discovering and prioritizing candidate resistance genes against soybean pests by integrating GWAS and gene coexpression networks.   \n",
       "4                      Error-Correcting Mean-Teacher: Corrections instead of consistency-targets applied to semi-supervised medical image segmentation.   \n",
       "..                                                                                                                                                  ...   \n",
       "95                                                         decoupleR: ensemble of computational methods to infer biological activities from omics data.   \n",
       "96                                                                           qmotif: determination of telomere content from whole-genome sequence data.   \n",
       "97                                                                                                     MAGI-MS: multiple seed-centric module discovery.   \n",
       "98                                       ExplorePipolin: reconstruction and annotation of piPolB-encoding bacterial mobile elements from draft genomes.   \n",
       "99                                                       LMPred: predicting antimicrobial peptides using pre-trained language models and deep learning.   \n",
       "\n",
       "                                                                                                                                                                                                                                                  Phrase  \n",
       "0                                                                                                                              LoFTK is an open software and is freely available to non-commercial users at https://github.com/CirculatoryHealth/LoFTK .  \n",
       "1                                                                                                                                      Source codes and supplementary materials are available at: https://github.com/tyqGitHub/TYQ/tree/master/GACNNMDA.  \n",
       "2   To present the Spanish-language SRM test, we created new recordings of the CRM with Spanish-language Translations and updated the freely available app (PART; https://ucrbraingamecenter.github.io/PART_Utilities/) to present materials in Spanish.  \n",
       "3                                                                            r/) and an R/Shiny package (https://github.com/almeidasilvaf/SoyPestGCN) that serve as a public resource to explore soybean-pest interactions at the transcriptional level.  \n",
       "4                                                                                                                                                                                                 Code is available at https://github.com/CloneRob/ECMT.  \n",
       "..                                                                                                                                                                                                                                                   ...  \n",
       "95                                                                                                                                                                        tml) for R and in GitHub (https://github.com/saezlab/decoupler-py) for Python.  \n",
       "96                           qmotif is implemented in Java and source code is available at https://github.com/AdamaJava/adamajava, with instructions on how to build and use the application available from https://adamajava.readthedocs.io/en/latest/.  \n",
       "97                                                                                                       MAGI-MS is free and available at https://github.com/jchow32/MAGI-MS. Supplementary data are available at <i>Bioinformatics Advances</i> online.  \n",
       "98                                                                                                                                                                  ll code is available and can be accessed here: github.com/pipolinlab/ExplorePipolin.  \n",
       "99                                                                    All codes are available and can be accessed here: https://github.com/williamdee1/LMPred_AMP_Prediction. Supplementary data are available at <i>Bioinformatics Advances</i> online.  \n",
       "\n",
       "[100 rows x 8 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_colwidth = None\n",
    "df[[\"Id\", \"DOI\", \"PubDate\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"FullJournalName\", \"Title\", \"Phrase\"]].head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Id</th>\n",
       "      <th>DOI</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>FullJournalName</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Id, DOI, PubDate, GitHub_link_raw, GitHub_link_clean, FullJournalName, Title, Phrase]\n",
       "Index: []"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[[\"Id\", \"DOI\", \"PubDate\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"FullJournalName\", \"Title\", \"Phrase\"]][(df['GitHub_link_raw'] == '') & (df['Phrase'] == '')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Id</th>\n",
       "      <th>DOI</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>FullJournalName</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Id, DOI, PubDate, GitHub_link_raw, GitHub_link_clean, FullJournalName, Title, Phrase]\n",
       "Index: []"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[[\"Id\", \"DOI\", \"PubDate\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"FullJournalName\", \"Title\", \"Phrase\"]][df['Phrase'] == '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Id</th>\n",
       "      <th>DOI</th>\n",
       "      <th>PubDate</th>\n",
       "      <th>GitHub_link_raw</th>\n",
       "      <th>GitHub_link_clean</th>\n",
       "      <th>FullJournalName</th>\n",
       "      <th>Title</th>\n",
       "      <th>Phrase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Id, DOI, PubDate, GitHub_link_raw, GitHub_link_clean, FullJournalName, Title, Phrase]\n",
       "Index: []"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[[\"Id\", \"DOI\", \"PubDate\", \"GitHub_link_raw\", \"GitHub_link_clean\", \"FullJournalName\", \"Title\", \"Phrase\"]][df['GitHub_link_raw'] == '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('C:/Users/nadia/pubmed_results_10000.tsv', sep='\\t', index=False, columns=df.columns[:-1].tolist())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
